---
title: "Chromosomal curing drives prophage evolution and host epidemiology in pneumococci"
author: "Nicholas Croucher"
date: "2023-07-20"
output:
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      message = FALSE,
                      warning = FALSE,
                      fig.height = 6,
                      fig.width = 8)
library(magrittr)
library(tidyverse)
library(cowplot)
library(ggpubr)
library(genoPlotR)
library(rstatix)
library(magick)
# BiocManager::install("ballgown")
library(ballgown)
library(circlize)
library(gridBase)
library(ComplexHeatmap)
library(gridExtra)
library(kableExtra)
library(ape)
library(phytools)
library(ggtree)
library(tidyverse)
library(rlang)
library(vctrs)

# knitr options
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
theme_set(theme_bw() + theme(strip.background = element_blank()))

### Replacement R functions ###

position_dodge <- function(width = NULL, preserve = "total") {
  ggproto(NULL, PositionDodge,
          width = width,
          preserve = rlang::arg_match0(preserve, c("total", "single"))
  )
}

#' @rdname ggplot2-ggproto
#' @format NULL
#' @usage NULL
#' @export
PositionDodge <- ggproto(
  `_class` = "PositionDodge",
  `_inherit` = ggplot2::Position,
  width = NULL,
  preserve = "total",
  setup_params = function(self, data) {
    flipped_aes <- has_flipped_aes(data)
    data <- flip_data(data, flipped_aes)
    if (is.null(data$xmin) &&
        is.null(data$xmax) && is.null(self$width)) {
      cli::cli_warn(c("Width not defined",
                      "i" = "Set with {.code position_dodge(width = ...)}"))
    }
    
    if (identical(self$preserve, "total")) {
      n <- NULL
    } else {
      panels <- unname(split(data, data$PANEL))
      ns <- vapply(panels,
                   function(panel) {
                     # we need to find the maximum number of times that the
                     # same xmin is used, but making sure we count each group
                     # only once
                     if (is.null(panel$group))
                       return(1)
                     groups <-
                       unname(split(panel, panel$group))
                     xmins <-
                       vapply(groups, function(group)
                         group$xmin[1], double(1))
                     max(table(xmins))
                   },
                   double(1))
      n <- max(ns)
    }
    
    list(width = self$width,
         n = n,
         flipped_aes = flipped_aes)
  },
  
  setup_data = function(self, data, params) {
    data <- flip_data(data, params$flipped_aes)
    if (!"x" %in% names(data) &&
        all(c("xmin", "xmax") %in% names(data))) {
      data$x <- (data$xmin + data$xmax) / 2
    }
    flip_data(data, params$flipped_aes)
  },
  
  compute_panel = function(data, params, scales) {
    data <- flip_data(data, params$flipped_aes)
    collided <- ggplot2:::collide(
      data,
      params$width,
      name = "position_dodge",
      strategy = pos_dodge,
      n = params$n,
      check.width = FALSE
    )
    flip_data(collided, params$flipped_aes)
  }
)

# Dodge overlapping interval.
# Assumes that each set has the same horizontal position.
pos_dodge <- function(df, width, n = NULL) {
  unique_count <- vctrs::vec_unique_count(df$group)
  if (is.null(n)) {
    n <- unique_count
  }
  
  if (n == 1)
    return(df)
  
  if (!all(c("xmin", "xmax") %in% names(df))) {
    df$xmin <- df$x
    df$xmax <- df$x
  }
  
  d_width <- max(df$xmax - df$xmin)
  
  # Have a new group index from 1 to number of groups.
  # This might be needed if the group numbers in this set don't include all of 1:n
  groupidx <- match(df$group, sort(ggplot2:::unique0(df$group)))

  # Find the center for each group, then use that to calculate xmin and xmax
  df$x <- df$x + width * ((groupidx - 0.5) / unique_count - .5)
  df$xmin <- df$x - d_width / n / 2
  df$xmax <- df$x + d_width / n / 2
  
  df
}

#####

mutant_labels <- c(
  # 08B02743
  "08B02743" = ""~phi*"08B02743"~"",
  "08B02743_phageJanus" = ""~phi*'08B02743::Janus'~"",
  "08B02743_intcat" = ""~phi*'08B02743'~italic(int)*'::'*italic(cat)~"",
  "08B02743_repcat" = ""~phi*'08B02743'~italic(rep)*'::'*italic(cat)~"",
  "08B02743_lytAcat" =""~phi*'08B02743'~italic(lyt)*'::'*italic(cat)~"",
  "08B02743_recAJanus" = ""~phi*'08B02743'~italic(recA)*'::Janus'~"",
  "08B02743_Znregcat" = ""~phi*'08B02743'[italic(szaL)*'::'*italic(cat)]~"",
  "08B02743_Znregcat_recA" = ""~phi*'08B02743'[italic(szaL)*'::'*italic(cat)]~italic(recA)*'::Janus'~"",
  "08B02743_immAcat" = ""~phi*'08B02743'~italic(immA)*'::'*italic(cat)~"",
  "08B02743_immAcat_recAJanus" = ""~phi*'08B02743'~italic(immA)*'::'*italic(cat)~italic(recA)*'::Janus'~"",
  # RMV4
  "RMV4" = ""~phi*'RMV4'~"",
  "RMV4_phageJanus" = ""~phi*'RMV4::Janus'~"",
  "RMV4_intcat" = ""~phi*'RMV4'~italic(int)*'::'*italic(cat)~"",
  "RMV4_repcat" = ""~phi*'RMV4'~italic(rep)*'::'*italic(cat)~"",
  "RMV4_lytAcat" = ""~phi*'RMV4'~italic(lyt)*'::'*italic(cat)~"",
  "RMV4_recAJanus" = ""~phi*'RMV4'~italic(recA)*'::Janus'~"",
  "RMV4_Znregcat" = ""~phi*'RMV4'[italic(immARcat)]~"",
  "RMV4_Znregcat_recA" = ""~phi*'RMV4'[italic(immARcat)]~italic(recA)*'::Janus'~"",
  "RMV4_spaLJanus" = ""~phi*'RMV4'~italic(spaL)~'::Janus'~"",
  "RMV4_dprAJanus" = ""~phi*'RMV4'~italic(dprA)~'::Janus'~"",
  # RMV8
  "RMV8" = ""~phi*'RMV8'~"",
  "RMV8_phageJanus" = ""~phi*'RMV8::Janus'~"",
  "RMV8_phageCat" = ""~phi*'RMV8::'*italic(cat)~"",
  "RMV8_intcat" = ""~phi*"RMV8"~italic(int)*'::'*italic(cat)~"",
  "RMV8 int::Jan" = ""~phi*"RMV8"~italic(int)*'::Janus'~"",
  "RMV8_repcat" = ""~phi*"RMV8"~italic(rep)*'::'*italic(cat)~"",
  "RMV8_lytAcat" = ""~phi*"RMV8"~italic(lyt)*'::'*italic(cat)~"",
  "RMV8_recAJanus" = ""~phi*"RMV8"~italic(recA)*'::Janus'~"",
  "RMV8_phageJanus_recAJanus" = ""~phi*"RMV8"~phi*'RMV8::Janus'~italic(recA)*'::Janus'~"",
  "RMV8_intcat_recAJanus" = ""~phi*"RMV8"~italic(int)*'::'*italic(cat)~italic(recA)*'::Janus'~"",
  "RMV8_phage_ermB" = ""~phi*'RMV8::'*italic(ermB)~"",
  "RMV8_intJanus_phage_ermB" = ""~"["*phi*"RMV8"~italic(int)*"::Janus"*"]"*"::"*italic(ermB)~"",
  "RMV8_int_restored" = ""~phi*"RMV8"~italic(int)*'::Janus restored'~"",
  "RMV8_comEA" = ""~phi*"RMV8"~italic(comRA)*'::Janus'~"",
  "RMV8_plac_recA" = ""~phi*'RMV8'~italic(recA)[italic(lacI)]~"",
  # RMV8 hybrid
  "RMV8_hybrid" = ""~phi*'RMV8'[italic(immARcat)]~"",
  "RMV8_hybrid_phage_ermB" = ""~phi*"RMV8"[italic(immARcat)]*"::"*italic(ermB)~"",
  "RMV8_Znregcat" = ""~phi*'RMV8'[italic(immARcat)]~"",
  "RMV8_Znregcat_phageJanus" = ""~phi*'RMV8'[italic(immARcat)]*'::Janus'~"",
  "RMV8_hybrid_intcat" = ""~phi*"RMV8"[italic(immARcat)]~italic(int)*'::'*italic(cat)~"",
  "RMV8_Znregcat_recA" = ""~phi*'RMV8'[italic(immARcat)]~italic(recA)*'::Janus'~"",
  "RMV8_hybrid_recAJanus" = ""~phi*"RMV8"[italic(immARcat)]~italic(recA)*'::Janus'~"",
  "RMV8_hybrid_intJanus_phage_ermB" = ""~"["*phi*"RMV8"[italic(immARcat)]~italic(int)*"::Janus"*"]"*"::"*italic(ermB)~"",
  "RMV8_hybrid_plac_recA" = ""~phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]~"",
  # R6
  "R6" = "R6",
  "R6_dutKO_kanpheS" = "R6"~~italic(scr)*'::Janus',
  "R6_restored" = "R6"~~italic(scr)[italic(attB)]~'restored',
  "R6_mod_RNA" = "R6"~~italic(scr)[italic(attL)],
  # 11930 genotypes
  "11930_delta" = "11930"~Delta*italic(ivr),
  "11930_scRNA_Janus" = "11930"~Delta*italic(ivr)~"[CIPhR"~italic(scr)*"]::Janus",
  "11930_scRNA_mod" = "11930"~Delta*italic(ivr)~italic(scr)[italic(attL)],
  "11930_scRNA_restored" = "11930"~Delta*italic(ivr)~italic(scr)[italic(attB)]
)

mutant_palette <- c(
  # 08B02043
  "08B02743" = "navy",
  "08B02743_phageJanus" = "purple",
  "08B02743_intcat" = "mediumblue",
  "08B02743_repcat" = "royalblue",
  "08B02743_lytAcat" = "steelblue1",
  "08B02743_recAJanus" = "midnightblue",
  "08B02743_Znregcat" = "lightskyblue1",
  "08B02743_Znregcat_recA" = "lightskyblue2",
  "08B02743_immAcat" = 'lightskyblue3',
  "08B02743_immAcat_recAJanus" = 'lightskyblue4',
  # RMV4
  "RMV4" = 'red',
  "RMV4_intcat" = 'darkred',
  "RMV4_lytAcat" = 'tomato',
  "RMV4_phageJanus" = 'orange',
  "RMV4_repcat" = 'coral',
  "RMV4_recAJanus" = 'firebrick4',
  "RMV4_Znregcat" = 'gold4',
  "RMV4_Znregcat_recA" = 'gold',
  "RMV4_spaLJanus" = "tomato",
  "RMV4_dprAJanus" = "tomato",
  # RMV8
  "RMV8" = "darkgreen",
  "RMV8_phageJanus" = "yellowgreen",
  "RMV8_phageCat" = "yellowgreen",
  "RMV8 int::Jan" = "yellowgreen",
  "RMV8_intcat" = "darkolivegreen3",
  "RMV8_repcat" = "limegreen",
  "RMV8_lytAcat" = "lawngreen",
  "RMV8_recAJanus" = "darkolivegreen",
  "RMV8_phageJanus_recAJanus" = "palegreen",
  "RMV8_intcat_recAJanus" = "olivedrab1",
  "RMV8_phage_ermB" = "yellowgreen",
  "RMV8_intJanus_phage_ermB" = "seagreen3",
  "RMV8_int_restored" = "green",
  "RMV8_comEA" = "olivedrab2",
  "RMV8_plac_recA" = "olivedrab3",
  # RMV8 hybrid
  "RMV8_Znregcat" = "aquamarine",
  "RMV8_hybrid_phage_ermB" = "lightblue1",
  "RMV8_Znregcat_recA" = "aquamarine4",
  "RMV8_Znregcat_phageJanus" = "cyan",
  "RMV8_hybrid" = "aquamarine",
  "RMV8_hybrid_intcat" = "turquoise",
  "RMV8_hybrid_recAJanus" = "aquamarine4",
  "RMV8_hybrid_phage_ermB" = "turquoise4",
  "RMV8_hybrid_intJanus_phage_ermB" = "turquoise2",
  "RMV8_hybrid_plac_recA" = "turquoise3",
  # R6
  "R6" = "goldenrod1",
  "R6_dutKO_kanpheS" = "darkorange",
  "R6_restored" = "goldenrod4",
  "R6_mod_RNA" = "brown",
  # 11930 genotypes
  "11930_delta" = "purple",
  "11930_scRNA_Janus" = "violet",
  "11930_scRNA_mod" = "magenta",
  "11930_scRNA_restored" = "blue"
)

condition_palette <- c(
  "None" = "grey",
  "No supplement" = "grey",
  "No mitomycin C" = "grey",
  "No IPTG" = "grey",
  "MMC" = "#8491B4FF",
  "Mitomycin C" = "#8491B4FF",
  "Mitomycin C & CSP" = "#8491B4FF",
  "Low mitomycin C" = "#8491B4FF",
  "High MMC" = "#3C5488FF",
  "High mitomycin C" = "#3C5488FF",
  "CSP" = "#F39B7FFF",
  "CSP & genomic DNA" = "purple",
  "Mitomycin C & CSP & genomic DNA" = "#B09C85FF",
  "Genomic DNA" = "#91D1C2FF",
  "Genomic DNA & IPTG" = "violet",
  "Mitomycin C & IPTG" = "magenta",
  "Mitomycin C & genomic DNA & IPTG" = "yellow",
  "Calcium chloride" = "green",
  "Manganese chloride" = "darkseagreen2",
  "Citrate" = "slateblue1",
  "Hydrogen peroxide" = "yellowgreen",
  "Zinc chloride" = "seagreen2",
  "1mM IPTG" = "peachpuff1",
  "3mM IPTG" = "peachpuff2",
  "6mM IPTG" = "peachpuff3"
)

get_growth_curve_df <- function(df, threshold = 24) {
  df %<>%
      dplyr::rename(any_of(c(Time = "OD600"))) %>%
      tidyr::pivot_longer(cols = -Time,
                          names_to = "Genotype",
                          values_to = "OD600") %>%
      dplyr::mutate(Genotype = gsub("\\.[0-9]$","",Genotype)) %>%
      tidyr::separate(Time,
                      sep = ':',
                      into = c("Hours","Minutes","Seconds")) %>%
      dplyr::mutate(Hours = as.numeric(as.character(Hours))) %>%
      dplyr::mutate(Minutes = dplyr::if_else(Minutes == 30, 0.5, 0)) %>%
      dplyr::mutate(Time = Hours+Minutes) %>%
      dplyr::select(-Hours,-Minutes,-Seconds) %>%
      dplyr::mutate(MMC = dplyr::case_when(
                        grepl("5ng_µl_MMC",Genotype) ~ "Low mitomycin C",
                        grepl("5ng_ul_MMC",Genotype) ~ "Low mitomycin C",
                        grepl("10ng_µl_MMC",Genotype) ~ "High mitomycin C",
                        grepl("10ng_ul_MMC",Genotype) ~ "High mitomycin C",
                        grepl("No_Mitomycin",Genotype) ~ "No mitomycin C",
                        grepl("No_Mitomycin",Genotype) ~ "No mitomycin C",
                        grepl("_no_mitomycin",Genotype) ~ "No mitomycin C",
                        grepl("no_Mitomycin",Genotype) ~ "No mitomycin C",
                        grepl("_No_mitomycin",Genotype) ~ "No mitomycin C",
                        grepl("No_MMC",Genotype) ~ "No mitomycin C",
                        grepl("Mitomycin",Genotype) ~ "Mitomycin C",
                        grepl("mitomycin",Genotype) ~ "Mitomycin C",
                        grepl("MMC",Genotype) ~ "Mitomycin C",
                        TRUE ~ "No mitomycin C"
                      )
                    ) %>%
      dplyr::mutate(gDNA = dplyr::case_when(
                      grepl("withGD",Genotype) ~ "Imported DNA",
                      grepl("no_GD",Genotype) ~ "No imported DNA",
                      grepl("_noGD",Genotype) ~ "No imported DNA",
                      grepl("_GD",Genotype) ~ "Imported DNA",
                      grepl("\\.GD",Genotype) ~ "Imported DNA",
                      TRUE ~ "No imported DNA",
                      )
                    ) %>%
      dplyr::mutate(IPTG = dplyr::case_when(
                      grepl("1mM_IPTG",Genotype) ~ "1mM IPTG",
                      grepl("3mM_IPTG",Genotype) ~ "3mM IPTG",
                      grepl("6mM_IPTG",Genotype) ~ "6mM IPTG",
                      grepl("IPTG",Genotype) ~ "IPTG",
                      TRUE ~ "No IPTG",
                      )
                    ) %>%
      dplyr::mutate(`Growth condition` = dplyr::case_when(
                      grepl("zinc_chloride",Genotype) ~ "Zinc chloride",
                      grepl("Mn_chloride",Genotype) ~ "Manganese chloride",
                      grepl("citrate",Genotype) ~ "Citrate",
                      grepl("hydrogen_peroxide",Genotype) ~ "Hydrogen peroxide",
                      grepl("Ca_chloride",Genotype) ~ "Calcium chloride",
                      grepl("CSP",Genotype) ~ "CSP",
                      IPTG != "No IPTG" ~ "IPTG",
                      MMC != "No mitomycin C" ~ "Mitomycin C",
                      gDNA != "No imported DNA" ~ "Extracellular DNA",
                      TRUE ~ "No supplement",
                      )
                    ) %>%
      dplyr::mutate(Genotype = dplyr::case_when(
                        # 2743 genotypes
                        Genotype == "X2743_ivrtvr" ~ "08B02743",
                        Genotype == "X2743_ivrtvr_intcat" ~ "08B02743_intcat",
                        Genotype == "X2743_ivrtvr_lytAcat" ~ "08B02743_lytAcat",
                        Genotype == "X2743_ivrtvr_phageKO" ~ "08B02743_phageJanus",
                        Genotype == "X2743_ivrtvr_repcat" ~ "08B02743_repcat",
                        grepl("X2743_zn..cat_recA",Genotype) ~ "08B02743_Znregcat_recA",
                        grepl("X2743_recAKO",Genotype) ~ "08B02743_recAJanus",
                        grepl("X2743_zn..cat",Genotype) ~ "08B02743_Znregcat",
                        grepl("X2743.ivrtvr_prophageKO",Genotype) ~ "08B02743_phageJanus",
                        grepl("X2743.ivrtvr_recAKO",Genotype) ~ "08B02743_recAJanus",
                        grepl("X2743.ivrtvr",Genotype) ~ "08B02743",
                        grepl("X2743",Genotype)  ~ "08B02743",
                        # RMV4 genotypes
                        Genotype == "RMV4" ~ "RMV4",
                        Genotype == "RMV4_intcat" ~ "RMV4_intcat",
                        Genotype == "RMV4_lytA" ~ "RMV4_lytAcat",
                        Genotype == "RMV4_phageKO" ~ "RMV4_phageJanus",
                        Genotype == "RMV4_repcat" ~ "RMV4_repcat",
                        grepl("RMV4_prophageJanus_",Genotype)  ~ "RMV4_phageJanus",
                        grepl("RMV4\\.ivr\\.tvrR\\.prophage\\.\\.Janus",Genotype)  ~ "RMV4_phageJanus",
                        grepl("RMV4_Zncat_recAJanus_",Genotype)  ~ "RMV4_Znregcat_recA",
                        grepl("RMV4_Zncat_",Genotype)  ~ "RMV4_Znregcat",
                        grepl("RMV4_recA..Janus_",Genotype)  ~ "RMV4_recAJanus",
                        grepl("RMV4\\.ivr\\.tvrR\\.recA",Genotype)  ~ "RMV4_recAJanus",
                        grepl("RMV4_",Genotype)  ~ "RMV4",
                        grepl("RMV4\\.",Genotype)  ~ "RMV4",
                        # RMV8 genotypes
                        Genotype == "Co" ~ "RMV8",
                        Genotype == "Col4" ~ "RMV8",
                        Genotype == "Col4_intcat" ~ "RMV8_intcat",
                        Genotype == "Col4_lytAcat" ~ "RMV8_lytAcat",
                        Genotype == "Col4_phageKO" ~ "RMV8_phageJanus",
                        Genotype == "Col4_repcat" ~ "RMV8_repcat",
                        grepl("RMV_Hybrid_proermB",Genotype) ~ "RMV8_hybrid_phage_ermB",
                        grepl("RMV_Hybrid_intJan_proermB",Genotype) ~ "RMV8_hybrid_intJanus_phage_ermB",
                        grepl("RMV8_Znregcat_prophage::Janus",Genotype) ~ "RMV8_Znregcat_phageJanus",
                        grepl("RMV8_hybrid_",Genotype)  ~ "RMV8_Znregcat",
                        grepl("RMV8_hybrid\\.",Genotype)  ~ "RMV8_Znregcat",
                        grepl("RMV8_hybrid",Genotype)  ~ "RMV8_Znregcat",
                        grepl("RMV8_prophage::Janus",Genotype) ~ "RMV8_phageJanus",
                        grepl("RMV8_prophageKO_",Genotype) ~ "RMV8_phageJanus",
                        grepl("RMV8_prophageKO_recAJanus",Genotype) ~ "RMV8_phageJanus_recAJanus",
                        grepl("RMV8_recAJanus",Genotype) ~ "RMV8_recAJanus",
                        grepl("RMV8_Znregcat",Genotype) ~ "RMV8_Znregcat",
                        grepl("RMV8_reg2743",Genotype) ~ "RMV8_Znregcat",
                        grepl("RMV8_prophage_KO",Genotype) ~ "RMV8_phageJanus",
                        grepl("RMV8_phageJanus",Genotype) ~ "RMV8_phageJanus",
                        grepl("RMV8\\.proermB",Genotype) ~ "RMV8_phage_ermB",
                        grepl("RMV_intJan_proermB",Genotype) ~ "RMV8_intJanus_phage_ermB",
                        grepl("RMV8_",Genotype) ~ "RMV8",
                        grepl("RMV8\\.",Genotype)  ~ "RMV8",
                        # 11930 genotypes
                        grepl("X11930_ivr_dutkanpheS",Genotype) ~ "11930_scRNA_Janus",
                        grepl("X11930_ivr_modified_csRNA",Genotype) ~ "11930_scRNA_mod",
                        grepl("X11930_2kb_normal_seq",Genotype) ~ "11930_scRNA_restored",
                        grepl("X11930_ivr",Genotype) ~ "11930_delta",
                        # R6
                        grepl("dut_kan_pheS",Genotype) ~ "R6_dutKO_kanpheS",
                        grepl("R6x_dutKO",Genotype) ~ "R6_dutKO_kanpheS",
                        grepl("modified_scRNA",Genotype) ~ "R6_mod_RNA",
                        grepl("R6x_modified",Genotype) ~ "R6_mod_RNA",
                        grepl("scRNA_restored_repeat",Genotype) ~ "R6_restored",
                        grepl("R6x_complemented",Genotype) ~ "R6_restored",
                        grepl("R6x",Genotype) ~ "R6",
                        TRUE ~ Genotype
                      )
                    ) %>%
      dplyr::group_by(Genotype,Time,MMC,gDNA,IPTG,`Growth condition`) %>%
      dplyr::summarise(Median = median(OD600),
                       Min = min(OD600),
                       Max = max(OD600),
                       Observations = list(OD600)
                      ) %>%
      dplyr::ungroup() %>%
      dplyr::mutate(Genotype = factor(Genotype,
                                      levels = names(mutant_labels)[names(mutant_labels) %in% Genotype])) %>%
      dplyr::filter(Time <= threshold)
}

read_excision_file <- function(fn) {
  
  df <-
    read.csv(fn) %>%
    dplyr::mutate(Time = dplyr::case_when(
            grepl("_T0",mutants) ~ "0",
            grepl("_T20",mutants) ~ "20",
            grepl("_T60",mutants) ~ "60",
            grepl("_T120",mutants) ~ "120",
            grepl("T0_",mutants) ~ "0",
            grepl("T20_",mutants) ~ "20",
            grepl("T60_",mutants) ~ "60",
            grepl("T120_",mutants) ~ "120",
            grepl("T0+",mutants) ~ "0",
            grepl("T20+",mutants) ~ "20",
            grepl("T60+",mutants) ~ "60",
            grepl("T120+",mutants) ~ "120",
        )
    ) %>%
    dplyr::mutate(Condition = dplyr::case_when(
            grepl("+CSP",mutants) ~ "CSP",
            grepl("_CSP",mutants) ~ "CSP",
            grepl("+MMC",mutants) ~ "Mitomycin C",
            grepl("_MMC",mutants) ~ "Mitomycin C",
            TRUE ~ "No supplement"
        )
    ) %>%
    dplyr::select(-mutants)
  
}

read_expression_file <- function(fn) {
  
  df <-
    read.csv(fn) %>%
    dplyr::mutate(Time = dplyr::case_when(
            grepl("T0_",mutants) ~ "0",
            grepl("_T0",mutants) ~ "0",
            grepl("T20_",mutants) ~ "20",
            grepl("_T20",mutants) ~ "20",
            grepl("T60_",mutants) ~ "60",
            grepl("_T60",mutants) ~ "60",
            grepl("T120_",mutants) ~ "120",
            grepl("_T120",mutants) ~ "120"
        )
    ) %>%
    dplyr::mutate(Condition = dplyr::case_when(
            grepl("_CSP",mutants) ~ "CSP",
            grepl("_MMC",mutants) ~ "Mitomycin C",
            TRUE ~ "No supplement"
        )
    ) %>%
    dplyr::select(-mutants) %>%
    dplyr::mutate(gene  = dplyr::case_when(
        gene == "RMV8_lytA" ~ "lytA",
        gene == "Imma_2743" ~ "reg",
        gene == "immA" ~ "reg",
        gene == "Lex" ~ "reg",
        gene == "Int_RMV8" ~ "int",
        gene == "dnaC" ~ "rep",
        TRUE ~ gene
      )
    )

}

modRNAPlot <- function (data, ranges = 0, add = FALSE, hl = NULL, seqcols = NULL, 
    seqTF = FALSE, labTF = FALSE, nt = FALSE, dp = 0.5, modspec = FALSE, 
    modp = NULL, mod = NULL, modcol = NULL, tsize = 0.5, main = "", 
    pointSize = 2, lineWd = 2) 
{
    l = 1
    if (seqTF) {
        s <- data$seq
        sequence <- paste(s, collapse = "")
    }
    if (is.null(dim(ranges)) & length(seqcols) < 1) {
        l = 0
    }
    if (is.null(dim(ranges))) {
        ranges <- NULL
        print("default ranges used")
        ranges$min[1] <- 0
        ranges$max[1] <- dim(data)[1]
        ranges$desc[1] <- c("RNA Molecule")
        ranges$col[1] <- -1
        ranges <- as.data.frame(ranges)
    }
    s2 <- dim(data)[1]
    if (length(hl) < 1) {
        print("no sequences to match")
    }
    else {
        hlranges <- NULL
        for (i in 1:length(hl)) {
            t <- strsplit(hl[i], "")
            s1 <- length(t[[1]])
            groups <- NULL
            if (s2 <= s1) {
                groups[[1]] <- as.character(data$seq)
            }
            else {
                for (j in 1:(s2 - s1 + 1)) {
                  groups[[j]] <- as.character(data$seq[j:(j + (s1 - 1))])
                }
            }
            groups <- lapply(groups, paste, collapse = "")
            groups <- unlist(groups)
            ind <- c(1:length(groups))[groups == hl[i]]
            ind <- data$num[ind]
            print(ind)
            hlranges$min <- c(hlranges$min, ind)
            hlranges$max <- c(hlranges$max, (ind + (s1 - 1)))
            hlranges$desc <- c(hlranges$desc, rep(hl[i], length(ind)))
            hlranges$col = c(hlranges$col, rep(seqcols[i], length(ind)))
        }
        print(hlranges)
        hlranges <- as.data.frame(hlranges)
        ranges <- rbind(ranges, hlranges)
    }
    s1 <- max(c(data$x, data$y))
    s2 <- min(c(data$x, data$y))
    if (add == FALSE) {
        plot(data$x, data$y, type = "n", xlab = "", ylab = "", 
            axes = FALSE, xlim = c(s2, s1), ylim = c(s2, s1))
        for (i in data$num) {
            if (data$bound[data$num == i] != -1) {
                lines(c(data$x[data$num == i], data$x[data$num == 
                  data$bound[data$num == i]]), c(data$y[data$num == 
                  i], data$y[data$num == (data$bound[data$num == 
                  i])]), lwd = lineWd)
            }
        }
    }
    else {
        if (add == TRUE) {
            points(data$x, data$y, type = "n")
            for (i in data$num) {
                if (data$bound[data$num == i] != -1) {
                  lines(c(data$x[data$num == i], data$x[data$num == 
                    data$bound[data$num == i]]), c(data$y[data$num == 
                    i], data$y[data$num == (data$bound[data$num == 
                    i])]), lwd = lineWd)
                }
            }
        }
    }
    x <- data$x
    y <- data$y
    if (nt) {
    }
    else {
        lines(x, y, lw = lineWd)
        points(x, y, pch = 16, cex = pointSize)
    }
    if (l == 1) {
        if (labTF) {
            legend(s2[1], s1[1], legend = ranges$desc[ranges$col != 
                0], col = abs(ranges$col[ranges$col != 0]), bty = "n", 
                lty = 1)
        }
    }
    if (nt == TRUE) {
        x1 <- data$x[1]
        x2 <- data$x[2]
        y1 <- data$y[1]
        y2 <- data$y[2]
        vx <- (x1 - x2)
        vy <- (y1 - y2)
        dist <- sqrt(vx^2 + vy^2)
        vx <- vx/dist
        vy <- vy/dist
        text((data$x[1] + (vx * dp)), (data$y[1] + (vy * dp)), 
            "5'", font = 2, cex = tsize)
        x3 <- (-1 * (y1 - y2))
        y3 <- (x1 - x2)
        vx <- x3/dist
        vy <- y3/dist
        text((data$x[1] + (vx * dp)), (data$y[1] + (vy * dp)), 
            data$seq[1], cex = tsize, font = 2)
        t1 <- length(data$x)
        t2 <- (length(data$x) - 1)
        x1 <- data$x[t1]
        x2 <- data$x[t2]
        y1 <- data$y[t1]
        y2 <- data$y[t2]
        vx <- (x1 - x2)
        vy <- (y1 - y2)
        dist <- sqrt(vx^2 + vy^2)
        vx <- vx/dist
        vy <- vy/dist
        text((data$x[t1] + (vx * dp)), (data$y[t1] + (vy * dp)), 
            "3'", cex = tsize, font = 2)
        x3 <- (-1 * (y1 - y2))
        y3 <- (x1 - x2)
        vx <- x3/dist
        vy <- y3/dist
        text((data$x[t1] - (vx * dp)), (data$y[t1] - (vy * dp)), 
            data$seq[t1], font = 2, cex = tsize)
        for (i in 2:(length(data$x) - 1)) {
            x1 <- data$x[(i - 1)]
            x2 <- data$x[(i + 1)]
            y1 <- data$y[(i - 1)]
            y2 <- data$y[(i + 1)]
            x3 <- (-1 * (y1 - y2))
            y3 <- (x1 - x2)
            dist <- sqrt(x3^2 + y3^2)
            x3 <- x3/dist
            y3 <- y3/dist
            text(data$x[i] + (x3 * dp), data$y[i] + (y3 * dp), 
                data$seq[i], font = 2, cex = tsize)
        }
    }
    if (modspec == TRUE) {
        for (i in 1:length(modp)) {
            points(data$x[modp[i]], data$y[modp[i]], pch = mod[i], 
                col = modcol[i])
        }
    }
    for (i in 1:dim(ranges)[1]) {
        keep <- data$num >= ranges$min[i] & data$num <= ranges$max[i]
        if (ranges$col[i] != 0) {
            if (ranges$col[i] == -1) {
            }
            else {
                if (ranges$col[i] == 3) {
                  points(data$x[keep], data$y[keep], col = ranges$col[i], 
                    pch = "x")
                }
                else {
                  points(data$x[keep], data$y[keep], col = ranges$col[i], 
                    pch = 19, cex = pointSize)
                }
            }
        }
    }
    title(main)
}

```

# Intragenomic conflict between prophage and transformation

![](~/Documents/transformationMGE/prophage_deletion/figures/deletion_by_recombination.png)

# The regulation of transformation

![](~/Documents/transformationMGE/prophage_deletion/figures/competence_and_RecA.png)

# The regulation of prophage activation

![](~/Documents/transformationMGE/prophage_deletion/figures/phage_regulation.png)

# What is the distribution of prophage diversity across pneumococci?

# Estimate the number of pneumococci infected with prophage

``` {bash, eval = FALSE}

cd /data/pam/team284/nc3/scratch/GPS/IS_search
source /software/pathogen/etc/pathogen.profile
export LSB_DEFAULT_USERGROUP="team284"
bsub -o phage_blast.o -e phage_blast.e -M 96000 -R 'select[mem>96000] rusage[mem=96000] span[hosts=1]' -n 4 blastn -db gps.mfa -query all_prophage.mfa -out phage_elements.blast.out -num_alignments 1000000  -evalue 0.05 -outfmt 7  -num_threads 4;
perl -lane 'if ($F[0] ne "#") {if ($F[3] > 500) {my @db = split(/\./,$F[1]);print "$F[0]\t$db[1]\t$F[3]\t$F[10]";}}' phage_elements.blast.out > summarised.phage_elements.blast.out
# Add header Query   Database        Length  Evalue
python identify_longest_phage_hits_pandas.py  

```

``` {r Estimate prevalence of phage}

phage_blast_threshold <- 5000

# In the 20047 sequences
phage_hit_lengths.df <-
  read.csv("./sequence_analysis/longest_phage_hits.csv")

phage_hit_length_plot <-
  ggplot(phage_hit_lengths.df,
         aes(x = Length)) +
    geom_density(fill = "blue", alpha = 0.25, linetype = 0) +
    geom_vline(xintercept = phage_blast_threshold, colour = "red", linetype = 3) +
    ylab("Density") +
    theme_bw()

```

# Manual investigation *att* sites

``` {r Manual investigation of att sites}

att_sites.df <-
  data.frame(
    position = c(926553,
                    998998,
                    1704863,
                    231234,
                    350515,
                    1841795,
                    267259,
                    1407631,
                    24066,
                    1833535,
                    24715,
                    1624601,
                    1465604,
                    438893,
                    986969,
                    1713565
                 ),
    name = c("attB[ccrB]", # L
                "attB[hlpA]", # L
                "attB[ssbB]", # R
                "attB[ccnB]", # L
                "attB[cbpG]", # L
                "attB[tgt]", # R
                "attB[ydiL]", # L
                "attB[MM1]", # R
                "attB[OXC]", # L
                "attB[comYC]", # R
                "attB[scRNA]", # L
                "attB[pfbA]", # R
                "attB[relA]", # R
                "attB[pyrG]", # L
                "attB[yjbK]", # L
                "attB[ply]" # R
                ),
    assembly = c("21999_7#175",
                    "22841_2#133",
                    "11893_1#48",
                    "22695_3#282",
                    "15841_5#24",
                    "13710_2#13",
                    "19341_2#124",
                    "ATCC700669",
                    "OXC141",
                    "PMEN2",
                    "21946_6#218",
                    "23164_8#100",
                    "18090_2#20",
                    "21127_1#110",
                    "17175_6#28",
                    "22841_1#183"
                    ),
    accession = c("ERS1813719",
                    "ERS1699606",
                    "ERS367616",
                    "ERS1698859",
                    "ERS638944",
                    "ERS455132",
                    "ERS1021109",
                    "FM211187",
                    "FQ312027",
                    "CP002176",
                    "ERS1460212",
                    "ERS2527991",
                    "ERS740678",
                    "ERS1299787",
                    "ERS708124",
                    "ERS1699476"
                    ),
    description = c("Disrupts ccrB within PPI-1",
                       "Downstream of hlpA",
                       "Disrupts ssbB",
                       "Modifies csRNA2",
                       "Disruption of protease (see CP053210)",
                       "Affects promoter of tgt",
                       "Interrupts ydiL upstream of folP",
                       "Into stop codon of unknown gene",
                       "Alter csRNA3",
                       "Disrupts comYC",
                       "Modifies scRNA",
                       "Disrupts pfbA",
                       "In BOX element upstream of relA",
                       "Within RUP element downstream of pyrG",
                       "Disrupts promoter of yjbK",
                       "Downstream of ply"
                    )
  )

# Variable sites not included:
# - insertion at dnaN in 13154_2#84 - same location as PRCI (may be misassembly)
# - insertion at cps locus in 15995_1#42 (15B/C) - too variable, may be misassemnly due to tnp

```

# Insertion site analysis

We now look at the distribution of MGE insertion sites:

``` {r Prophage insertion sites, fig.width = 18, fig.height = 8}

gps.df <-
  read.csv("./sequence_analysis/gps_info.csv")

phage_insertions.df <-
  read.csv(file = "./sequence_analysis/Spn_att.csv") %>%
    dplyr::mutate(att = paste0(gsub("att_","italic(attB)[italic(",att),")]")) %>%
    dplyr::mutate(att = dplyr::case_when(
                      att == "italic(attB)[italic(OXC)]" ~ "italic(attB)[OXC]",
                      att == "italic(attB)[italic(MM1)]" ~ "italic(attB)[MM1]",
                      att == "italic(attB)[italic(scRNA)]" ~ "italic(attB)[italic(scr)]",
                      TRUE ~ att
                    )
                  ) %>%
    dplyr::filter(att != "italic(attB)[italic(licR)]")

summarise_by_site.df <-
  phage_insertions.df %>%
         dplyr::filter(status == "Infected") %>%
         dplyr::group_by(att) %>%
         dplyr::summarise(Infections = n_distinct(genome)) %>%
         dplyr::ungroup() %>%
         dplyr::arrange(desc(Infections)) %>%
         dplyr::mutate(att=factor(att,levels = unique(att),ordered = T))

phage_by_site_plot <-
  ggplot(summarise_by_site.df,
         aes(x = att,
             y = Infections)) +
    geom_col(fill = "blue") +
    geom_text(aes(label=Infections), vjust=-0.5) +
    scale_x_discrete(labels = function(l) parse(text=l)) +
    xlab(expression(italic(attB)~'site')) +
    ylab('Number of infected sites') +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

```

# Quantify transformation inhibition

``` {r Transformation inhibition by insertion}

transformation_inhibition_sites <- c("italic(attB)[italic(ssbB)]","italic(attB)[italic(comYC)]","italic(attB)[OXC]","italic(attB)[italic(yjbK)]","italic(attB)[italic(scr)]")

transformation_inhibition.df <-
  phage_insertions.df %>%
    # dplyr::filter(status == "Infected") %>%
    # dplyr::mutate(Inhibition = dplyr::if_else(att %in% transformation_inhibition_sites & status == "Infected", "Inhibition", NA_character_)) %>%
    dplyr::group_by(genome) %>%
    dplyr::mutate(Inhibition = dplyr::case_when(
        any(att %in% transformation_inhibition_sites & status == "Infected") ~ 'atop("Inhibition of transformation","by prophage insertion")',
        any(status == "Infected") ~ 'atop("No inhibition of transformation","by prophage insertion")',
        TRUE ~ '"Non-lysogenic"'
      )
    ) %>%
    # tidyr::fill(Inhibition,.direction = "updown") %>%
    # dplyr::mutate(Inhibition = dplyr::if_else(is.na(Inhibition),"No inhibition",Inhibition)) %>%
    dplyr::ungroup() %>%
    dplyr::group_by(Inhibition) %>%
    dplyr::summarise(Isolate_count = n_distinct(genome)) %>%
    dplyr::ungroup()

ggplot(transformation_inhibition.df %>%
         dplyr::select(Inhibition,Isolate_count),
       aes(x = Inhibition,
           y = Isolate_count)) +
  geom_col(colour="blue", fill = "blue") +
  scale_x_discrete(labels = function(l) parse(text=l)) +
  ylab("Isolate count") +
  xlab(expression("Classification of isolate by"~italic(attB)~"status")) +
  geom_text(aes(label=Isolate_count), vjust=-0.5)

```

# Test whether there are any unidentified insertion sites

``` {r Identify new insertion sites}

unidentified_att_sites.df <-
  phage_insertions.df %>%
    dplyr::group_by(genome) %>%
    dplyr::summarise(Disrupted_sites = sum(status == "Infected")) %>%
    dplyr::ungroup() %>%
    dplyr::left_join(
      gps.df %>%
        dplyr::mutate(Lane.Id = gsub("#","_",Lane.Id)) %>%
        dplyr::select(Lane.Id,Strain,Serotype),
      by = c("genome"="Lane.Id")
    ) %>%
    dplyr::left_join(
      phage_hit_lengths.df %>%
        dplyr::group_by(Database) %>%
        dplyr::summarise(Lysogenic = dplyr::if_else(max(Length) > phage_blast_threshold,"Lysogenic","Non-lysogenic")) %>%
        dplyr::ungroup(),
      by = c("genome"="Database")
    )

```

# Number of phage per isolate

``` {r Plot number of phage infections per GPSC}

phage_by_gpsc.df <-
  unidentified_att_sites.df %>%
    dplyr::select(Disrupted_sites,Strain) %>%
    dplyr::rowwise() %>%
    dplyr::mutate(rn = runif(1)) %>%
    dplyr::ungroup() %>%
    dplyr::group_by(Strain) %>%
    dplyr::slice_max(n = 1, order_by = rn) %>%
    dplyr::ungroup() %>%
    dplyr::select(-rn)

max_insertions <- max(unidentified_att_sites.df$Disrupted_sites)

library(fitdistrplus)
fitted_poisson.out <-
  fitdist(data = phage_by_gpsc.df$Disrupted_sites,
          distr = "pois",
          method = "mme")

estimate_lambda <- paste0("lambda~'='~",signif(fitted_poisson.out$estimate,3))

poisson_phage_by_gpsc.df <-
  tibble(
    x = 0:max_insertions,
    y = dpois(x=0:max_insertions,fitted_poisson.out$estimate)
  )

phage_by_isolate_plot <-
  ggplot(phage_by_gpsc.df %>%
           dplyr::group_by(Disrupted_sites) %>%
           dplyr::summarise(Frequency = n()/nrow(phage_by_gpsc.df)) %>%
           dplyr::ungroup(),
         aes(x = Disrupted_sites,
             y = Frequency)
         ) +
    geom_col(colour = "blue", fill = "blue") +
    geom_line(data = poisson_phage_by_gpsc.df,
              aes(x = x,
                  y = y),
              inherit.aes = FALSE,
              linetype = 3,
              colour =  "red") +
    geom_text(x = 4, y = 0.35, label = estimate_lambda, parse = T) +
    xlab(expression(atop("Disrupted"~italic(attB),"sites per isolate"))) +
    #xlim(c(NA,12)) +
    theme_bw() +
    theme(legend.position = "none")

```

# Plot integrase phylogeny

``` {bash Make integrase phylogeny, eval=FALSE}

muscle -align int_sequences.aa -output int_sequences.aln
FastTree int_sequences.aln > int_sequences.tree

```

``` {r Plot integrase phylogeny}

int.tree <-
  ape::read.tree("./sequence_analysis/int_sequences.tree")

int.tree <- phytools::midpoint.root(int.tree)

int.tree$tip.label <- gsub("att_","italic(attB)[italic(",paste0(int.tree$tip.label,")]"))
int.tree$tip.label[int.tree$tip.label == "italic(attB)[italic(MM1)]"] <- "italic(attB)[MM1]"
int.tree$tip.label[int.tree$tip.label == "italic(attB)[italic(OXC)]"] <- "italic(attB)[OXC]"
int.tree$tip.label[int.tree$tip.label == "italic(attB)[italic(scRNA)]"] <- "italic(attB)[italic(scr)]"

int_groupings <-
    list(
      "comYC" = c("italic(attB)[italic(comYC)]","italic(attB)[italic(ccrB)]","italic(attB)[italic(pfbA)]"),
      "ssbB" = c("italic(attB)[italic(ssbB)]"),
      "OXC" = c("italic(attB)[OXC]","italic(attB)[italic(ccnB)]","italic(attB)[italic(ply)]","italic(attB)[italic(relA)]"),
      "MM1" = c("italic(attB)[MM1]","italic(attB)[italic(ydiL)]","italic(attB)[italic(cbpG)]"),
      "scRNA" = c("italic(attB)[italic(scr)]","italic(attB)[italic(pyrG)]"),
      "hlpA" = c("italic(attB)[italic(hlpA)]"),
      "yjbK" = c("italic(attB)[italic(yjbK)]","italic(attB)[italic(tgt)]")
    )

clades <- NULL
mrcas <- NULL

for (i in 1:length(int_groupings)) {
  clades[i] <- names(int_groupings)[i]
  mrcas[i] <- MRCA(int.tree,int_groupings[[i]])
}

clade.df <- tibble(
  "clade" = clades,
  "mrca" = mrcas
)

int_ggtree <-
  ggtree(int.tree, ladderize = TRUE) +
    geom_hilight(data = clade.df,
                 aes(node = mrca,
                     fill = clade),
                 to.bottom = TRUE,
                 extend = 0.5
    ) +
    geom_tiplab(parse = TRUE) +
    geom_treescale(x = 1) +
    theme(legend.position = "none") +
    ggplot2::xlim(0, 2.75)

```

# Tree of the att sites

``` {bash Generate att site, eval=FALSE}

# Set to 25 bp
cd /Users/ncrouche/Documents/transformationMGE/prophage_deletion/bioinformatics
python get_central_att_site.py > central_att_sites.fa
muscle -align central_att_sites.fa -output central_att_sites.aln
FastTree -nt < central_att_sites.aln > central_att_sites.tree

```

``` {r Plot att site tree}

att_tree <- read.tree("./sequence_analysis/central_att_sites.tree")

att_tree <- phytools::midpoint.root(att_tree)

att_tree$tip.label <- gsub("att_","italic(attB)[italic(",paste0(att_tree$tip.label,")]"))
att_tree$tip.label[att_tree$tip.label == "italic(attB)[italic(MM1)]"] <- "italic(attB)[MM1]"
att_tree$tip.label[att_tree$tip.label == "italic(attB)[italic(OXC)]"] <- "italic(attB)[OXC]"
att_tree$tip.label[att_tree$tip.label == "italic(attB)[italic(scRNA)]"] <- "italic(attB)[italic(scr)]"

att_sit_tree <-
  ggtree(att_tree, ladderize = TRUE) +
    geom_tiplab(parse = TRUE) +
    geom_treescale() +
    coord_cartesian(clip="off") +
      ggtree::xlim(0, 3)

```

# Distances of att sites to BOX elements

``` {r Distance to BOX elements, fig.height = 11, fig.width = 8}

box.positions <-
  read.table(file = "./sequence_analysis/R6_BOX_positions.list") %>% dplyr::pull()

box_att_site_distances.df <-
  att_sites.df %>%
    dplyr::rowwise() %>%
    dplyr::mutate(distance_to_BOX = min(abs(position - box.positions))) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(name = gsub("attB\\[","italic(attB)[italic(", gsub("]",")]",name))) %>%
    dplyr::mutate(name = dplyr::case_when(
                      name == "italic(attB)[italic(OXC)]" ~ "italic(attB)[OXC]",
                      name == "italic(attB)[italic(MM1)]" ~ "italic(attB)[MM1]",
                      name == "italic(attB)[italic(srpRNA)]" ~ "italic(attB)[italic(scr)]",
                      name == "italic(attB)[italic(scRNA)]" ~ "italic(attB)[italic(scr)]",
                      TRUE ~ name
                    )
                  ) %>%
  dplyr::mutate(name=factor(name,levels = unique(summarise_by_site.df$att)))

att_BOX_dist_plot <-
  ggplot(box_att_site_distances.df,
         aes(x = name,
             y = distance_to_BOX)) +
    geom_col(fill = "blue") +
    scale_x_discrete(labels = function(l) parse(text=l)) +
    ylab("Distance to nearest\nBOX element") +
    xlab(expression(italic(attB)~"site")) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

cowplot::plot_grid(
  plotlist = list(
    int_ggtree,
    att_sit_tree,
    att_BOX_dist_plot
  ),
  labels = "AUTO",
  nrow = 3
)

```

# What regulators are present on these prophage?

# Identification of regulators using the Massachusetts collection

``` {r Regulators in Massachusetts}

regulator_palette <- c(
  "ImmAR-like" = "royalblue",
  "ImmA-like" = "royalblue",
  "C1-like" = "navyblue",
  "No identified\nregulator" = "darkgrey",
  "Polylysogenic\nisolate" = "lightgrey",
  "C1-like\n& ImmAR-like" = "cornflowerblue"
)

ma_phage.df <-
  read.csv("./sequence_analysis/prophage_regulatory_summary.csv") %>%
    dplyr::mutate(Regulator =
                    dplyr::case_when(
                      regulator == "None" ~ "No identified\nregulator",
                      regulator == "DUF955" ~ "ImmA-like",
                      regulator == "Peptidase_S24" ~ "C1-like"
                    )
    ) %>%
    dplyr::mutate(Regulator = factor(Regulator,
                                     levels = c("No identified\nregulator","C1-like","ImmA-like")))

phage_reg_by_length_plot <-
  ggplot(ma_phage.df,
         aes(x = length,
             fill = Regulator)) +
    geom_density(alpha = 0.25, linetype = 0) +
    xlab("Number of CDSs in prophage") +
    ylab("Density") +
    scale_fill_discrete(type = regulator_palette) +
    theme(legend.position = "bottom")

```

# Searching for prophage regulatory proteins

Each set of streptococcal proteins was then aligned with MAFFT (v7.305b), and hidden Markov model representations of the alignments were generated using HMMer3:

```{bash, eval = FALSE}

# Load HMMer
hmmer/3.2.1=hfc679d8_0-c1

# Also search for manually identified phage regulatory sequences
python get_phage_regulatory_sequences.py # in ~/Documents/phage_biology/regulatory_cogs
mafft phage_DUF955.faa > phage_DUF955.aln
hmmbuild phage_DUF955.hmm phage_DUF955.aln
hmmpress phage_DUF955.hmm

# Per genome analysis 
module load hmmer/3.2.1=hfc679d8_0-c1
module load agat/1.2.0--pl5321hdfd78af_0
f=`cat $1`;
PREFIX=${f%.velvet.gff};
perl -e 'open(my $gff, "<", $ARGV[0]) or die; my $print = 0; while (my $line = readline($gff)) {if ($print == 1) {print $line;} elsif ($line eq "##FASTA\n") {$print = 1;}}' ${PREFIX}.velvet.gff > ${PREFIX}.velvet.fa;
agat_sp_extract_sequences.pl -g ${PREFIX}.velvet.gff -f ${PREFIX}.velvet.fa -t cds -p --keep_attributes --type cds --table 11 --clean_final_stop -o ${PREFIX}.out &> /dev/null;
rm ${PREFIX}.velvet.fa;
perl -lane 'if ($_ =~ /^>/) {my @d = split(/\|/);print ">".$d[$#d]} else {print $_;}' ${PREFIX}.out > ${PREFIX}.aa;
rm ${PREFIX}.out;
hmmsearch -o DUF955_${PREFIX}.hmm_search.out --tblout DUF955_${PREFIX}.hmm_search.tab --noali -E 1e-2 --domE 1e-2 --cpu 1 phage_DUF955.hmm ${PREFIX}.aa;
hmmsearch -o peptidaseS24_${PREFIX}.hmm_search.out --tblout peptidaseS24_${PREFIX}.hmm_search.tab --noali -E 1e-2 --domE 1e-2 --cpu 1 phage_peptidaseS24.hmm ${PREFIX}.aa;

# Run analysis
bsub -q small -J  phage_annotation[1-20047] -o phage_search.%I.o -e phage_search.%I.e -M 2000 -R 'select[mem>2000] rusage[mem=2000]' ./get_gps_proteins.sh annotation.\${LSB_JOBINDEX}

# Combine results
grep -h -v ^# *hmm_search.tab > gps_phage_domains.tab

```

# Read hits from HMM search

This search identifies any match with an e value < 0.01. However, we still expect a large number of false positives at this threshold, because the database contains so many seqences (42,538,047 proteins). We therefore load the hits and plot their p values to decide on a threshold:

``` {r Load and plot HMM results, fig.height = 11, fig.width = 8}

# Write a function to load the HMM hits
process_hmm_hits<-function(fn) {
  hmm_colnames<-c("target name","seq_accession","query name","dom_accession","seq_E-value","seq_score","seq_bias","dom_E-value","dom_score","dom_bias","domn_exp","domn_reg","domn_clu","domn_ov","domn_env","domn_dom","domn_rep","domn_inc","description of target")
  raw.df<-read.table(file=fn,
                             comment.char = "",
                             header = FALSE)

  colnames(raw.df)<-hmm_colnames
  raw.df %<>%
    dplyr::mutate(Rank=rank(`seq_E-value`)) %>%
    dplyr::mutate(Lane.Id = gsub("_([0-9]+)$","",`target name`))
  return(raw.df)
}

# Load the hits with the function
hmm_hits.df <-
  process_hmm_hits("./sequence_analysis/gps_phage_domains.tab") %>%
    dplyr::mutate(
      Regulator = dplyr::case_when(
        `query name` == "phage_DUF955" ~ "ImmAR-like",
        `query name` == "phage_peptidaseS24" ~ "C1-like"
      )
    )

# Specify thresholds
threshold.df<-data.frame(
  "query name" = c("phage_DUF955","phage_peptidaseS24"),
  "threshold" = c(1e-30,1e-30)
)

hmm_hits.df %<>%
  dplyr::left_join(threshold.df, by=c("query name"="query.name"))

# Plot the E values
phage_hmm_plot <-
  ggplot(hmm_hits.df, aes(x=Rank,
                          y=`seq_E-value`,
                          colour = Regulator)) +
    geom_line() +
    scale_y_log10() +
    facet_wrap(~Regulator, scales = "free") +
    ylab("E value of alignment") +
    xlab("Alignment rank by E value") +
    theme(legend.position = "none") +
    geom_hline(aes(yintercept=threshold), linetype = 3) +
    scale_colour_manual(values = regulator_palette)

cowplot::plot_grid(
  plotlist = list(
    phage_reg_by_length_plot,
    phage_hmm_plot
  ),
  nrow = 2,
  labels = "AUTO"
)

```

# Calculate prevalence of prophage and regulators

``` {r Discard false positives}

# Apply threshold
filtered.hmm_hits.df<-
  hmm_hits.df %>%
    dplyr::filter(`seq_E-value` < threshold)

lysogeny_comparison.df <-
  tibble(
    "Method" = c("Prophage BLAST","att site BLAST","Regulator HMM"),
    "Count" = c(
      nrow(phage_hit_lengths.df %>% dplyr::filter(Length > phage_blast_threshold)),
      nrow(unidentified_att_sites.df %>% dplyr::filter(Disrupted_sites > 0)),
      nrow(
        filtered.hmm_hits.df %>%
          dplyr::group_by(Lane.Id) %>%
          dplyr::summarise(Count = n()) %>%
          dplyr::ungroup() %>%
          dplyr::filter(Count > 0)
      )
    )
  )
  
lysogeny_comparison.df %>%
  kableExtra::kable() %>%
  kableExtra::kable_styling(latex_options = "scale_down")

```

# Analyse congruence between analyses

``` {r Congruence between analyses}

congruence.df <-
  unidentified_att_sites.df %>%
    dplyr::left_join(
      filtered.hmm_hits.df %>%
        dplyr::select(Lane.Id,`query name`,`target name`) %>%
        tidyr::pivot_wider(id_cols = c(Lane.Id),
                           names_from = `query name`,
                           values_from = `target name`,
                           values_fn = function(x) sum(!is.na(x)),
                           values_fill = 0
                           ) %>%
        dplyr::mutate(Lane.Id = gsub("#","_",Lane.Id)),
      by = c("genome"="Lane.Id")
    ) %>%
    dplyr::mutate(
      phage_DUF955 = dplyr::if_else(is.na(phage_DUF955),0,phage_DUF955),
      phage_peptidaseS24 = dplyr::if_else(is.na(phage_peptidaseS24),0,phage_peptidaseS24)
    ) %>%
    dplyr::mutate(Regulator = dplyr::if_else(phage_DUF955 == 0 & phage_peptidaseS24 == 0, "No regulator", "Regulator")) %>%
    dplyr::mutate(Summary = dplyr::case_when(
                      Regulator == "Regulator" & Disrupted_sites > 0 & Lysogenic == "Lysogenic" ~ "'Consensus on lysogeny'",
                      Regulator == "No regulator" & Disrupted_sites == 0 & Lysogenic == "Non-lysogenic" ~ "'Consensus on non-lysogeny'",
                      
                      Regulator == "No regulator" & Disrupted_sites == 0 & Lysogenic == "Lysogenic" ~ "atop('Phage with unknown regulator','and'~italic(attB)~'site')",
                      Regulator == "No regulator" & Disrupted_sites > 0 & Lysogenic == "Lysogenic" ~ "'Phage with unknown regulator'",
                      Regulator == "Regulator" & Disrupted_sites == 0 & Lysogenic == "Lysogenic" ~ "'Phage with unknown'~italic(attB)~'site'",
                      
                      Regulator == "Regulator" & Disrupted_sites == 0 & Lysogenic == "Non-lysogenic" ~ "'Orphan prophage regulator'",
                      Regulator == "No regulator" & Disrupted_sites > 0 & Lysogenic == "Non-lysogenic" ~ "'Disrupted'~italic(attB)~'site'",
                      Regulator == "Regulator" & Disrupted_sites > 0 & Lysogenic == "Non-lysogenic" ~ "atop('Undetected phage with known','regulator and'~italic(attB)~'site')",
                      
                      TRUE ~ "'Other'"
                    )
                  ) %>%
    dplyr::mutate(Regulator_count = phage_DUF955 + phage_peptidaseS24)

phage_analysis_congruence_plot <-
  ggplot(congruence.df,
         aes(x = Summary)) +
    geom_bar() +
    geom_text(stat='count', aes(label=..count..), vjust=-0.5, size = 4) +
    ylab("Count") +
    scale_x_discrete(labels = function(l) parse(text=l)) +
    scale_y_continuous(expand = c(0,1000)) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

```


Here we can see some of these domains now have more hits than there are in the genomes, which is more true positives than expected. Hence we may need to make the cutoffs more specific.

# Tally regulators per isolate

``` {r Tally phage regulators per isolate}

phage_gene_distribution.df <-
  filtered.hmm_hits.df %>%
    dplyr::select(Lane.Id,`query name`) %>%
    dplyr::group_by(Lane.Id,`query name`) %>%
    dplyr::summarise(count = n()) %>%
    dplyr::ungroup() %>%
    dplyr::rename(Domain = `query name`) %>%
    dplyr::left_join(
      gps.df,
      by = c("Lane.Id")
    )

# ggplot(phage_gene_distribution.df,
#        aes(x = count,
#            fill = Domain,
#            colour = Domain)) +
#   geom_bar(position = position_dodge(preserve = "single"))

```

# Compare tallies at each site

``` {r Compare phage number estimates, fig.height = 11, fig.width = 8}

phage_count_cor <-cor.test(congruence.df$Disrupted_sites,congruence.df$Regulator_count)$estimate

cor_string = paste0("italic(R)*'='*",signif(phage_count_cor,2))

phage_count_congruence_plot <-
  ggplot(congruence.df %>%
           dplyr::group_by(Disrupted_sites,Regulator_count) %>%
           dplyr::summarise(Frequency = n()) %>%
           dplyr::ungroup(),
         aes(x = Disrupted_sites,
             y = Regulator_count,
             size = Frequency)) +
    geom_point() +
    geom_smooth(data = congruence.df,
                inherit.aes = FALSE,
                aes(x = Disrupted_sites,
                    y = Regulator_count), 
                method = "lm") +
    ylab("Number of identified\nphage regulators") +
    xlab(expression("Number of disrupted"~italic(attB)~"sites")) +
    annotate("text",
             x = 1,
             y = 6.5,
             label = cor_string,
             parse = T)

cowplot::plot_grid(
  plotlist = list(
    phage_hit_length_plot,
    phage_count_congruence_plot + theme(legend.position = "right"),
    phage_analysis_congruence_plot
  ),
  nrow = 3,
  labels = "AUTO",
  rel_heights = c(0.3,0.3,0.4)
)

```

# Type of regulator at each insertion site

``` {r Link regulators and insertion site}

att_sites_domains.df <-
  unidentified_att_sites.df %>%
    dplyr::left_join(phage_gene_distribution.df %>%
                       dplyr::select(Lane.Id,Domain,count) %>%
                       dplyr::mutate(Lane.Id = gsub("#","_",Lane.Id)) %>%
                       tidyr::pivot_wider(id_cols = c(Lane.Id),names_from = Domain,values_from = count,values_fill = 0),
                     by = c("genome" = "Lane.Id")
                     ) %>%
    #dplyr::mutate(Domain = dplyr::if_else(is.na(Domain),"None",Domain)) %>%
    dplyr::mutate(
      Count_ImmR = dplyr::if_else(is.na(phage_DUF955),0,phage_DUF955),
      Count_cI = dplyr::if_else(is.na(phage_peptidaseS24),0,phage_peptidaseS24)
    ) %>%
    dplyr::mutate(reg_domains = dplyr::case_when(
                      Count_ImmR >= 1 & Count_cI >= 1 ~ "C1-like\n& ImmAR-like",
                      Count_ImmR == 0 & Count_cI >= 1 ~ "C1-like",
                      Count_ImmR >= 1 & Count_cI == 0 ~ "ImmAR-like",
                      Count_ImmR == 0 & Count_cI == 0 ~ "No identified\nregulator"
                    )
                  ) %>%
    dplyr::left_join(
      phage_insertions.df %>%
        dplyr::filter(status == "Infected") %>%
        dplyr::group_by(genome) %>%
        dplyr::reframe(infected_sites = list(att)) %>%
        dplyr::distinct() %>%
        dplyr::group_by(genome) %>%
        dplyr::mutate(infected_site_str = paste(infected_sites, collapse=',')) %>%
        dplyr::ungroup(),
      by = c("genome")
    ) %>%
  dplyr::mutate(infected_site_str=factor(infected_site_str,levels = unique(summarise_by_site.df$att)))

phage_summary_per_attB.df <-
  att_sites_domains.df %>%
    dplyr::filter(!is.na(infected_site_str)) %>%
    dplyr::group_by(reg_domains,infected_site_str) %>%
    dplyr::summarise(count = n()) %>%
    dplyr::mutate(att = factor(infected_site_str,
                               levels = levels(summarise_by_site.df$att),ordered = T)) %>%
    tidyr::pivot_wider(id_cols = att, names_from = reg_domains, values_from = count,values_fill = 0) %>%
    dplyr::left_join(summarise_by_site.df, by = c("att")) %>%
    dplyr::mutate(`Polylysogenic\nisolate` = Infections - `C1-like\n& ImmAR-like` - `C1-like` - `ImmAR-like` - `No identified\nregulator`) %>%
    tidyr::pivot_longer(cols = -c(att,Infections), names_to = "Classification", values_to = "count") %>%
    dplyr::mutate(Classification = factor(Classification,
                                          levels = c("No identified\nregulator",
                                                     "Polylysogenic\nisolate",
                                                     "C1-like",
                                                     "ImmAR-like",
                                                     "C1-like\n& ImmAR-like")
                                          )
                  )

guide_nrow = 2
g <- guide_legend(nrow = guide_nrow,
                  title.position = NULL)

regulator_by_site_plot <-
  ggplot(phage_summary_per_attB.df,
           aes(x = att,
               y = count,
               colour = Classification,
               fill = Classification)) +
      geom_col(position = position_stack()) +
      scale_x_discrete(labels = function(l) parse(text=l)) +
      scale_colour_manual(values = regulator_palette, aesthetics = c("fill","colour"), name = "") +
      geom_text(aes(y = Infections, label=Infections), size = 2, colour = "black", vjust=-0.5) +
      xlab(expression(italic(attB)~"site")) +
      ylab("Number of inferred prophage insertions") +
      theme_bw() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
            legend.position = "bottom") +
      guides(fill = g, colour = g)

```

- The missing regulatory systems at the hup site correspond to difficult-to-assemble transposon-like regulators, e.g. see CP035235

# Fisher's exact test on differences across sites

``` {r Differences in regulators across sites}

reg_differences.table <-
  table(
      att_sites_domains.df %>%
        dplyr::filter(Disrupted_sites == 1) %>%
        dplyr::filter(reg_domains %in% c("ImmAR-like","C1-like")) %>%
        dplyr::select(reg_domains,infected_site_str)
  )

chisq.test(reg_differences.table)

```

# How do prophage inhibit competence while integrated?

# Plot the insertion sites in R6

``` {r Insertion sites in chromosome}

# read GTF from Kallisto
# rmv8.granges.list<-gffReadGR("~/Documents/RMS/pv_RNA_seq/RMV8_annotations/SAMEA1463109_ncRNA.gtf",
#                         splitByTranscript = TRUE,
#                         identifier = "gene_id")
r6.granges.list<-gffRead("./sequence_analysis/R6.gff",nrows = 2315)

r6.df <-
    data.frame(
      "chr" = "genome",
      "start" = as.numeric(ifelse(r6.granges.list$start<r6.granges.list$end,r6.granges.list$start,r6.granges.list$end)),
      "end" = as.numeric(ifelse(r6.granges.list$start<r6.granges.list$end,r6.granges.list$end,r6.granges.list$start)),
      "strand" = ifelse(r6.granges.list$strand == "+",1,0)
    )

# instructions: https://jokergoo.github.io/circlize_book/book/

# set variables
segment_widths = 5
track_height = 0.1
rect_height = 5

# annotate MGEs
att.df<-data.frame(
  "chr" = "genome",
  "start" = att_sites.df$position,
  "end" = att_sites.df$position,
  "strand" = 1,
  "gene_id" =  att_sites.df$name,
  "gene_name" = att_sites.df$name,
  "target_id" = att_sites.df$name
)

check_row<-function(x) {
  s<-sum(!is.na(x[8:length(x)]))
  ifelse(s > 0,TRUE,FALSE)
}

# create plot
circos.clear()

# png(file = "~/Documents/RMS/pv_RNA_seq/RMV8_rna_seq/RMV8_rna_seq.png",
#     height = 4.5,
#     width = 4.5,
#     units = "in",
#     res = 600)

plot.new()
circle_size = unit(1, "snpc") # snpc unit gives you a square region
pushViewport(viewport(x = 0,
                      y = 0,
                      width = circle_size,
                      height = circle_size,
    just = c("center", "top")))
par(new = TRUE)
circos.par("start.degree" = 90, "track.margin" = c(0,0))
circos.genomicInitialize(r6.df, plotType = NULL)
# add labels
circos.genomicLabels(
  #filtered_annotation.df %>% dplyr::select(chr,start,end,gene_name),
  #labels = 1,
  att.df %>% dplyr::select(chr,start,end),
  labels = gsub("\\]","",gsub("attB\\[","",att.df %>% dplyr::select(gene_id) %>% pull())),
  #labels.column = 5,
  side = "outside",
  connection_height = 0.01,
  line_col = "black",
  line_lwd = 1,
  cex = 0.7
)
# annotate all CDSs
circos.genomicTrack(
  data = r6.df,
  ylim = c(-1*rect_height,rect_height),
  panel.fun = function(region,value,...) {
    i = getI(...)
    circos.genomicRect(region,
                      value,
                      ytop = ifelse(value[[1]] == 0, 0, rect_height), #c(0,rect_height),
                      ybottom = ifelse(value[[1]] == 0, 0-rect_height, 0),#c(0-rect_height,0),#
                      col = "black", #ifelse(value == 0, "coral2", "blue"),
                      border = "black",
                      ...)
  },
  stack = FALSE,
  numeric.column = "strand",
  bg.border = "white",
  track.height = track_height
)

# title at centre
Lines <- list(bquote(italic("S. pneumoniae\n")),
              bquote(R6))
text(0, -0.1, do.call(expression, Lines), cex = 1)

```

# Effect of integration at *att*comYC

![](~/Documents/transformationMGE/prophage_deletion/figures/attcomYC_effect.png)

# Effect of integration at *att*OXC

![](~/Documents/transformationMGE/prophage_deletion/figures/attcsRNA_effect.png)

# Effect of insertion at *att*srpRNA

``` {r Plot RNA change, message=FALSE,error=FALSE,results='hide',fig.show='hide',plot=FALSE}


mod_ct2coord <- function (input) 
{
  #messages <- validCT(input)
  messages <- c()
  all <- NULL
  if (length(messages) == 0) {
    group <- 0
    firstrun <- 1
    mnp <- min(input$pos)
    map <- max(input$pos)
    a1 <- map + 1
    a2 <- map + 2
    b1 <- mnp - 1
    b2 <- mnp - 2
    new1 <- NULL
    new1$pos[1] <- a1
    new1$before[1] <- (a1 - 1)
    new1$after[1] <- (a1 + 1)
    new1$seq[1] <- "A"
    new1$pos2[1] <- a1
    new1$bound[1] <- b1
    new2 <- NULL
    new2$pos[1] <- a2
    new2$before[1] <- (a2 - 1)
    new2$after[1] <- (a2 + 1)
    new2$seq[1] <- "A"
    new2$pos2[1] <- a2
    new2$bound[1] <- b2
    new3 <- NULL
    new3$pos[1] <- b1
    new3$before[1] <- (b1 - 1)
    new3$after[1] <- (b1 + 1)
    new3$seq[1] <- "A"
    new3$pos2[1] <- b1
    new3$bound[1] <- a1
    new4 <- NULL
    new4$pos[1] <- b2
    new4$before[1] <- (b2 - 1)
    new4$after[1] <- (b2 + 1)
    new4$seq[1] <- "A"
    new4$pos2[1] <- b2
    new4$bound[1] <- a2
    input <- rbind(new1, new2, input, new3, new4)
    input <- as.data.frame(input)
    mnp <- min(input$pos)
    output <- NULL
    nextNT <- input[input$pos == mnp, ]
    if (nextNT$bound != 0) {
      output$pos[1] <- mnp
      output$x[1] <- 0
      output$y[1] <- 0
      output$pos[2] <- nextNT$bound
      output$x[2] <- 0
      output$y[2] <- 1
    }
    stems <- NULL
    stems[[1]] <- c(output$pos[1], output$pos[2])
    j <- 3
    prev <- mnp
    npos <- input$after[input$pos == mnp]
    nextNT <- input[input$pos == npos, ]
    mp <- max(input$pos)
    while (length(stems) > 0) {
      newstems <- NULL
      newloops <- NULL
      ns <- 1
      nl <- 1
      for (i in 1:length(stems)) {
        s1 <- stems[[i]]
        p1 <- s1[1]
        p2 <- s1[2]
        if (firstrun == 1) {
          x3 <- -1
          y3 <- 0
          firstrun <- 0
        }
        else {
          prev <- input$before[input$pos == p1]
          x3 <- output$x[output$pos == prev]
          y3 <- output$y[output$pos == prev]
        }
        x1 <- output$x[output$pos == p1]
        y1 <- output$y[output$pos == p1]
        x2 <- output$x[output$pos == p2]
        y2 <- output$y[output$pos == p2]
        sout <- stemCords(input, p1, p2, x1[1], y1[1], 
          x2[1], y2[1], x3[1], y3[1])
        sdat <- sout[[1]]
        sdat <- as.data.frame(sdat)
        sdat <- sdat[sdat$pos != p1, ]
        sdat <- sdat[sdat$pos != p2, ]
        l <- dim(sdat)[1]
        if (l > 0) {
          s <- length(output$pos)
          s <- s + 1
          for (sd in 1:length(sdat$pos)) {
            output$pos[s] <- sdat$pos[sd]
            output$x[s] <- sdat$x[sd]
            output$y[s] <- sdat$y[sd]
            output$group[s] <- group
            s <- s + 1
          }
          group <- group + 1
        }
        newloops[[nl]] <- sout[[2]]
        nl <- nl + 1
      }
      newstems <- NULL
      for (i in 1:length(newloops)) {
        lps <- newloops[[i]]
        lp1 <- loopLength(input, lps[1])
        if (length(lp1) > 1) {
          nstm <- lp1[[2]]
          for (nt in 1:length(nstm)) {
            newstems[[ns]] <- nstm[[nt]]
            ns <- ns + 1
          }
        }
        pp <- input$before[input$pos == lps[1]]
        lout <- genCords(lp1, lps[1], lps[2], output, 
          1)
        pt <- lp1[[1]]
        tout <- lout
        lpl <- length(pt$pos)
        pt$pos <- pt$pos[lpl:1]
        lout$pos <- pt$pos
        lout <- as.data.frame(lout)
        lout <- lout[lout$pos != lps[1], ]
        lout <- lout[lout$pos != lps[2], ]
        s <- length(output$pos)
        s <- s + 1
        for (lo in 1:length(lout$pos)) {
          output$pos[s] <- lout$pos[lo]
          output$x[s] <- lout$x[lo]
          output$y[s] <- lout$y[lo]
          output$group[s] <- group
          s <- s + 1
        }
        group <- group + 1
      }
      stems <- newstems
    }
    my <- min(output$y) - 5
    xy <- max(output$y) + 5
    mx <- min(output$x) - 5
    xx <- max(output$x) + 5
    output <- as.data.frame(output)
    input <- as.data.frame(input)
    all <- merge(output, input, by = "pos")
    names(all)[1] <- "num"
    all
    rmax <- dim(all)[1]
    rmax <- rmax - 2
    all <- all[3:rmax, ]
  }
  else {
    print(messages)
  }
  all
}


library(RRNA)
library(cowplot)

# from http://rna.tbi.univie.ac.at//cgi-bin/RNAWebSuite/RNAfold.cgi default settings
old.par <- par(no.readonly = TRUE)

par(oma=c(0,0,0,0),mar=c(0,0,0,0))

# Intact scRNA
# short_scrna.cm <- RRNA::makeCt("..(((((((((((((..(((.....((((....(((....)))....))))...)))....))))))).)))))).....", # minimum free energy prediction
#                          "GUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUU") # R6 sequence
# short_scrna.dat<-mod_ct2coord(short_scrna.cm)
# short_scrna.dat$id=1
wt_scrna.cm <- RRNA::makeCt("...(((((((.(((((((..(((.....((((....(((....)))....))))...)))....)))))))...((((((((((((.........)))).........((((....))))...............))))))))((((((((((....)))))))))).............)))))))..", # minimum free energy prediction
                         "AUAGUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUUUCGUGCUUUUUCCGAAUAAAUAAGAUAGAAUAAUCUAGAAUAAAUGAUAAUAGAAAAGAGAAAAUUAUGAAAAUUCGUGGUUUUGAAUUGGUUUCGAGUUUUACAG") # R6 sequence
wt_scrna.dat<-mod_ct2coord(wt_scrna.cm)
wt_scrna.dat$id=1

# Modified scRNA
# modified_scrna.cm <- makeCt("..(((((((((((((..(((.....((((....(((....)))....))))...)))....))))))).))))))......((((.....((((((((((((((........))))))))))))))(((((((((............))))))))).....))))...", # minimum free energy prediction
#                     "GUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUUCGUGCACUUUAAAAACCCUUUAAAAUCAACACUUUAAGGGGUUUUUGUUUGUCUUGUAUAAGAAAAAGGGGCAGACGAGGGGCACAAU") # ATCC 700669 sequence
# modified_scrna.dat<-mod_ct2coord(modified_scrna.cm)
# modified_scrna.dat$id=2
modified_scrna.cm <- makeCt("...(((((((.(((((((..(((.....((((....(((....)))....))))...)))....))))))).(((((((((((((((((...(((((((((((((((........))))))))))))))).......)))))..)))))))))))).....(((((.(((((((((..............))))))))))))))..)))))))..", # minimum free energy prediction
                    "AUAGUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUUCGUGCACUUUAAAAACCCUUUAAAAUCAACACUUUAAGGGGUUUUUGUUUGUCUUGUAUAAGAAAAAGGGGCAGACGAGGGGCACAAUUUAAAUGAAAAUUCGUGGUUUUGAAUUGGUUUCGAGUUUUACAG") # ATCC 700669 sequence
modified_scrna.dat<-mod_ct2coord(modified_scrna.cm)
modified_scrna.dat$id=2

# Bacillus scRNA
bacillus_full_structure <- "(((((......))))).....((((((((((..((..((((((((((....((((((((.....(.(((((((....((.....))....)))))))).....)))))))).((((((.((((((((((...((.((((((((..((((......((((((((((((((((((((((.(((((.....((((....(((....)))....))))...)))))...)))))).)).))))))...))))))))...))))..))))))))))...))))))))))))))))..........))))))))))..))..)))))))))).(((((((((((....)))))))))))."
bacillus_full_sequence <- "GCAUCGUAAUAGAUGCAACAUAAAAUUUGAUAUUUAUUUUGCCGUGCUAAGCGGGGAGGUAGCGGUGCCCUGUACCUGCAAUCCGCUCUAGCAGGGCCGAAUCCCUUCUCGAGGUUCGUUUACUUUAAGGCCUGCCUUAAGUAAGUGGUGUUGACGUUUGGGUCCUGCGCAAUGGGAAUUCAUGAACCAUGUCAGGUCCGGAAGGAAGCAGCAUUAAGUGAAACCUCUCAUGUGCCGCAGGGUUGCCUGGGCCGAGCUAACUGCUUAAGUAACGCUUAGGGUAGCGAAUCGACAGAAGGUGCACGGUAAAUCAAUCAUCAAAUUUUCAGACUCACCUAAUAUUAGGUGAGUUUU"
bacillus_processed_structure <- substr(bacillus_full_structure,38,312)
bacillus_full_sequence <- substr(bacillus_full_sequence,38,312)

bacillus_scrna.cm <- makeCt(bacillus_processed_structure, # minimum free energy prediction
                            bacillus_full_sequence) # B. subtilis 168 (AL009126)
bacillus_scrna.dat<-ct2coord(bacillus_scrna.cm)
bacillus_scrna.dat$id=3

# Align plots
combined_dat <- rbind(wt_scrna.dat,modified_scrna.dat,bacillus_scrna.dat)
#adat=alignCoord(combined_dat,1,floor(length(short_scrna.cm$seq)/2),0,0)
adat=alignCoord(combined_dat,1,floor(length(wt_scrna.cm$seq)/2),0,0)

# Plot unmmodified short RNA
modRNAPlot(adat[adat$id==1,],
           seqcol=c("turquoise"),
           pointSize = 0.5,
           hl = c("AUAGUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUUUCG")
           )
wt_scrna_plot<-recordPlot()
#intact_scrna_gg<-ggdraw(intact_scrna_plot,ylim=c(0.3,0.95),clip='on')
wt_scrna_gg<-ggdraw(wt_scrna_plot,clip='on',ylim=c(0.0,0.65),xlim=c(0.3,1))

# Plot modified RNA
modRNAPlot(adat[adat$id==2,],
           pointSize = 0.5,
           hl = c("AUAGUGGAGCAACAGUUCUGCGUGAAGCGGGUCAGGGGAGGAAUCCAGCAGCCCUAAGCGAUUUGAAUUGUGUGCUCUUUUUUCGU"),
           seqcols = c("turquoise")
           )
modified_scrna_plot<-recordPlot()
#disrupted_scrna_gg<-ggdraw(disrupted_scrna_plot,ylim=c(0.0,0.45),clip='on')
modified_scrna_gg<-ggdraw(modified_scrna_plot,clip='on',ylim=c(0.0,0.75),xlim=c(0.1,1))

# Plot modified RNA
modRNAPlot(adat[adat$id==3,] %>% dplyr::mutate(x = -1*x, y = -1*y),
           seqcol=c("turquoise"),
           pointSize = 0.5,
           hl = c("CCGUGCUAAGCGGGGAGGUAGCGGUGCCCUGUACCUGCAAUCCGCUCUAGCAGGGCCGAAUCCCUUCUCGAGGUUCGUUUACUUUAAGGCCUGCCUUAAGUAAGUGGUGUUGACGUUUGGGUCCUGCGCAAUGGGAAUUCAUGAACCAUGUCAGGUCCGGAAGGAAGCAGCAUUAAGUGAAACCUCUCAUGUGCCGCAGGGUUGCCUGGGCCGAGCUAACUGCUUAAGUAACGCUUAGGGUAGCGAAUCGACAGAAGGUGCACGG")
           )
bacillus_scrna_plot<-recordPlot()
#modified_scrna_gg<-ggdraw(modified_scrna_plot,ylim=c(0.15,0.65),clip='on')
bacillus_scrna_gg<-ggdraw(bacillus_scrna_plot,clip='on',ylim=c(0.3,1),xlim=c(0,0.8))

```

``` {R Actual plot output, fig.height = 11, fig.width = 8}

(three_rna_plot <-
  cowplot::plot_grid(
    plotlist = list(
      #bacillus_scrna_gg,
      wt_scrna_gg,
      modified_scrna_gg
    ),
    nrow = 1,
    labels = c("   i","   ii"),
    #scale = c(0.75,0.5,0.75),
    scale = c(0.75,0.75),
    rel_heights = c(1,1),
    greedy = FALSE
  )
)

par(old.par)

```

# Disruption of *att*srpRNA

``` {r Comparison of scr genes}

scr.df <-
  read.csv(file = "./transformation_data/Modified_scRNA_transformation_final_final_03052024.csv") %>%
    dplyr::filter(mutants != "scRNA_two_SNPs") %>%
    dplyr::mutate(mutants = dplyr::case_when(
          mutants == "R6x" ~ "R6",
          mutants == "dut_KO" ~ "R6_dutKO_kanpheS",
          mutants == "Modified_scRNA" ~ "R6_mod_RNA",
          mutants == "restored_scRNA" ~ "R6_restored"
        )
    ) %>%
    dplyr::mutate(mutants = factor(mutants,
                                   levels = c("R6","R6_dutKO_kanpheS","R6_mod_RNA","R6_restored")))


ncrna_transformation_changes_plot <-
  ggplot(scr.df,
         aes(x = mutants,
             y = Transformation_efficiency,
             colour = mutants,
             fill = mutants)
         ) +
    geom_point(position = position_jitter(width = 0.1)) +
    geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  position = position_dodge(preserve = 'single')) +
    ylab(expression(atop("Transformation efficiency","(10"^{-4}~"transformants per cfu)"))) +
    xlab("") +
    scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% levels(scr.df$mutants)],
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
    scale_y_continuous(trans = "log1p") +
    scale_x_discrete(labels = function(l) parse(text=mutant_labels[match(levels(scr.df$mutants),names(mutant_labels))])) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none") +
    stat_compare_means(method = "wilcox",
                       paired = FALSE,
                       ref.group = "R6",
                       hide.ns = TRUE,
                       label = "p.signif",
                       method.args = list(p.adjust.method = "holm"))

```

# Fitness effect of disrupting *att*srpRNA

``` {r Fitness effect of disrupting srpRNA, fig.width=6}

att_srpRNA_growth.df <-
  get_growth_curve_df(read.csv("./growth_curves/R6x_growth_repeat_titration.csv"))

# plot
r6_scr_growth_plot <-
  ggplot(att_srpRNA_growth.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = Genotype,
               fill = Genotype
            )
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(att_srpRNA_growth.df$Genotype)],
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8))

```

# Can prophage be deleted through transformation?

# Plot the regulator region

``` {bash assemble regulatory region, eval = FALSE}

tracy assemble -r 08B02743_lysogeny.fa -t 6 -o immAR_locus_assembly -f 0.01 Hybrid_catF-IntdownRev-RMV8-Frag1_F.ab1 Hybrid_catF-IntdownRev-RMV8-Frag2_F.ab1
tracy assemble -r RMV8_lysogeny.fa -t 5 -o int_locus_assembly -f 0.01 Hybrid_catF-IntdownRev-RMV8-purA_R.ab1 Hybrid_catF-IntdownRev-RMV8-int_R.ab1
cat immAR_locus_assembly.cons.fa cat.fa int_locus_assembly.cons.fa > RMV8_hybrid_2_contigs.mfa
~/Documents/Scripts/compare_genomes.pl cat.fa RMV8_hybrid_2_contigs.mfa 
# Use ACT to identify the boundaries with the cat gene
cat upstream_immAR_region.fa cat.fa downstream_int_region.fa > RMV8_immARcat_lysogeny.fasta
# Use Artemis to generate RMV8_immARcat_lysogeny.dna
# Use Artemis to annotate manually to generate RMV8_immARcat_lysogeny.embl

```

# Growth curves showing fitness costs

``` {r Plot growth curves of knock outs, fig.width = 8, echo=FALSE}

phage_mutant_growth.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/RMV8_growth_curves.csv")) %>%
      dplyr::mutate(phage = "phi*'RMV8'"),
    get_growth_curve_df(read.csv("./growth_curves/2743_growth_curves.csv")) %>%
      dplyr::mutate(phage = "phi*'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_phageKO_normal_growth_curve.csv")) %>%
      dplyr::mutate(phage = "phi*'RMV4'")
  ) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'")))

# guide legend
guide_nrow = 5
g <- guide_legend(nrow = guide_nrow,
                  size = 0.75,
                  title.position = "left",
                  title = "Genotype")
      
# plot
phage_mutant_growth_plot <-
  ggplot(phage_mutant_growth.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(phage_mutant_growth.df$Genotype)],
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_wrap(phage~.,
                 scales = 'free_x',
                 labeller = label_parsed)

```

# Assaying deletion by transformation

- Measure the ratio of deletions to the acquisition of a base substitution in *rpoB* causing resistance to rifampicin

### Deletion analysis

```{r Revised deletion rates for different mutants, fig.width = 6, echo=FALSE, message = FALSE}

rif_threshold <- 5

deletions.df <-
  dplyr::bind_rows(
    read.csv("./transformation_data/RMV8_rates_deletions.csv") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::rename(Rif.colonies = Rif.colonies,
                    Rif.volume = Volume,
                    Rif.per_1 = per_1,
                    Kan.colonies = Kan_colonies,
                    Kan.volume = Volume.1,
                    Kan.per_1 = per_1.1,
                    Deletion_ratio = Transformation.efficiency),
    read.csv("./transformation_data/2743_rates_deletions_raw.csv") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::rename(Rif.colonies = Number.of.Rif..colonies,
                    Rif.volume = Volume.of.transformants_Rif,
                    Rif.per_1 = per_1,
                    Kan.colonies = Number.of.Kan..colonies,
                    Kan.volume = Volume.of.transformants_Kan,
                    Kan.per_1 = per_1.1,
                    Deletion_ratio = Transformat),
    read.csv("./transformation_data/RMV4_rates_deletions_raw.csv") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::rename(Rif.colonies = Rif.colonies,
                    Rif.volume = Volume,
                    Rif.per_1 = per_1,
                    Kan.colonies = Kan_colonies,
                    Kan.volume = Volume.1,
                    Kan.per_1 = per_1.1,
                    Deletion_ratio = Transformation.efficiency),
    read.csv("./transformation_data/RMV8_Hybrid_int_rates_deletions.csv") %>%
      dplyr::rename(Mutants = Genotyupe) %>%
      dplyr::mutate(phage = dplyr::if_else(grepl("hybrid",Mutants),
                                           "phi*'RMV8'[italic(immARcat)]",
                                           "phi*'RMV8'")
                    ) %>%
      dplyr::rename(Rif.colonies = Rif.number.of.colonies,
                    Rif.volume = Volume..on.rif.plate,
                    Rif.per_1 = Rif.colonies.per.1.microlitres,
                    Kan.colonies = ermB.number.of.colonies,
                    Kan.volume = Volume.on.ery.plate,
                    Kan.per_1 = ery.colonies.per.1.microlitres,
                    Deletion_ratio = deletions),
    read.csv("./transformation_data/2743_additional_deletions.csv") %>%
      dplyr::rename(Mutants = mutants) %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::rename(Rif.colonies = Count_rif,
                    Rif.volume = Vol_rif,
                    Kan.colonies = Count_kan,
                    Kan.volume = Vol_kan) %>%
      dplyr::mutate(Rif.per_1 = Rif.colonies/Rif.volume,
                    Kan.per_1 = Kan.colonies/Kan.volume) %>%
      dplyr::mutate(Deletion_ratio = Kan.per_1/Rif.per_1)
  ) %>%
  dplyr::rename(Genotype = Mutants) %>%
  dplyr::mutate(Genotype = dplyr::case_when(
    # 2743 genotypes
    Genotype == "2743 ivr tvr" ~ "08B02743",
    Genotype == "2743 ivr tvr int::cat" ~ "08B02743_intcat",
    Genotype == "2743 ivr tvr lytA::cat" ~ "08B02743_lytAcat",
    Genotype == "2743 ivr tvr rep::cat " ~ "08B02743_repcat",
    Genotype == "2743 ivr tvr rep::cat" ~ "08B02743_repcat",
    Genotype == "2743 ivr tvr immA::cat" ~ "08B02743_immAcat",
    # RMV4 genotypes
    Genotype == "RMV4 ivr" ~ "RMV4",
    Genotype == "RMV4 ivr tvrR" ~ "RMV4",
    Genotype == "RMV4 ivr int::cat" ~ "RMV4_intcat",
    Genotype == "RMV4 ivr lytA::cat" ~ "RMV4_lytAcat",
    Genotype == "RMV4 ivr tvrR lytA::cat" ~ "RMV4_lytAcat",
    Genotype == "RMV4 ivr rep::cat" ~ "RMV4_repcat",
    Genotype == "RMV4 ivr tvrR int::cat" ~ "RMV4_intcat",
    Genotype == "RMV4 ivr tvrR rep::cat" ~ "RMV4_repcat",
    # RMV8 genotypes
    Genotype == "Col4" ~ "RMV8",
    Genotype == "RMV8 int::Jan" ~ "RMV8 int::Jan",
    Genotype == "RMV8_intcat" ~ "RMV8_intcat",
    Genotype == "Col4_Int_KO" ~ "RMV8_intcat",
    Genotype == "Col4_lytA_KO" ~ "RMV8_lytAcat",
    Genotype == "RMV8_lytA_KO" ~ "RMV8_lytAcat",
    Genotype == "Col4_rep_KO" ~ "RMV8_repcat",
    Genotype == "RMV8_rep_KO" ~ "RMV8_repcat",
    grepl("RMV8 int::Jan _Complemented",Genotype) ~ "RMV8_int_restored",
    grepl("RMV8 int::Jan",Genotype) ~ "RMV8 int::Jan",
    # RMV8 hybrid genotypes
    Genotype == "RMV8 hybrid" ~ "RMV8_hybrid",
    Genotype == "RMV8 hybrid int::Jan" ~ "RMV8_hybrid_intcat",
    TRUE ~ Genotype
  )
  ) %>%
  dplyr::mutate(mutant = dplyr::case_when(
      grepl("rest",Genotype) ~ "int restored",
      grepl("int::Jan",Genotype) ~ "intJan",
      grepl("int",Genotype) ~ "int",
      grepl("lyt",Genotype) ~ "lytA",
      grepl("rep",Genotype) ~ "rep",
      grepl("immA",Genotype) ~ "immA",
      TRUE ~ "WT"
    )
  ) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = names(mutant_labels)[names(mutant_labels) %in% Genotype])) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]"))) %>%
  dplyr::mutate(mutant = factor(mutant,
                                levels = c("WT","int","intJan","immA","rep","lytA","int restored"))) %>%
  dplyr::filter(Rif.colonies >= rif_threshold)

# Plot function
plot_phage_transformation_data <- function(df,var,lab,min_val = 0.1,log_p_increment=1.1) {
  
  y_var <- sym(var)
  
  test_str <- paste0(var," ~ mutant")
  
  # Run test
  test.df <-
    df %>%
      dplyr::group_by(phage) %>%
      dplyr::arrange(Genotype) %>%
      dplyr::mutate(mutant_count = n_distinct(mutant)) %>%
      dplyr::ungroup() %>%
      dplyr::filter(mutant_count > 1) %>%
      dplyr::group_by(phage) %>%
          rstatix::wilcox_test(formula(test_str),
                               ref.group = "WT",
                               paired = FALSE,
                               p.adjust.method = "holm") %>%
          dplyr::ungroup() %>%
          rstatix::add_xy_position(x = "cluster",
                                   dodge = 1,
                                   step.increase = 0.1,
                                   scales = "free_y") %>%
          dplyr::filter(p.adj.signif != "ns") %>%
      dplyr::left_join(
        df %>%
          dplyr::select(phage,Genotype,mutant) %>%
          dplyr::distinct() %>%
          dplyr::rename(named_group1 = Genotype),
        by = c("phage","group1"="mutant")
      ) %>%
      dplyr::left_join(
        df %>%
          dplyr::select(phage,Genotype,mutant) %>%
          dplyr::distinct() %>%
          dplyr::rename(named_group2 = Genotype),
        by = c("phage","group2"="mutant")
      )
  
  if (nrow(test.df) > 0) {
    test.df %<>%
      dplyr::group_by(phage) %>%
      dplyr::arrange(named_group2) %>%
      dplyr::mutate(rank = 1:n(),
                    min_y_pos = min(y.position)) %>%
      #dplyr::mutate(new_y.position = lag(y.position)**log_p_increment) %>%
      dplyr::mutate(new_y.position = log(min_y_pos,10)+(rank-1)*log_p_increment) %>%
      dplyr::mutate(new_y.position = dplyr::if_else(is.na(new_y.position),y.position,new_y.position)) %>%
      dplyr::mutate(y.position = new_y.position) %>%
      dplyr::ungroup()
  }
  
  # Legend
  g <- guide_legend(nrow = 2,
                    title = NULL)
  
  # Process x axis labels
  abbreviations_for_x <-
    df %>%
        dplyr::arrange(phage,Genotype) %>%
        dplyr::select(Genotype) %>%
        dplyr::distinct() %>%
        dplyr::pull()
  
  labels_for_x <- NULL
  
  for (x in abbreviations_for_x) {
      labels_for_x <- c(labels_for_x,mutant_labels[x])
  }
  
  
  labels_for_x[!(grepl(" italic",labels_for_x) | grepl("restored",labels_for_x))] <- bquote('atop(scriptscriptstyle(""),atop(textstyle("Intact"),textstyle("prophage")))')
  labels_for_x[grepl("restored",labels_for_x) & grepl("int",labels_for_x)] <- bquote('atop(scriptscriptstyle(""),atop(textstyle(italic(int)*"::Janus"),textstyle("restored")))')
  labels_for_x <- sub('phi \\* \\S+ ~', '', labels_for_x)
  labels_for_x <- lapply(labels_for_x, FUN = function(x) parse(text=x)[[1]])
  
  naming_vec = labels_for_x
  names(naming_vec) = abbreviations_for_x
  
  # Construct plot
  deletion_plot <-
    ggplot(df,
           aes(x = Genotype,
               y = !!y_var,
               fill = Genotype,
               colour = Genotype)) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  position = position_dodge(preserve = 'single')) +
      geom_point(position = position_jitter(width = 0.25)) +
      geom_hline(yintercept = 1, linetype = 3) +
      scale_y_continuous(trans = "log10",
                         limits = c(min_val,max(df[[var]],
                                                10**(test.df$y.position),
                                                na.rm = TRUE)
                                    )                         ) +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% (deletions.df %>% dplyr::select(Genotype) %>% dplyr::distinct())],
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
      scale_x_discrete(labels = naming_vec) +
      ylab(lab) +
      xlab("Genotype") +
      facet_grid(~phage,
                 scales = 'free_x',
                 space = 'free_x',
                 labeller = label_parsed) +
      theme_bw() +
      theme(legend.position = "none",
            legend.text = element_text(size = 8),
            axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g)
    
  if (nrow(test.df) > 0) {

    deletion_plot <-
      deletion_plot +
        ggpubr::stat_pvalue_manual(test.df,
                                    xmin = "named_group1",
                                    xmax = "named_group2", #y.position = "y.position",xmin="group1",xmax = "group2",
                                    size = 2,
                                    label = "{scales::pvalue(p.adj)}",
                                    tip.length = 0,
                                    inherit.aes = FALSE)

  }
  
  deletion_plot
  
}

deletion_plot <-
  plot_phage_transformation_data(deletions.df,"Deletion_ratio","Ratio of prophage\ndeletions to point mutations",log_p_increment=0.25)

```


# Rif transformation changes

``` {r Rif and kan transformation changes, fig.height = 11, fig.width = 8}

rif_transformation_plot <-
  plot_phage_transformation_data(deletions.df,"Rif.per_1","Rifampicin-resistant\ncolonies per microlitre")

kan_transformation_plot <-
  plot_phage_transformation_data(deletions.df,"Kan.per_1","Kanamycin-resistant\ncolonies per microlitre",log_p_increment=0.25)

cowplot::plot_grid(
  plotlist = list(
    kan_transformation_plot,
    rif_transformation_plot
  ),
  nrow = 2,
  labels = "AUTO"
)

```

# Control deletion rates for *int* mutants

``` {r Growth curve for control genotypes, fig.width = 8, fig.height = 8}

g <- guide_legend(nrow = 2,
                  direction = 'horizontal',
                    title = NULL)

control_growth.df <-
  get_growth_curve_df(read.csv("./growth_curves/RMV8_mutants_pro_ermB_control.csv"))

# Growth curve plots over time
control_growth_curve_plot <-
  ggplot(control_growth.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(control_growth.df$Genotype)],
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8)) +
      guides(colour = g, fill = g)


control_deletions.df <-
  read.csv("./transformation_data/control_prophageKO.csv") %>%
    dplyr::filter(Rif.number.of.colonies >= 5) %>%
    dplyr::mutate(Genotype_colour = dplyr::case_when(
        Genotype == "RMV8  (prophage::ermB)" ~ "RMV8_phage_ermB",
        Genotype == "RMV8 integrase::Janus (prophage::ermB)" ~ "RMV8_intJanus_phage_ermB",
        Genotype == "RMV8 hybrid zn::cat (prophage::ermB)" ~ "RMV8_hybrid_phage_ermB",
        Genotype == "RMV8 hybrid zn::cat integrase::Janus(prophage::ermB)" ~ "RMV8_hybrid_intJanus_phage_ermB",
        TRUE ~ Genotype
      )
    ) %>%
    dplyr::mutate(
      Genotype = dplyr::case_when(
        Genotype == "RMV8  (prophage::ermB)" ~ '""~phi*"RMV8"*"::"*italic(ermB)~""',
        Genotype == "RMV8 integrase::Janus (prophage::ermB)" ~ '""~"["*phi*"RMV8"~italic(int)*"::Janus"*"]"*"::"*italic(ermB)~""',
        Genotype == "RMV8 hybrid zn::cat (prophage::ermB)" ~ '""~phi*"RMV8"[italic(immARcat)]*"::"*italic(ermB)~""',
        Genotype == "RMV8 hybrid zn::cat integrase::Janus(prophage::ermB)" ~ '""~"["*phi*"RMV8"[italic(immARcat)]~italic(int)*"::Janus"*"]::"*italic(ermB)~""',
        TRUE ~ Genotype
      )
    ) %>%
    dplyr::mutate(Genotype = factor(Genotype,
                                    levels = c('""~phi*"RMV8"*"::"*italic(ermB)~""',
                                               '""~"["*phi*"RMV8"~italic(int)*"::Janus"*"]"*"::"*italic(ermB)~""',
                                               '""~phi*"RMV8"[italic(immARcat)]*"::"*italic(ermB)~""',
                                               '""~"["*phi*"RMV8"[italic(immARcat)]~italic(int)*"::Janus"*"]::"*italic(ermB)~""'
                                               )
                                    )
    )

int_mutant_control_deletions_plot <-
  ggplot(control_deletions.df,
         aes(x = Genotype,
             y = deletions,
             colour = Genotype_colour,
             fill = Genotype_colour)) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                scale = "width",
                alpha = 0.15,
                position = position_dodge(preserve = 'single')) +
    geom_point(position = position_jitter(width = 0.1)) +
    ylab("Ratio of prophage\ndeletions to point mutations") +
    scale_y_continuous(trans = "log10") +
    scale_x_discrete(labels = function(l) parse(text=levels(control_deletions.df$Genotype))) +
    scale_colour_manual(values = mutant_palette[unique(control_deletions.df$Genotype_colour)],
                        aesthetics = c("fill","colour")) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none")

cowplot::plot_grid(
  plotlist = list(
    cowplot::plot_grid(
      plotlist = list(
        control_growth_curve_plot + theme(legend.position = "none"),
        int_mutant_control_deletions_plot + theme(legend.position = "none")
      ),
      rel_widths = c(0.6,0.4),
      labels = "AUTO"
    ),
    cowplot::get_plot_component(control_growth_curve_plot + guides(colour = g, fill = g) + theme(legend.position = "bottom"), 'guide-box-bottom', return_all = TRUE)
  ),
  nrow = 2,
  rel_heights = c(0.8,0.2)
)

```

# Deletion ratios with active *ivr* and *tvr* loci

``` {r Active pvRMS ratios, fig.height = 11, fig.width = 8}

active_pvrms_ratios.df <-
  read.csv("./transformation_data/rates_of_deletion_Ion_Aug_2022.csv") %>%
    dplyr::filter(Genotype != "RMV4_2743Φ") %>%
    dplyr::mutate(Genotype = dplyr::if_else(Genotype == "2743","08B02743",Genotype)) %>%
    dplyr::mutate(phage = dplyr::case_when(
                      grepl("2743",Genotype) ~ "phi*'08B02743'~italic(ivr)^'+'~italic(tvr)^'+'",
                      grepl("RMV4",Genotype) ~ "phi*'RMV4'~italic(ivr)^'+'~italic(tvr)^'+'",
                      grepl("RMV8",Genotype) ~ "phi*'RMV8'~italic(ivr)^'+'~italic(tvr)^'+'"
                    )
                  ) %>%
    dplyr::mutate(mutant = dplyr::case_when(
                      grepl("lytA",Genotype) ~ "lytA",
                      grepl("int",Genotype) ~ "int",
                      grepl("rep",Genotype) ~ "rep",
                      TRUE ~ "WT"
                    )
                  ) %>%
    dplyr::filter(Rifampicin >= rif_threshold) %>%
    dplyr::mutate(phage = factor(phage,
                                 levels = c("phi*'RMV8'~italic(ivr)^'+'~italic(tvr)^'+'",
                                            "phi*'08B02743'~italic(ivr)^'+'~italic(tvr)^'+'",
                                            "phi*'RMV4'~italic(ivr)^'+'~italic(tvr)^'+'")
                                 )
                  )

# active_pvrms_deletions_plot <-
#   plot_phage_transformation_data(active_pvrms_ratios.df,"Ratio","Ratio of prophage\ndeletions to point mutations",min_val = 0.04)

pvrms_versions.df <-
  dplyr::bind_rows(
    read.csv("./transformation_data/2743_tvrR_rates_deletions.csv") %>%
      dplyr::mutate(mutant = dplyr::case_when(
          Genotype == "2743 ivr tvrR" ~ "WT",
          Genotype == "2743 ivr tvrR int::cat" ~ "int",
          Genotype == "2743 ivr tvrR lytA::cat" ~ "lytA",
          Genotype == "2743 ivr tvrR rep::cat" ~ "rep"
        )
      ) %>%
      dplyr::mutate(Genotype = dplyr::case_when(
          Genotype == "2743 ivr tvrR" ~ "08B02743",
          Genotype == "2743 ivr tvrR int::cat" ~ "08B02743_intcat",
          Genotype == "2743 ivr tvrR lytA::cat" ~ "08B02743_lytAcat",
          Genotype == "2743 ivr tvrR rep::cat" ~ "08B02743_repcat"
        )
      ) %>%
      dplyr::mutate(phage = 'phi*"08B02743"~italic(ivr)^"-"~"fixed"~italic(tvr)'),
    active_pvrms_ratios.df %>%
      dplyr::filter(grepl("08B02743",Genotype)) %>%
      dplyr::mutate(phage = 'phi*"08B02743"~italic(ivr)^"+"~italic(tvr)^"+"'),
    deletions.df %>%
      dplyr::filter(grepl("08B02743",Genotype)) %>%
      dplyr::rename(Ratio = Deletion_ratio) %>%
      dplyr::mutate(phage = 'phi*"08B02743"~italic(ivr)^"-"~italic(tvr)^"-"')
  ) %>%
  dplyr::mutate(
    phage = factor(phage,
                   levels = c(
                       'phi*"08B02743"~italic(ivr)^"+"~italic(tvr)^"+"',
                       'phi*"08B02743"~italic(ivr)^"-"~"fixed"~italic(tvr)',
                       'phi*"08B02743"~italic(ivr)^"-"~italic(tvr)^"-"'
                    )
                   )
  )

# modified_2743_pvrms_deletions_plot <-
#   plot_phage_transformation_data(pvrms_versions.df,"Ratio","Ratio of prophage\ndeletions to point mutations",min_val = 0.04)

pv_rms_in_rmv.df <-
  dplyr::bind_rows(
    # 2743 informative comparison
    pvrms_versions.df %>%
      dplyr::filter(Genotype != "08B02743_immAcat") %>%
      dplyr::filter(phage != 'phi*"08B02743"~italic(ivr)^"-"~"fixed"~italic(tvr)') %>%
      dplyr::mutate(phage = dplyr::if_else(phage == 'phi*"08B02743"~italic(ivr)^"-"~italic(tvr)^"-"',
                                           'phi*"08B02743"~Delta*italic(ivr)~Delta*italic(tvr)',
                                           phage)) %>%
      dplyr::mutate(phage = paste0('""~',phage,'~""')),
    # RMV4 with active pvRMS
    read.csv("./transformation_data/RMV4_Ion_deletion_data.csv") %>%
      dplyr::mutate(mutant = dplyr::case_when(
          Genotype == "WT" ~ "WT",
          Genotype == "∆int" ~ "int",
          Genotype == "lytA::cat" ~ "lytA",
          Genotype == "rep::cat" ~ "rep"
        )
      ) %>%
      dplyr::mutate(Genotype = dplyr::case_when(
          Genotype == "WT" ~ "RMV4",
          Genotype == "∆int" ~ "RMV4_intcat",
          Genotype == "lytA::cat" ~ "RMV4_lytAcat",
          Genotype == "rep::cat" ~ "RMV4_repcat"
        )
      ) %>%
      dplyr::mutate(phage = '""~phi*"RMV4"~italic(ivr)^"+"~Delta*italic(tvrR)~""'),
    # RMV8 with active pvRMS
    read.csv("./transformation_data/RMV8_Ion_deletion_data.csv") %>%
      dplyr::mutate(mutant = dplyr::case_when(
          Genotype == "WT" ~ "WT",
          Genotype == "int::cat" ~ "int",
          Genotype == "lytA::cat" ~ "lytA",
          Genotype == "rep::cat" ~ "rep"
        )
      ) %>%
      dplyr::mutate(Genotype = dplyr::case_when(
          Genotype == "WT" ~ "RMV8",
          Genotype == "∆int" ~ "RMV8_intcat",
          Genotype == "int::cat" ~ "RMV8_intcat",
          Genotype == "lytA::cat" ~ "RMV8_lytAcat",
          Genotype == "rep::cat" ~ "RMV8_repcat"
        )
      ) %>%
      dplyr::mutate(phage = '""~phi*"RMV8"~italic(ivr)^"+"~Delta*italic(tvrR)~""'),
    # RMV4 data without pvRMS
    deletions.df %>%
      dplyr::filter(grepl("RMV4",Genotype)) %>%
      dplyr::rename(Ratio = Deletion_ratio) %>%
      dplyr::mutate(phage = '""~phi*"RMV4"~Delta*italic(ivr)~Delta*italic(tvrR)~""'),
    # RMV8 data without pvRMS
    deletions.df %>%
      dplyr::filter(grepl("RMV8",Genotype) & !grepl("ybrid",Genotype)) %>%
      dplyr::filter(mutant %in% c("WT","int","lytA")) %>%
      dplyr::rename(Ratio = Deletion_ratio) %>%
      dplyr::mutate(phage = '""~phi*"RMV8"~Delta*italic(ivr)~Delta*italic(tvrR)~""')
  ) %>%
  dplyr::mutate(
    phage = factor(phage,
                   levels = c(
                       '""~phi*"08B02743"~italic(ivr)^"+"~italic(tvr)^"+"~""',
                       '""~phi*"08B02743"~Delta*italic(ivr)~Delta*italic(tvr)~""',
                       '""~phi*"RMV8"~italic(ivr)^"+"~Delta*italic(tvrR)~""',
                       '""~phi*"RMV8"~Delta*italic(ivr)~Delta*italic(tvrR)~""',
                       '""~phi*"RMV4"~italic(ivr)^"+"~Delta*italic(tvrR)~""',
                       '""~phi*"RMV4"~Delta*italic(ivr)~Delta*italic(tvrR)~""'
                    )
                   )
  ) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("08B02743",
                                             "08B02743_intcat",
                                             "08B02743_immAcat",
                                             "08B02743_repcat",
                                             "08B02743_lytAcat",
                                             "RMV8",
                                             "RMV8_intcat",
                                             "RMV8 int::Jan",
                                             "RMV8_repcat",
                                             "RMV8_lytAcat",
                                             "RMV8_int_restored",
                                             "RMV4",
                                             "RMV4_intcat",
                                             "RMV4_repcat",
                                             "RMV4_lytAcat")
                                  )
                )

# modified_other_pvrms_deletions_plot <-
#   plot_phage_transformation_data(pv_rms_in_rmv.df,"Ratio","Ratio of prophage\ndeletions to point mutations",log_p_increment=0.25,min_val = 0.04)

cowplot::plot_grid(
  plotlist = list(
    plot_phage_transformation_data(pv_rms_in_rmv.df %>% 
                                     dplyr::filter(grepl("08B02743",phage)),
                                   "Ratio",
                                   "Ratio of prophage\ndeletions to point mutations",
                                   log_p_increment=0.25,
                                   min_val = 0.04),
    plot_phage_transformation_data(pv_rms_in_rmv.df %>% 
                                     dplyr::filter(grepl("RMV8",phage)),
                                   "Ratio",
                                   "Ratio of prophage\ndeletions to point mutations",
                                   log_p_increment=0.25,
                                   min_val = 0.04),
    plot_phage_transformation_data(pv_rms_in_rmv.df %>% 
                                     dplyr::filter(grepl("RMV4",phage)),
                                   "Ratio",
                                   "Ratio of prophage\ndeletions to point mutations",
                                   log_p_increment=0.25,
                                   min_val = 0.04)
  ),
  labels = "AUTO",
  nrow = 3
)

```


# Gel image of *tvr* loci

![](~/Documents/Admin/AIon/Laboratory_data/Year_3/Pictures_or_figures/Images/methylationpatterns_i_II_iii_III_WT_nullp_int_rep_NORMAL_AND_KAN_MUTANTS_NICE_PIC.jpg)

![](tvr_orientation_gel.pdf)

# Rates of prophage insertion by transformation

- RMV8 zn::cat (Hybrid) int::Janus RpoB*

``` {r Prophage insertion, fig.width = 6}

insertion.df <-
  read.csv("./transformation_data/RMV8_prophage_insertion_RAW_DATA.csv") %>%
  dplyr::filter(Recipient != "") %>%
  dplyr::filter(Volume.of.transformants.on.Rif == 5)

ggplot(insertion.df,
       aes(x = Number.of.colonies.on.Rif.plate,
           y = Number.of.colonies.on.Kan.Chl.plate)) +
  geom_point() +
  xlab(expression('Number of rifampicin-resistant colonies in 5'~mu*'L')) +
  ylab(expression('Number of kanamycin- and chloramphenicol-resistant colonies in 400'~mu*'L')) +
  theme_bw()

```

# What stimuli trigger prophage activation?

## Effects of CSP on overnight excision levels of RMV4

``` {r RMV4 effect of CSP on overnight excision levels}

rmv4_overnight_excision_df <-
  dplyr::bind_rows(
    read.csv("./qPCR/22072024_Overnight_0.2_20hr_RMV4.csv")
  ) %>%
  dplyr::mutate(Condition = dplyr::case_when(
      grepl("CSP",mutants) ~ "CSP",
      grepl("MMC",mutants) ~ "MMC",
      TRUE ~ "No supplement"
    )
  ) %>%
  dplyr::mutate(Genotype = dplyr::case_when(
      grepl("_spaL_",mutants) ~ "RMV4_spaLJanus",
      grepl("SpaL_",mutants) ~ "RMV4_spaLJanus",
      grepl("_dprA_",mutants) ~ "RMV4_dprA",
      grepl("DprA_",mutants) ~ "RMV4_dprA",
      TRUE ~ "RMV4"
    )
  )

ggplot(rmv4_overnight_excision_df,
       aes(x = Genotype,
           y = ADbyAB,
           colour = Condition)) +
  geom_point(position = position_jitter(width=0.1)) +
  theme_bw()

```

## Activation of prophage depends on the regulatory systems

```{r Activation levels of prophage, fig.width = 10, echo=FALSE}

activation.df <-
  dplyr::bind_rows(
    read.csv("./qPCR/RMV8_col4_ADbyAB_phage_Hybrid_July_v2.csv") %>%
      dplyr::rename(ratio = ADbyAB),
    read.csv("./qPCR/2743_ADAB_overnights.csv"),
    read.csv("./qPCR/RMV4_ADbyAB_overnights.csv"),
    read.csv("./qPCR/overnight_ADbyAB.csv") %>%
      dplyr::slice_tail(n=-2) %>% # remove first two lines - 3rd technical replicate failed
      dplyr::filter(!(mutants == "2743_immAcat" & ADbyAB > 0.001)) %>% # remove experiments with incorrect mutant
      dplyr::rename(ratio = ADbyAB),
    read.csv("./qPCR/immA_new_replicate_ADbyAB.csv") %>%
      dplyr::mutate(mutants = dplyr::if_else(grepl("2743_immA",mutants),"2743_immAcat",mutants)) %>%
      dplyr::rename(ratio = ADbyAB),
    # RMV4 spaL and dprA 
    read.csv("./qPCR/Overnight_CSP_18hr.csv") %>%
      dplyr::filter(!(grepl("CSP",mutants))) %>%
      dplyr::mutate(mutants = dplyr::case_when(
          grepl("_spaL_",mutants) ~ "RMV4_spaLJanus",
          grepl("_dprA_",mutants) ~ "RMV4_dprAJanus",
          TRUE ~ "RMV4"
        )
      ) %>%
      dplyr::rename(ratio = ADbyAB),
    # Extra replicates
    read.csv("./qPCR/Overnight_AdbyAB_extra_July.csv") %>%
      dplyr::rename(ratio = ADbyAB),
    read.csv("./qPCR/Overnight_RMV8_ADbyAB.csv") %>%
      dplyr::rename(ratio = ADbyAB),
    read.csv("./qPCR/Overnight_extra_ADbyAB_August.csv") %>%
      dplyr::rename(ratio = ADbyAB)
  ) %>%
  dplyr::rename(Genotype = mutants) %>%
  dplyr::mutate(phage = dplyr::case_when(
                    grepl("2743_zncat",Genotype) ~ "phi*'08B02743'",# ~ "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                    grepl("RMV8_Reg2743",Genotype) ~ "phi*RMV8[italic(immARcat)]",
                    grepl("RMV8_reg2743",Genotype) ~ "phi*RMV8[italic(immARcat)]",
                    grepl("RMV4 2743zncat",Genotype) ~ "phi*RMV4[italic(immARcat)]",
                    grepl("RMV4_zncat_",Genotype) ~ "phi*RMV4[italic(immARcat)]",
                    grepl("2743",Genotype) ~ "phi*'08B02743'",
                    grepl("RMV8",Genotype) ~ "phi*'RMV8'",
                    grepl("RMV4",Genotype) ~ "phi*'RMV4'",
                    TRUE ~ Genotype
                  )
                ) %>%
  dplyr::mutate(phage = factor(phage,
                               levels  = c("phi*'RMV8'",
                                           "phi*'08B02743'",
                                           "phi*'RMV4'",
                                           "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                                           "phi*RMV8[italic(immARcat)]",
                                           "phi*RMV4[italic(immARcat)]"
                                           )
                               )
                ) %>%
  dplyr::mutate(Genotype = dplyr::case_when(
                    # RMV8
                    Genotype == "RMV8" ~ "RMV8",
                    Genotype == "RMV8 integrase::cat" ~ "RMV8_intcat",
                    Genotype == "RMV8 lytA::cat" ~ "RMV8_lytAcat",
                    Genotype == "RMV8 rep::cat" ~ "RMV8_repcat",
                    Genotype == "RMV8 recA::Janus" ~ "RMV8_recAJanus",
                    Genotype == "RMV8 integrase::cat recA::Janus" ~ "RMV8_intcat_recAJanus",
                    Genotype == "RMV8_Reg2743" ~ "RMV8_Znregcat",
                    Genotype == "RMV8_Reg2743_recA::Janus" ~ "RMV8_Znregcat_recA",
                    Genotype == "RMV8_reg2743_2_RR2-1" ~ "RMV8_Znregcat_recA",
                    Genotype == "RMV8_reg2743_2_RR2-2" ~ "RMV8_Znregcat_recA",
                    # RMV4
                    Genotype == "RMV4" ~ "RMV4",
                    Genotype == "RMV4_WT" ~ "RMV4",
                    Genotype == "RMV4 int::cat" ~ "RMV4_intcat",
                    Genotype == "RMV4 int::ermB" ~ "RMV4_intcat",
                    Genotype == "RMV4_int" ~ "RMV4_intcat",
                    Genotype == "RMV4 recA::Janus" ~ "RMV4_recAJanus",
                    Genotype == "RMV4_recA" ~ "RMV4_recAJanus",
                    Genotype == "RMV4 2743zncat" ~ "RMV4_Znregcat",
                    Genotype == "RMV4_zncat_hybrid" ~ "RMV4_Znregcat",
                    Genotype == "RMV4 2743zncat recA::Janus" ~ "RMV4_Znregcat_recA",
                    Genotype == "RMV4_zncat_recA::Jan" ~ "RMV4_Znregcat_recA",
                    Genotype == "RMV4 2743zncat intermB" ~ "RMV4_Znregcat_intermB",
                    Genotype == "RMV4_SpaL_1" ~ "RMV4_spaLJanus",  ####
                    Genotype == "RMV4_spaLJanus" ~ "RMV4_spaLJanus",  ####
                    Genotype == "RMV4_spaL" ~ "RMV4_spaLJanus",
                    Genotype == "RMV4_dprAJanus" ~ "RMV4_dprAJanus",  ####
                    # 08B02743
                    Genotype == "2743" ~ "08B02743",
                    Genotype == "2743_WT" ~ "08B02743",
                    Genotype == "2743_immAcat" ~ "08B02743_immAcat",
                    Genotype == "2743_immAcat_recA" ~ "08B02743_immAcat_recAJanus", #### ADD PALETTE
                    Genotype == "2743_immA_recA" ~ "08B02743_immAcat_recAJanus",
                    Genotype == "2743 intcat" ~ "08B02743_intcat",
                    Genotype == "2743_intcat" ~ "08B02743_intcat",
                    Genotype == "2743_int" ~ "08B02743_intcat",
                    Genotype == "2743 recA" ~ "08B02743_recAJanus",
                    Genotype == "2743_zncat" ~ "08B02743_Znregcat",
                    Genotype == "2743_repcat" ~ "08B02743_repcat",
                    Genotype == "2743_lytA_Jan" ~ "08B02743_lytAcat",
                    Genotype == "2743_lytA_cat" ~ "08B02743_lytAcat",
                    Genotype == "2743_lytA" ~ "08B02743_lytAcat",
                    Genotype == "2743_zncat_recA" ~ "08B02743_Znregcat_recA",
                    grepl("2743_recA",Genotype) ~ "08B02743_recAJanus"
                  )
                ) %>%
  dplyr::filter(Genotype != "RMV4_Znregcat_intermB") %>%
  dplyr::filter(Genotype != "RMV4_dprAJanus") %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("RMV8",
                                             "RMV8_intcat",
                                             "RMV8_repcat", 
                                             "RMV8_lytAcat", 
                                             "RMV8_recAJanus",
                                             "RMV8_intcat_recAJanus",
                                             "08B02743",
                                            "08B02743_intcat",
                                            "08B02743_repcat", 
                                            "08B02743_lytAcat",
                                            "08B02743_recAJanus",
                                            "08B02743_immAcat",
                                            "08B02743_immAcat_recAJanus",
                                            "08B02743_Znregcat",
                                            "08B02743_Znregcat_recA", 
                                            "RMV4",
                                            "RMV4_intcat",
                                            "RMV4_recAJanus",
                                            "RMV4_spaLJanus",
                                            #"RMV4_dprAJanus",
                                            #"08B02743_Znregcat",
                                            #"08B02743_Znregcat_recA", 
                                            "RMV8_Znregcat",
                                            "RMV8_Znregcat_recA",
                                            "RMV4_Znregcat", 
                                            "RMV4_Znregcat_recA")
                                  )
                ) %>%
  dplyr::group_by(Genotype,phage) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

# Function for trimming phage names
trim_phage_name <- function(x) {
  
  x <- sub('phi \\* \\S+ ~', '', x)
  x <- sub('~phi*\'08B02743\'', '', x)
  return(parse_expr(x))
  
}

  # Process x axis labels
  abbreviations_for_x <-
    activation.df %>%
        dplyr::arrange(phage,Genotype) %>%
        dplyr::select(Genotype) %>%
        dplyr::distinct() %>%
        dplyr::pull()
  
  labels_for_x <- NULL
  
  for (x in abbreviations_for_x) {
      labels_for_x <- c(labels_for_x,mutant_labels[x])
  }
  
  labels_for_x[labels_for_x=='"" ~ phi * "RMV8" ~ italic(int) * "::" * italic(cat) ~ italic(recA) * "::Janus" ~ ""'] <- bquote('atop(scriptscriptstyle(""),atop(textstyle(""~italic(int) * "::" * italic(cat)),textstyle(italic(recA) * "::Janus"~"")))')
  labels_for_x[!(grepl(" italic",labels_for_x) | grepl("restored",labels_for_x))] <- bquote('""~"Intact prophage"~""')
  #labels_for_x[labels_for_x == '"" ~ phi * "08B02743"[italic(szaL) * "::" * italic(cat)] ~ ""'] <- bquote('""~"Intact prophage"~""')
  #labels_for_x <- sub('phi * "08B02743"[italic(szaL) * "::" * italic(cat)] ~ ', '', labels_for_x, fixed = TRUE)
  labels_for_x[labels_for_x=="\"\" ~  italic(int) * \"::\" * italic(cat) ~ italic(recA) * \"::Janus\" ~ \"\""] <- bquote('atop(scriptscriptstyle(""),atop(textstyle(""~italic(int) * "::" * italic(cat)),textstyle(italic(recA) * "::Janus"~"")))')
  labels_for_x[labels_for_x=='"" ~ phi * "08B02743"[italic(szaL) * "::" * italic(cat)] ~ ""'] <- bquote('"" ~ italic(szaL) * "::" * italic(cat) ~ ""')
  labels_for_x[labels_for_x=='"" ~ phi * "08B02743"[italic(szaL) * "::" * italic(cat)] ~ italic(recA) * "::Janus" ~ ""'] <- bquote('atop(scriptscriptstyle(""),atop(textstyle(""~italic(szaL) * "::" * italic(cat)),textstyle(italic(recA) * "::Janus"~"")))')
  labels_for_x[labels_for_x=='"" ~ phi * "08B02743" ~ italic(immA) * "::" * italic(cat) ~ italic(recA) * "::Janus" ~ ""'] <- bquote('atop(scriptscriptstyle(""),atop(textstyle(""~italic(immA) * "::" * italic(cat)),textstyle(italic(recA) * "::Janus"~"")))')
  labels_for_x <- lapply(labels_for_x,trim_phage_name)
  
  #labels_for_x <- sub('phi \\* \\S+ ~', '', labels_for_x)
  #labels_for_x <- sub('~phi*\'08B02743\'', '', labels_for_x)

  naming_vec = labels_for_x
  names(naming_vec) = abbreviations_for_x


  
phage_mutant_label_order <- match(levels(activation.df$Genotype),names(mutant_labels))
phage_mutant_labels <- mutant_labels[phage_mutant_label_order]

(excision_levels_plot <-
  ggplot(activation.df,
         aes(x = Genotype,
             y = ratio,
             colour = Genotype,
             fill = Genotype,
             group = Genotype)) +
    geom_violin(draw_quantiles = c(0.5),
                scale = "width",
                colour = "black",
                alpha = 0.15,
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
               position = position_jitter(width = 0.25),
               size = 0.75) +
    scale_y_continuous(trans = "log10") +
    scale_colour_manual(labels = phage_mutant_labels,
                          values = mutant_palette,
                          aesthetics = c("fill","colour")) +
    geom_hline(yintercept = 1e-3, linetype = 3) +
    scale_x_discrete(labels = naming_vec,
                     expand = c(0,1)) +
    ylab(expression(italic(attB)~"to"~italic(attL)~"ratio")) +
    xlab("") +
    facet_grid(.~phage,
              scales = "free_x",
              space = "free_x",
              labeller = label_parsed) +
    theme(legend.position = "none",
          strip.text = element_text(size = 7),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 7),
          panel.spacing = unit(0, "lines"))
)

```

# RecA and ssDNA synergistically activate prophage

```{r Effect of DNA and MMC, fig.width = 8, echo=FALSE}

guide_nrow = 4
g <- guide_legend(nrow = guide_nrow,
                    title = NULL)

# RMV8
phi_rmv8_growth.df <-
  get_growth_curve_df(purrr::map(list(
                                  "./growth_curves/1207_withoutMitomycin_RMV8.csv",
                                  "./growth_curves/1207_withMitomycin_RMV8.csv",
                                  "./growth_curves/1207_withoutMitomycin_RMV8_PhageKO.csv",
                                  "./growth_curves/1207_withMitomycin_RMV8_PhageKO.csv",
                                  "./growth_curves/1807_RMV8_recAKO_without_mitomycin.csv",
                                  "./growth_curves/1807_RMV8_recAKO_with_mitomycin.csv",
                                  "./growth_curves/1807_RMV8_phageKO_recAKO_without_mitomycin.csv",
                                  "./growth_curves/1807_RMV8_phageKO_recAKO_with_mitomycin.csv"
                                ),
                               read.csv) %>%
                      purrr::reduce(left_join, by = "OD600"),
    threshold = 7.5
  ) %>%
  dplyr::mutate(phage = "phi*'RMV8'") %>%
  dplyr::mutate(dataset = 1)

# RMV8 hybrid
phi_rmv8_hybrid_growth.df <-
  get_growth_curve_df(purrr::map(list(
                                  "./growth_curves/1207_withMitomycin_Hybrid_RMV8.csv",
                                  "./growth_curves/1207_withoutMitomycin_Hybrid_RMV8.csv"
                                ),
                               read.csv) %>%
                      purrr::reduce(left_join, by = "OD600"),
    threshold = 7.5
  ) %>%
  dplyr::mutate(phage = "phi*RMV8[Zn*italic(reg)*'::'*italic(cat)]") %>%
  dplyr::mutate(dataset = 2)


# 08B02743
phi_08B02743_hybrid_growth.df <-
  get_growth_curve_df(dplyr::left_join(read.csv("./growth_curves/2743_CSP_Mitomycin_DNA.csv"),
                                     read.csv("./growth_curves/2743_prophageKO_CSP_Mitomycin_DNA.csv"),
                                     by = c("OD600")) %>%
                      dplyr::left_join(read.csv("./growth_curves/2743_recAKO_CSP_Mitomycin_DNA.csv")),
    threshold = 7.5
  ) %>%
  dplyr::mutate(phage = "phi*'08B02743'") %>%
  dplyr::mutate(dataset = 3)

# second dataset
phi_rmv8_and_hybrid_with_mmc.df <-
  get_growth_curve_df(read.csv("./growth_curves/2nd_replicate_RMV8.csv"),
    threshold = 7.5) %>%
    dplyr::mutate(phage = dplyr::if_else(Genotype == "RMV8_Znregcat","phi*RMV8[Zn*italic(reg)*'::'*italic(cat)]","phi*'RMV8'")) %>%
  dplyr::mutate(dataset = 4)

# third dataset
phi_rmv8_and_hybrid_with_without_mmc.df <-
  get_growth_curve_df(read.csv("./growth_curves/3rd_replicate_RMV8.csv"),
    threshold = 7.5) %>%
  dplyr::mutate(phage = dplyr::case_when(Genotype == "RMV8_Znregcat" ~ "phi*RMV8[Zn*italic(reg)*'::'*italic(cat)]",
                                         Genotype == "RMV8" ~ "phi*'RMV8'",
                                         Genotype == "RMV8_phageJanus" ~ "phi*'RMV8'")
                ) %>%
  dplyr::mutate(dataset = 5)

# RMV4 data
phi_rmv4_and_hybrid_with_without_mmc.df <-
  get_growth_curve_df(read.csv("./growth_curves/RMV4_mitomycin_CSP_GD_growth.csv"),
    threshold = 7.5) %>%
  dplyr::mutate(phage = dplyr::case_when(
                        grepl("RMV4_Zn",Genotype) ~ "phi*RMV4[Zn*italic(reg)*'::'*italic(cat)]",
                        grepl("RMV4",Genotype)  ~ "phi*'RMV4'"
                  )
                )%>%
  dplyr::mutate(dataset = 6)

# Fifth replicate - RMV8 data
phi_rmv8_and_hybrid_with_without_mmc_replicate_2.df <-
  get_growth_curve_df(read.csv("./growth_curves/growth_fifth_replicate.csv"),
    threshold = 7.5) %>%
  dplyr::mutate(phage = dplyr::case_when(
                        grepl("RMV8_phageJanus",Genotype) ~ "phi*RMV8",
                        grepl("RMV8_Znregcat",Genotype) ~ "phi*RMV4[Zn*italic(reg)*'::'*italic(cat)]",
                        grepl("RMV8",Genotype)  ~ "phi*'RMV8'"
                  )
                ) %>%
  dplyr::mutate(dataset = 7)

# Sixth replicate - 2743, RMV4 and RMV8 data
additional_exposure_replicates.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/2743_two_replicate_MMC_GD.csv") %>% 
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "control", 
                                         replacement = "X2743"),
                  matches("control")) %>%
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "MMC", 
                                         replacement = "X2743.MMC"), 
                  matches("MMC")),
        threshold = 7.5) %>%
      dplyr::mutate(phage = "phi*'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_two_replicate_MMC_GD.csv") %>% 
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "control", 
                                         replacement = "RMV4"),
                  matches("control")) %>%
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "MMC", 
                                         replacement = "RMV4.MMC"), 
                  matches("MMC")),
        threshold = 7.5) %>%
      dplyr::mutate(phage = "phi*RMV4'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_two_replicate_MMC_GD.csv") %>% 
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "control", 
                                         replacement = "RMV8"),
                  matches("control")) %>%
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "MMC", 
                                         replacement = "RMV8.MMC"), 
                  matches("MMC")),
        threshold = 7.5) %>%
      dplyr::mutate(phage = "phi*RMV8'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_two_replicate_MMC_GD.csv") %>% 
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "control", 
                                         replacement = "RMV8_recAJanus"),
                  matches("control")) %>%
      rename_with(~ stringr::str_replace(.x, 
                                         pattern = "MMC", 
                                         replacement = "RMV8_recAJanus.MMC"), 
                  matches("MMC")),
        threshold = 7.5) %>%
      dplyr::mutate(phage = "phi*RMV8'")
  ) %>%
  dplyr::mutate(dataset = 8)

# combine plots
mmc_dna_growth.df <-
  dplyr::bind_rows(
    phi_rmv8_growth.df,
    phi_rmv8_hybrid_growth.df,
    phi_08B02743_hybrid_growth.df,
    phi_rmv8_and_hybrid_with_mmc.df,
    phi_rmv8_and_hybrid_with_without_mmc.df,
    phi_rmv4_and_hybrid_with_without_mmc.df,
    phi_rmv8_and_hybrid_with_without_mmc_replicate_2.df,
    additional_exposure_replicates.df
  ) %>%
  # identify first replicates for growth curve plotting
  dplyr::group_by(Genotype) %>%
  dplyr::mutate(first_replicate = (dataset == min(dataset))) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'",
                                          "phi*'08B02743'",
                                          "phi*RMV8[Zn*italic(reg)*'::'*italic(cat)]",
                                          "phi*'RMV4'",
                                          "phi*RMV4[Zn*italic(reg)*'::'*italic(cat)]")
                               )
                ) %>%
  dplyr::mutate(gDNA = dplyr::if_else(gDNA=="Imported DNA","Imported~DNA","No~DNA")) %>%
  dplyr::mutate(MMC = dplyr::if_else(MMC=="Mitomycin C","Mitomycin~C","No~mitomycin~C"))

mmc_dna_endpoints.df <-
  mmc_dna_growth.df %>%
  # unnest observations
  dplyr::filter(Time == max(Time)) %>%
  dplyr::select(Genotype,phage,MMC,gDNA,Observations,dataset) %>%
  tidyr::unnest(Observations) %>%
  # identify paired samples
  dplyr::group_by(Genotype,dataset,gDNA,MMC) %>%
  tibble::rowid_to_column("index") %>%
  dplyr::ungroup() %>%
  # get differences
  dplyr::group_by(Genotype,dataset) %>%
  dplyr::mutate(control_level = dplyr::if_else(
                      MMC == "No~mitomycin~C" & gDNA == "No~DNA",
                      Observations,
                      NA_real_
                    )
                ) %>%
  tidyr::fill(control_level,.direction="updown") %>%
  dplyr::ungroup() %>%
  # remove control measurements
  dplyr::filter(MMC != "No~mitomycin~C" | gDNA != "No~DNA") %>%
  # calculate differences
  dplyr::mutate(Difference = Observations - control_level) %>%
  # make labels
  dplyr::mutate(Condition = factor(
                  dplyr::case_when(
                    MMC == "No~mitomycin~C" & gDNA == "No~DNA" ~ "No supplement",
                    MMC == "No~mitomycin~C" & gDNA == "Imported~DNA" ~ "Genomic DNA",
                    MMC == "Mitomycin~C" & gDNA == "No~DNA" ~ "Mitomycin C & CSP",
                    MMC == "Mitomycin~C" & gDNA == "Imported~DNA" ~ "Mitomycin C & CSP & genomic DNA"
                  )
                )
              ) %>%
  dplyr::filter(Condition != "Control" & Genotype %in% c("RMV8",
                                                         "08B02743",
                                                         "RMV4",
                                                         "RMV8_Znregcat",
                                                         "RMV8_phageJanus",
                                                         "RMV8_recAJanus"
                                                         )
                ) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c(
                                      "RMV8",
                                      "RMV8_phageJanus",
                                      "RMV8_recAJanus",
                                      "08B02743",
                                      "RMV4",
                                      "RMV8_Znregcat"
                                    )
                                  )
                )

# statistical test
d_width = 0.75
j_width = 0.25
endpoint_growth.stat.test <- 
  mmc_dna_endpoints.df %>%
    group_by(Genotype) %>%
    wilcox_test(Difference ~ Condition,
                ref.group = "Mitomycin C & CSP & genomic DNA",
                paired = FALSE,
                p.adjust.method = "holm") %>%
    rstatix::add_xy_position(x = "Locus", dodge = 0.8) %>%
    dplyr::ungroup() %>%
    dplyr::rowwise() %>%
    dplyr::mutate(genotype_position = which(levels(mmc_dna_endpoints.df$Genotype)==as.character(Genotype)),
                  condition_position_1 = which(levels(mmc_dna_endpoints.df$Condition)==group2),
                  condition_position_2 = which(levels(mmc_dna_endpoints.df$Condition)==group1)) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(xmin = genotype_position + 1/3*d_width*(condition_position_1-2),
                  xmax = genotype_position + 1/3*d_width*(condition_position_2-2)) %>%
    dplyr::filter(p.adj.signif != "ns")

# plot
(growth_endpoint_plot <-
  ggplot(mmc_dna_endpoints.df,
         aes(x = Genotype,
             y = Difference,
             fill = Condition,
             colour = Condition,
             width = d_width,#shape = as.factor(dataset)
             )) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                scale = "width",
                alpha = 0.15,
                position = position_dodge(width = d_width,
                                          preserve = 'single')
                ) +
    geom_point(position = position_jitterdodge(dodge.width = d_width,
                                               jitter.width = j_width
                                               )
               ) +
    geom_hline(yintercept = 0, linetype = 3) +
    scale_x_discrete(labels = ifelse(mutant_labels[levels(mmc_dna_endpoints.df$Genotype)]=='"" ~ phi * "RMV8" ~ italic(recA) * "::Janus" ~ ""',
                                     parse(text='atop("" ~ phi * "RMV8",italic(recA) * "::Janus" ~ "")'),
                                     mutant_labels[levels(mmc_dna_endpoints.df$Genotype)])) +
    ylab(expression('Effect on growth after 8 h ('*Delta*'OD'[600]*')')) +
    scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
    theme_bw() +
    
    ggpubr::stat_pvalue_manual(endpoint_growth.stat.test,
                                    size = 2,
                                    label = "{scales::pvalue(p.adj)}",
                                    tip.length = 0,
                                    inherit.aes = FALSE) +
    theme(legend.position = "bottom")
)

```

# Growth curve for IPTG inducible RMV8 hybrid mutant

``` {r Growth curve for IPTG inducible RMV8 hybrid mutant}

time_threshold <- 8

rmv8_hybrid_iptg_growth.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/RMV8_plac_recA_growth_GD_MMC.csv"), threshold = time_threshold) %>%
      dplyr::mutate(Genotype = "phi*'RMV8'~italic(recA)[italic(lacI)]"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_Hybrid_HF_plac_recA_growth_GD_MMC.csv"), threshold = time_threshold) %>%
      dplyr::mutate(Genotype = "phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]")
  ) %>%
  dplyr::mutate(dataset = 1) %>%
  dplyr::mutate(gDNA = paste0("'",gDNA,"'")) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("phi*'RMV8'~italic(recA)[italic(lacI)]","phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"))
                )

# Growth curve plots over time
rmv8_hybrid_iptg_growth_plot <-
  ggplot(rmv8_hybrid_iptg_growth.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = MMC,
               fill = MMC,
               linetype = IPTG,
               group = interaction(IPTG,MMC))
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g, linetype = g) +
      facet_grid(gDNA~Genotype,
                 scales = 'free_x',
                 labeller = label_parsed)

```

# Effect of MMC and gDNA and IPTG on prophage induction

``` {r Plot endpoint differences between matched samples with IPTG, echo = FALSE, fig.width = 8, fig.height = 11}

mmc_dna_iptg_endpoints.df <-
  rmv8_hybrid_iptg_growth.df %>%
  # Reset formatting
  dplyr::mutate(gDNA = gsub("'","",gDNA)) %>%
  # unnest observations
  dplyr::filter(Time == max(Time)) %>%
  dplyr::select(Genotype,MMC,gDNA,IPTG,Observations,dataset) %>%
  tidyr::unnest(Observations) %>%
  # identify paired samples
  dplyr::group_by(Genotype,dataset,gDNA,MMC,IPTG) %>%
  tibble::rowid_to_column("index") %>%
  dplyr::ungroup() %>%
  # get differences
  dplyr::group_by(Genotype,dataset) %>%
  dplyr::mutate(control_level = dplyr::if_else(
                      MMC == "No mitomycin C" & gDNA == "No imported DNA" & IPTG == "No IPTG",
                      Observations,
                      NA_real_
                    )
                ) %>%
  tidyr::fill(control_level,.direction="updown") %>%
  dplyr::ungroup() %>%
  # calculate differences
  dplyr::mutate(Difference = Observations - control_level) %>%
  # make labels
  dplyr::mutate(Condition = factor(
                  dplyr::case_when(
                    MMC == "No mitomycin C" & gDNA == "No imported DNA" & IPTG == "No IPTG" ~ "No supplement",
                    MMC == "No mitomycin C" & gDNA == "Imported DNA" & IPTG == "No IPTG" ~ "Genomic DNA",
                    MMC == "Mitomycin C" & gDNA == "No imported DNA" & IPTG == "No IPTG" ~ "Mitomycin C",
                    MMC == "Mitomycin C" & gDNA == "Imported DNA" & IPTG == "No IPTG" ~ "Mitomycin C & genomic DNA",
                    MMC == "No mitomycin C" & gDNA == "Imported DNA" & IPTG == "IPTG" ~ "Genomic DNA & IPTG",
                    MMC == "Mitomycin C" & gDNA == "No imported DNA" & IPTG == "IPTG" ~ "Mitomycin C & IPTG",
                    MMC == "Mitomycin C" & gDNA == "Imported DNA" & IPTG == "IPTG" ~ "Mitomycin C & genomic DNA & IPTG"
                  )
                )
              ) %>%
  dplyr::filter(Condition != "No supplement" & Genotype %in% c("phi*'RMV8'~italic(recA)[italic(lacI)]",
                                                              "phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"
                                                              )
                ) %>%
  dplyr::mutate(Genotype = factor(Genotype))

# statistical test
d_width = 0.75
j_width = 0.25
mmc_dna_iptg_endpoints.stat_test <- 
  mmc_dna_iptg_endpoints.df %>%
    group_by(Genotype) %>%
    wilcox_test(Difference ~ Condition,
                ref.group = "Mitomycin C & genomic DNA",
                paired = FALSE,
                p.adjust.method = "holm") %>%
    rstatix::add_xy_position(x = "Locus", dodge = 0.8) %>%
    dplyr::ungroup() %>%
    dplyr::rowwise() %>%
    dplyr::mutate(genotype_position = which(levels(mmc_dna_iptg_endpoints.df$Genotype)==as.character(Genotype)),
                  condition_position_1 = which(levels(mmc_dna_iptg_endpoints.df$Condition)==group2),
                  condition_position_2 = which(levels(mmc_dna_iptg_endpoints.df$Condition)==group1)) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(xmin = genotype_position + 1/3*d_width*(condition_position_1-3),
                  xmax = genotype_position + 1/3*d_width*(condition_position_2-3)) %>%
    dplyr::filter(p.adj.signif != "ns")

# plot
rmv8_hybrid_iptg_growth_endpoint_plot <-
  ggplot(mmc_dna_iptg_endpoints.df,
         aes(x = Condition,
             y = Difference,
             fill = Condition,
             colour = Condition,
             width = d_width,#shape = as.factor(dataset)
             )) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                position = position_dodge(width = d_width,
                                          preserve = 'single'),
                scale = "width",
                alpha = 0.15
                ) +
    geom_point(position = position_jitterdodge(dodge.width = d_width,
                                               jitter.width = j_width
                                               )
               ) +
    scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
    ylab(expression(atop('Effect on growth','after 8 h ('*Delta*'OD'[600]*')'))) +
    geom_hline(yintercept = 0, linetype = 3) +
    facet_grid(.~Genotype,
                   labeller = label_parsed) +
    theme_bw() +
    theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
                      strip.background = element_blank()
          ) +
    ggpubr::stat_pvalue_manual(mmc_dna_iptg_endpoints.stat_test,
                                    size = 2,
                                    label = "{scales::pvalue(p.adj)}",
                                    tip.length = 0,
                                    inherit.aes = FALSE)

cowplot::plot_grid(
  plotlist = list(
    rmv8_hybrid_iptg_growth_plot,
    rmv8_hybrid_iptg_growth_endpoint_plot
  ),
  nrow = 2,
  labels= "AUTO",
  rel_heights = c(0.6,0.4)
)

```

# ssDNA targeting the regulatory systems activate prophage

```{r Activation of prophage by ssDNA, fig.width = 8, echo=FALSE}

partial_deletions.df <-
  dplyr::bind_rows(
    read.csv("./transformation_data/2743_partial_using_GD_July.csv") %>%
      dplyr::mutate(phage = "phi*'08B02743'",
                    Genotype = "08B02743",
                    conditions = "gDNA"),
    read.csv("./transformation_data/2743_partial_using_PCR.csv") %>%
      dplyr::mutate(phage = "phi*'08B02743'",
                    Genotype = "08B02743") %>%
      dplyr::rename(conditions = condition),
    read.csv("./transformation_data/RMV8_partial_deletions_usingPCR_July_v3.csv") %>%
      dplyr::mutate(phage = "phi*'RMV8'",
                    Genotype = "RMV8"),
    read.csv("./transformation_data/RMV8_deletions_usingGD_July.csv") %>%
      dplyr::mutate(phage = "phi*'RMV8'",
                    Genotype = "RMV8",
                    conditions = "gDNA"),
    read.csv("./transformation_data/RMV8_hybrid_partial_deletions_usingPCR_July_v3.csv") %>%
      dplyr::mutate(phage = "phi*'RMV8'[italic(immARcat)]",
                    Genotype = "RMV8_Znregcat"),
    read.csv("./transformation_data/RMV8_hybrid_partial_usingGD_July_v3.csv") %>%
      dplyr::mutate(phage = "phi*'RMV8'[italic(immARcat)]",
                    Genotype = "RMV8_Znregcat",
                    conditions = "gDNA"),
    
        read.csv("./transformation_data/RMV4_partial_PCR_Partial_deletions.csv") %>%
      dplyr::mutate(phage = "phi*'RMV4'",
                    Genotype = "RMV4") %>%
      dplyr::rename(conditions = condition),
    
        read.csv("./transformation_data/RMV4_partial_deletions_usingGD.csv") %>%
      dplyr::mutate(phage = "phi*'RMV4'",
                    Genotype = "RMV4",
                    conditions = "gDNA"),
    
  ) %>%
  dplyr::mutate(mutants = dplyr::case_when(
                        mutants == "partial 1_GD" ~ "Proximal",
                        mutants == "partial 3_GD" ~ "Distal",
                        mutants == "ssbB_GD" ~ "control",
                        mutants == "partial 1" ~ "Proximal",
                        mutants == "modified_partial1" ~ "Proximal",
                        mutants == "partial 3" ~ "Distal",
                        mutants == "ssbB" ~ "control",
                        mutants == "partial1" ~ "Proximal",
                        mutants == "partial3" ~ "Distal",
                        mutants == "lytA" ~ "lytA",
                        mutants == "lytA_GD" ~ "lytA",
                        mutants == "gd_ssbB" ~ "control",
                        mutants == "gd_ssbB_KO" ~ "control",
                        mutants == "gd_partial1" ~ "Proximal",
                        mutants == "gd_modified1" ~ "Proximal",
                        mutants == "gd_partial3" ~ "Distal",
                        mutants == "gd_lytA" ~ "lytA",
                        mutants == "gd_partial 3" ~ "Distal"
                      )
                    ) %>%
  dplyr::mutate(conditions = dplyr::case_when(
                        conditions == "200ng" ~ "atop(PCR~amplicon,(200~ng))",
                        conditions == "50ng" ~ "atop(PCR~amplicon,(50~ng))",
                        conditions == "gDNA" ~ "atop(Genomic,DNA)"
                      )
                    ) %>%
  dplyr::rename(Locus = mutants) %>%
  dplyr::filter(Locus != "control") %>%
  dplyr::mutate(Locus = factor(Locus,
                               levels = c("Proximal",
                                          "Distal",
                                          "lytA"))
                ) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'",
                                          "phi*'08B02743'",
                                          "phi*'RMV4'",
                                          "phi*'RMV8'[italic(immARcat)]"))
                )

# Statistical test
partial_deletions.stat.test <- 
  partial_deletions.df %>%
    group_by(phage,conditions) %>%
    wilcox_test(deletions ~ Locus,
                ref.group = "Distal",
                paired = FALSE,
                p.adjust.method = "holm") %>%
    rstatix::add_xy_position(x = "Locus", dodge = 0.8) %>%
    dplyr::filter(p.adj.signif != "ns")

ggplot(partial_deletions.df,
       aes(x = Locus,
           y = deletions,
           colour = Genotype,
           fill = Genotype)) +
  geom_violin(draw_quantiles = c(0.5),
              colour = "black",
              scale = "width",
              alpha = 0.15,
              position = position_dodge(preserve = 'single')) +
  geom_point(position = position_jitter(width = 0.15)) +
  ylab("Ratio of prophage deletions to point mutations") +
  scale_y_continuous(trans = "log10") +
  scale_colour_manual(labels = mutant_labels[mutant_labels %in% unique(partial_deletions.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
  scale_x_discrete(labels = expression("Regulatory",
                                       "Distal",
                                       italic(lyt))) +
  facet_grid(conditions~phage,
             labeller = label_parsed) +
  theme(legend.position = "none",
            panel.spacing = unit(0, "lines")) +
  ggpubr::stat_pvalue_manual(partial_deletions.stat.test %>%
                               dplyr::mutate(y.position = log(y.position,10)),
                                    size = 2,
                                    label = "{scales::pvalue(p.adj)}",
                                    tip.length = 0,
                                    inherit.aes = FALSE)

```

# How quickly do prophage activate?

# Effect of *comEA* on RMV8

``` {r Growth of RMV8 comEA mutant}

guide_nrow = 2
g <- guide_legend(nrow = guide_nrow,
                    title = NULL)

rmv8_comEA_growth.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/RMV8_metals.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"RMV8"~""'),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_comEA_metals_senstivity_2_replicates.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"RMV8"~italic(comEA)*"::Janus"~""')
  ) %>%
    dplyr::group_by(Genotype,Time) %>%
    dplyr::mutate(Control = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_)) %>%
    tidyr::fill(Control,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(Adjusted_Median = Median - Control,
                  Adjusted_Min = Min - Control,
                  Adjusted_Max = Max - Control) %>%
    dplyr::filter(`Growth condition` != "No supplement") 

# Growth curve plots over time
(rmv8_comEA_growth_plot <-
  ggplot(rmv8_comEA_growth.df,
           aes(x = Time,
               y = Adjusted_Median,
               ymin = Adjusted_Min,
               ymax = Adjusted_Max,
               colour = `Growth condition`,
               fill = `Growth condition`)
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank()) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_wrap(~Genotype,
               label = label_parsed)
)

```

# dprA mutant metal sensitivity

``` {r Comparison of all dprA, fig.height = 11, fig.width = 8}

guide_nrow = 2
g <- guide_legend(nrow = guide_nrow,
                    title = NULL)

competence_mutant_growth.df <-
  dplyr::bind_rows(
    # 2743 immA repeat
      get_growth_curve_df(read.csv("./growth_curves/2743_immA_2nd_repeat_022024.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(immA)*"::"*italic(cat)~""'),
    # 2743 radA
      get_growth_curve_df(read.csv("./growth_curves/2743_radA_KO_metals.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(radA)*"::Janus"~""'),
    # 2743 dprA
      get_growth_curve_df(read.csv("./growth_curves/2743_dprAKO_R1_2604.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(dprA)*"::Janus"~""'),
    # 2743 radA immA
      #get_growth_curve_df(read.csv("2743_immA_radAKO_022024.csv")) %>%
      get_growth_curve_df(read.csv("./growth_curves/03052024_2743_immA_radA_KO_repeat.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(radA)*"::Janus"~italic(immA)*"::"*italic(cat)~""'),
    # 2743 dprA immA
    get_growth_curve_df(read.csv("./growth_curves/2743_immA_dprA_022024.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(dprA)*"::Janus"~italic(immA)*"::"*italic(cat)~""'),
    # 2743 recA cinA double mutant
      get_growth_curve_df(read.csv("./growth_curves/2743_cinArecA_KO_022024.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~"("*italic(recA)~italic(cinA)*")::Janus"~""')
  ) %>%
    dplyr::group_by(Genotype,Time) %>%
    dplyr::mutate(Control = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_)) %>%
    tidyr::fill(Control,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(Adjusted_Median = Median - Control,
                  Adjusted_Min = Min - Control,
                  Adjusted_Max = Max - Control) %>%
    dplyr::filter(`Growth condition` != "No supplement") 

# Growth curve plots over time
mutant_conditions_growth_plot <-
  ggplot(competence_mutant_growth.df,
           aes(x = Time,
               y = Adjusted_Median,
               ymin = Adjusted_Min,
               ymax = Adjusted_Max,
               colour = `Growth condition`,
               fill = `Growth condition`)
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank()) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_wrap(~Genotype,
               label = label_parsed)

dprA_comparison.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/RMV4_metals_dprAKO.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"RMV4"~italic(dprA)*"::Janus"~""'),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_metals_dprAKO.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"RMV8"~italic(dprA)*"::Janus"~""'),
        # 2743 dprA
    get_growth_curve_df(read.csv("./growth_curves/2743_dprAKO_R1_2604.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(dprA)*"::Janus"~""')
  ) %>%
    dplyr::group_by(Genotype,Time) %>%
    dplyr::mutate(Control = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_)) %>%
    tidyr::fill(Control,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(Adjusted_Median = Median - Control,
                  Adjusted_Min = Min - Control,
                  Adjusted_Max = Max - Control) %>%
    dplyr::filter(`Growth condition` != "No supplement") %>%
    dplyr::mutate(
      Genotype = factor(Genotype,
                        levels = c('""~phi*"RMV8"~italic(dprA)*"::Janus"~""','""~phi*"08B02743"~italic(dprA)*"::Janus"~""','""~phi*"RMV4"~italic(dprA)*"::Janus"~""'))
    )

# Growth curve plots over time
dprA_comparison_growth_plot <-
  ggplot(dprA_comparison.df,
           aes(x = Time,
               y = Adjusted_Median,
               ymin = Adjusted_Min,
               ymax = Adjusted_Max,
               colour = `Growth condition`,
               fill = `Growth condition`)
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank()) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_wrap(~Genotype,
               label = label_parsed)

dprA_double_mutants.df <-
  dplyr::bind_rows(
    get_growth_curve_df(read.csv("./growth_curves/2743_dprAKO_R1_2604.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(dprA)*"::Janus"~""'),
    #get_growth_curve_df(read.csv("2743_dprAKO_zncat_2_2604.csv")) %>%
     get_growth_curve_df(read.csv("./growth_curves/2743_dprAKO_zncat_2_2604.csv")) %>% 
      dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~italic(dprA)*"::Janus"~""'),
    # REMOVE
    get_growth_curve_df(read.csv("./growth_curves/2743_prophageJan_dprAcat_2604.csv")) %>%
      dplyr::mutate(Genotype = '""~phi*"08B02743::"*italic(cat)~~italic(dprA)*"::Janus"~""')
    # REMOVE
  ) %>%
    dplyr::group_by(Genotype,Time) %>%
    dplyr::mutate(Control = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_)) %>%
    tidyr::fill(Control,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(Adjusted_Median = Median - Control,
                  Adjusted_Min = Min - Control,
                  Adjusted_Max = Max - Control) %>%
    dplyr::filter(`Growth condition` != "No supplement") %>%
    dplyr::mutate(Genotype = factor(Genotype,
                                    levels = c('""~phi*"08B02743"~italic(dprA)*"::Janus"~""',
                                              '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~italic(dprA)*"::Janus"~""',
                                               '""~phi*"08B02743::"*italic(cat)~~italic(dprA)*"::Janus"~""'
                                               )
                                    )
                  )

# Growth curve plots over time
dprA_double_mutant_comparison_growth_plot <-
  ggplot(dprA_double_mutants.df,
           aes(x = Time,
               y = Adjusted_Median,
               ymin = Adjusted_Min,
               ymax = Adjusted_Max,
               colour = `Growth condition`,
               fill = `Growth condition`)
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank()) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_wrap(~Genotype,
               label = label_parsed)

(dprA_comparison_plot <-
  cowplot::plot_grid(
    plotlist = list(
      mutant_conditions_growth_plot + theme(legend.position = "none"),
      dprA_comparison_growth_plot + theme(legend.position = "none"),
      dprA_double_mutant_comparison_growth_plot + theme(legend.position = "none"),
      cowplot::get_plot_component(mutant_conditions_growth_plot, 'guide-box-bottom', return_all = TRUE)
    ),
    nrow = 4,
    ncol = 1,
    labels = c("A","B","C",""),
    rel_heights = c(2,1,1,0.5)
  )
)

```

``` {r Check on 2743 mutant phenotypes}

check_2743.df <-
  get_growth_curve_df(read.csv("./growth_curves/2743_zn_mutants_all_mitomycin_senstivity_repeat.csv")) %>%
    dplyr::mutate(Genotype = dplyr::case_when(
                      Genotype == "08B02743" ~ '""~phi*"08B02743"~""',
                      Genotype == "08B02743_Znregcat" ~ '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~""',
                      Genotype == "08B02743_Znregcat_recA" ~ '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~italic(recA)*"::Janus"~""',
                      Genotype == "08B02743_recAJanus" ~ '""~phi*"08B02743"~italic(recA)*"::Janus"~""'
                    )
                  )

# Growth curve plots over time
# Do not adjust by control because growth defect in recA mutant without MMC
(check_2743_mutant_comparison_growth_plot <-
  ggplot(check_2743.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = `Growth condition`,
               fill = `Growth condition`)
         ) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank()) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_wrap(~Genotype,
               label = label_parsed)
  )

```

# Combined expression and excision kinetics

``` {r Combined kinetics analysis}

phage_excision.df <-
  dplyr::bind_rows(
    
    # AD to AB ratio
    read_excision_file("./qPCR/2743_ADbyAB_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "08B02743") %>%
      dplyr::rename(ratio = ADbyAB) %>%
      dplyr::mutate(Measurement = "AD to AB ratio"),
   read_excision_file("./qPCR/ADbyAB_RMV8_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV8") %>%
      dplyr::rename(ratio = ADbyAB) %>%
      dplyr::mutate(Measurement = "AD to AB ratio"),
   read_excision_file("./qPCR/ADbyAB_RMV8_hybrid_1st_2nd_3rd.csv") %>%
      dplyr::rename(ratio = ADbyAB) %>%
      dplyr::mutate(phage = "RMV8_hybrid") %>%
      dplyr::mutate(Measurement = "AD to AB ratio"),
   read_excision_file("./qPCR/RMV4_ADbyAB_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV4") %>%
      dplyr::mutate(Measurement = "AD to AB ratio"),
   
   # BC to rpoA ratio
    read_excision_file("./qPCR/2743_BC_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "08B02743") %>%
      dplyr::mutate(Measurement = "BC total"),
   read_excision_file("./qPCR/RMV8_circular_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV8") %>%
      dplyr::mutate(Measurement = "BC total"),
   read_excision_file("./qPCR/RMV8_hybrid_circular_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV8_hybrid") %>%
      dplyr::mutate(Measurement = "BC total"),
   read_excision_file("./qPCR/RMV4_BC_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV4") %>%
      dplyr::mutate(Measurement = "BC total"),
   
   # Extracellular BC to rpoA ratio
   dplyr::bind_cols(
     read_excision_file("./qPCR/2743_RMV4_filtered_extra_ratio_with_Time0_v2.csv"),
     read.csv("./qPCR/2743_RMV4_filtered_extra_ratio_with_Time0_v2.csv") %>%
     dplyr::mutate(phage = dplyr::if_else(grepl("RMV4",mutants),"RMV4","08B02743")) %>%
     dplyr::select(phage)
   ) %>%
   dplyr::mutate(Measurement = "BC extracellular"),
   dplyr::bind_cols(
     read_excision_file("./qPCR/RMV8_RMV8_hybrid_filtered_extra_ratio_Time0_v2.csv"),
     read.csv("./qPCR/RMV8_RMV8_hybrid_filtered_extra_ratio_Time0_v2.csv") %>%
     dplyr::mutate(phage = dplyr::if_else(grepl("_Hybrid",mutants),"RMV8_hybrid","RMV8")) %>%
     dplyr::select(phage)
   ) %>%
   dplyr::mutate(Measurement = "BC extracellular")
  ) %>%
  dplyr::mutate(Time = factor(Time,
                                levels = c("0","20","60","120"))) %>%
  dplyr::mutate(phage = factor(dplyr::case_when(
                  phage == "RMV8" ~ "phi*'RMV8'",
                  phage == "RMV8_hybrid" ~ "phi*'RMV8'[italic(immARcat)]",
                  phage == "08B02743" ~ "phi*'08B02743'",
                  phage == "RMV4" ~ "phi*'RMV4'"
                ),
                levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]"))
  ) %>%
  dplyr::mutate(Measurement = factor(dplyr::case_when(
                  Measurement == "AD to AB ratio" ~ "italic(attB)~'to'~italic(attL)~'ratio'",
                  Measurement == "BC total" ~ "'Total'~italic(attP)",
                  Measurement == "BC extracellular" ~ "'Extracellular'~italic(attP)"
                ),
                levels = c("italic(attB)~'to'~italic(attL)~'ratio'","'Total'~italic(attP)","'Extracellular'~italic(attP)"))
  ) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,Measurement,phage,Time) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

phage_transcription.df <-
  dplyr::bind_rows(
    read_expression_file("./qPCR/2743_prophage_induction_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "08B02743"),
    read_expression_file("./qPCR/RMV4_prophage_induction_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV4"),
   read_expression_file("./qPCR/RMV8_prophage_induction_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV8"),
   read_expression_file("./qPCR/RMV8_hybrid_prophage_induction_1st_2nd_3rd.csv") %>%
      dplyr::mutate(phage = "RMV8_hybrid")
  ) %>%
  dplyr::filter(gene != "rpoA") %>%
  dplyr::mutate(gene = dplyr::if_else(gene == "lytA","lyt",gene)) %>%
  dplyr::mutate(Time = factor(Time,
                                levels = c("0","20","60","120"))) %>%
  dplyr::mutate(phage = factor(dplyr::case_when(
                  phage == "RMV8" ~ "phi*'RMV8'",
                  phage == "RMV8_hybrid" ~ "phi*'RMV8'[italic(immARcat)]",
                  phage == "08B02743" ~ "phi*'08B02743'",
                  phage == "RMV4" ~ "phi*'RMV4'"
                ),
                levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]"))
  ) %>%
  dplyr::mutate(gene = factor(paste0('italic(',gene,')'),
                                levels = c("italic(recA)","italic(int)","italic(reg)","italic(rep)","italic(lyt)"))) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,gene,phage,Time) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

```

# Plot expression and excision data

``` {r Plot expression and excision data, fig.width = 8, fig.height = 11}

expression_plot <-
   # Expression data
    ggplot(phage_transcription.df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Time,Condition,phage,gene),
               )
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      geom_hline(yintercept = 1, linetype = 2) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      facet_grid(gene~phage, scales = 'free_y', labeller = label_parsed) +
      theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines"))


excision_plot <-
   # Excision data
    ggplot(phage_excision.df,
           aes(x = Time,
               y = ratio,
               colour = Condition,
               fill = Condition,
               group = interaction(Time,Condition,phage,Measurement))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('DNA molecule ratio')) +
      xlab("Time (min)") +
      facet_grid(Measurement~phage, scales = 'free_y',
                 labeller = label_parsed) +
      theme(legend.position = "none",
            panel.spacing = unit(0, "lines"))


prophage_excision_summary_plot <-
  magick::image_ggplot(image_crop(image_trim((image_read_pdf("./summary_figure/prophage_excision_summary_figure.pdf")))))


(combined_expression_excision_plot <-
  cowplot::plot_grid(
    plotlist = list(
      expression_plot,
      prophage_excision_summary_plot,
      excision_plot),
    nrow = 3,
    rel_heights = c(1,0.15,0.7),
    labels = "AUTO"
  )
)

```

# Compare expression in RMV8 and RMV8 hybrid with and without RecA

``` {r RMV8 and RMV8 hybrid with and without RecA updated}

rmv8_hyrbid_recA_effect.df <-
  dplyr::bind_rows(
    read_expression_file("./qPCR/RMV8_recAKO_phage_induction_3replicates.csv") %>% 
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    phage = "phi*'RMV8'"),
    read_expression_file("./qPCR/04032024_RMV8_4th_replicate.csv") %>%
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    phage = "phi*'RMV8'"),
    read_expression_file("./qPCR/Hybrid_recAKO_induction_3_replicates.csv") %>% 
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    phage = "phi*'RMV8'[italic(immARcat)]")
  ) %>%
  dplyr::rename(Fold_change = fold_change) %>%
  dplyr::filter(gene != "RpoA" & gene != "rpoA") %>%
  dplyr::mutate(
      gene = dplyr::case_when(
        gene == "Int" ~ "italic(int)",
        gene == "comEA_2" ~ "italic(comEA)",
        gene == "RpoA" ~ "italic(rpoA)",
        gene == "immA_2743" ~ "italic(reg)",
        gene == "lytA" ~ "italic(lyt)",
        gene == "dnaC" ~ "italic(rep)",
        gene == "immA_2743" ~ "italic(reg)",
        gene == "lex" ~ "italic(reg)",
        TRUE ~ paste0("italic(",gene,")")
      )
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,gene,phage,Time) %>%
  dplyr::mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::filter(!(phage == "phi*'RMV8'" & RecA == "italic(recA)~'negative'" & `Biological replicate` == "3")) %>%
  dplyr::mutate(`Biological replicate` = dplyr::if_else(`Biological replicate` == "4","3",`Biological replicate`)) %>%
  dplyr::ungroup()

rmv8_hyrbid_recA_effect.df %<>%
  dplyr::bind_rows(
    phage_transcription.df %>%
      dplyr::filter(phage %in% c("phi*'RMV8'","phi*'RMV8'[italic(immARcat)]") & gene != "italic(rpoA)") %>%
      dplyr::mutate(RecA = "italic(recA)~'positive'")
  ) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(gene = factor(gene,levels = c("italic(recA)","italic(comEA)","italic(int)","italic(reg)","italic(rep)","italic(lyt)")))

(rmv8_recA_comparison_plot <-
    ggplot(rmv8_hyrbid_recA_effect.df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Condition,gene,phage,Time))
           ) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      geom_hline(yintercept = 1, linetype = 2) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~phage + RecA,
                 scales = 'free_y',
                 labeller = label_parsed)
)

```

# Expression in RecA mutant relative to wild type

``` {r RMV8 and RMV8 hybrid with and without RecA}

recA_standardised_wt.df <-
  dplyr::bind_rows(
    read_expression_file("./qPCR/RMV8_recA_comapred_RMV8_T0_3replicates.csv") %>%
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    base_phage = "RMV8",
                    phage = "phi*'RMV8'~italic(recA)*'::Janus'"),
    read_expression_file("./qPCR/RMV8_4th_replicate_normliased_to_WT_T0.csv") %>%
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    base_phage = "RMV8",
                    phage = "phi*'RMV8'~italic(recA)*'::Janus'"),
    read_expression_file("./qPCR/RMV8_hybrid_recA_normliased_RMV8hybrid_T0_three_replicates.csv") %>%
      dplyr::mutate(RecA = "italic(recA)~'negative'",
                    base_phage = "RMV8_hybrid",
                    phage = "phi*'RMV8'[italic(immARcat)]~italic(recA)*'::Janus'")
  ) %>%
  dplyr::rename(Fold_change = fold_change) %>%
  dplyr::filter(gene != "RpoA" & gene != "rpoA") %>%
  dplyr::mutate(
      gene = dplyr::case_when(
        gene == "Int" ~ "italic(int)",
        gene == "comEA_2" ~ "italic(comEA)",
        gene == "RpoA" ~ "italic(rpoA)",
        gene == "immA_2743" ~ "italic(reg)",
        gene == "lytA" ~ "italic(lyt)",
        gene == "dnaC" ~ "italic(rep)",
        gene == "immA_2743" ~ "italic(reg)",
        gene == "lex" ~ "italic(reg)",
        TRUE ~ paste0("italic(",gene,")")
      )
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(gene = factor(gene,
                              levels = c("italic(int)","italic(reg)","italic(rep)","italic(lyt)"))) %>%
  dplyr::group_by(Condition,gene,Time,base_phage) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup() %>%
  dplyr::filter(!(phage == "phi*'RMV8'~italic(recA)*'::Janus'" & `Biological replicate` == "3")) %>%
  dplyr::mutate(`Biological replicate` = dplyr::if_else(`Biological replicate` == "4","3",`Biological replicate`)) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'~italic(recA)*'::Janus'","phi*'RMV8'[italic(immARcat)]~italic(recA)*'::Janus'")))

recA_standardised_wt_plot <-
    ggplot(recA_standardised_wt.df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Condition,gene,base_phage,Time))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.5) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      geom_hline(yintercept = 1, linetype = 2) +
      theme(legend.position = "right",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~phage,
                 scales = 'free_y',
                 labeller = label_parsed)

```

# Effect of comEA mutation on gene expression

``` {r Effect of comEA mutation on gene expression}

comEA_phage_expression.df <-
  read_expression_file("./qPCR/RMV8_comEAKO_phage_induction_three_replicates.csv") %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(Genotype = "atop(phi*'RMV8',italic(comEA)~'::Janus')") %>%
  dplyr::filter(gene != "rpoA") %>%
  dplyr::mutate(gene = dplyr::if_else(gene == "RMV8_int","int",gene)) %>%
  dplyr::mutate(gene = dplyr::if_else(gene == "lytA","lyt",gene)) %>%
  dplyr::mutate(gene = paste0("italic(",gene,")")) %>%
  dplyr::rename(Fold_change = fold_change) %>%
  dplyr::group_by(Condition,gene,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

comEA_phage_expression.df %<>%
  dplyr::bind_rows(
    phage_transcription.df %>%
      dplyr::filter(phage == "phi*'RMV8'" & gene %in% c("italic(lyt)","italic(recA)","italic(int)")) %>%
      dplyr::mutate(Genotype = "phi*'RMV8'")
  ) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(gene = factor(gene,
                              levels = c("italic(recA)","italic(int)","italic(lyt)"))) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("phi*'RMV8'","atop(phi*'RMV8',italic(comEA)~'::Janus')")))

comEA_phage_induction_plot <-
    ggplot(comEA_phage_expression.df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Time,Condition,Genotype))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 cex = 0.75) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      geom_hline(yintercept = 1, linetype = 2) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed)

```

# Effect of comEA on phage excision

``` {r RMV8 comEA mutants}

rmv8_comEA.df <-
  dplyr::bind_rows(
    read_excision_file("./qPCR/comEA_cat_extracellular_BCbyrpoAthree_replicates.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular") %>%
      dplyr::rename(value = Extracellular_BC),
    read_excision_file("./qPCR/comEA_mutants_ADbyAB_three_replicates_July.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio") %>%
      dplyr::rename(value = ADbyAB),
    read_excision_file("./qPCR/comEA_mutants_BCbyRpoA_three_replicates_July.csv") %>%
      dplyr::mutate(Measurement = "BC total") %>%
      dplyr::rename(value = BCbyrpoA)
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C")),
                Genotype = "atop(phi*'RMV8',italic(comEA)~'::Janus')") %>%
  dplyr::rename(ratio = value) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(Measurement = factor(dplyr::case_when(
                  Measurement == "AD to AB ratio" ~ "italic(attB)~'to'~italic(attL)~'ratio'",
                  Measurement == "BC total" ~ "'Total'~italic(attP)",
                  Measurement == "BC extracellular" ~ "'Extracellular'~italic(attP)"
                ),
                levels = c("italic(attB)~'to'~italic(attL)~'ratio'","'Total'~italic(attP)","'Extracellular'~italic(attP)"))
  ) %>%
  dplyr::group_by(Condition,Measurement,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()


rmv8_comEA.df %<>%
  dplyr::bind_rows(
    phage_excision.df %>%
      dplyr::filter(phage == "phi*'RMV8'") %>%
      dplyr::mutate(Genotype = "phi*'RMV8'")
  ) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("phi*'RMV8'","atop(phi*'RMV8',italic(comEA)~'::Janus')")))

comEA_phage_excision_plot <-
  ggplot(rmv8_comEA.df,
         aes(x = Time,
             y = ratio,
             colour = Condition,
             fill = Condition,
             group = interaction(Time,Genotype,Condition,Measurement))
         ) +
    geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                alpha = 0.15,
                scale = "width",
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
               position = position_jitterdodge(jitter.width = 0.25),
               size = 0.75) +
    scale_y_continuous(trans = "log10") +
    scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
    ylab(expression('DNA molecule ratio')) +
    xlab("Time (min)") +
    facet_grid(Measurement~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed) +
    theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines"),
          strip.text.y = element_text(size = 7))

```

# Test for in situ replication in RMV8

``` {r in situ replication test}

in_situ.df <-
  dplyr::bind_rows(
    read_excision_file("./qPCR/RMV8_repbyrpoA_1st2nd.csv") %>%
      dplyr::mutate(Genotype = "phi*'RMV8'"),
    read_excision_file("./qPCR/RMV8_intcat_repbyrpoA_1st2nd.csv")  %>%
      dplyr::mutate(Genotype = "phi*'RMV8'~italic(int)*'::Janus'")
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

in_situ_replication_plot <-
  ggplot(in_situ.df,
         aes(x = as.factor(Time),
             y = repbyrpoA,
             colour = Condition,
             fill = Condition,
             group = interaction(Time,Genotype,Condition))
         ) +
    geom_violin(colour = "black",
                alpha = 0.15,
                draw_quantiles = c(0.5),
                scale = "width",
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
               position = position_jitterdodge()) +
    scale_y_continuous(trans = "log10") +
    ylab(expression('Relative'~italic(rep)~'DNA concentration')) +
    xlab("Time (min)") +
    scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
    facet_grid(.~Genotype, labeller = label_parsed) +
    theme(panel.spacing = unit(0, "lines"))

# Test for in situ replication in int mutants after 4 h
in_situ_int_mutant.df <-
  read.csv("./growth_curves/RMV8_Hybrid_intcat_2hour_4hour_repbyrpoA.csv") %>%
  dplyr::mutate(Genotype = dplyr::if_else(
                  grepl("hybrid",mutants),
                  "RMV8_hybrid_intcat",
                  "RMV8_intcat"
                ),
                `Growth condition` = dplyr::case_when(
                  grepl("MMC",mutants) ~ "Mitomycin C",
                  grepl("CSP",mutants) ~ "CSP",
                  TRUE ~ "No supplement"
                )
  ) %>%
  dplyr::filter(`Growth condition` != "CSP") %>%
  dplyr::mutate(`Growth condition` = factor(`Growth condition`,
                                            levels = c("No supplement","Mitomycin C"))) %>%
  dplyr::group_by(`Growth condition`,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

int_in_situ_replication_plot <-
  ggplot(in_situ_int_mutant.df,
         aes(x = Genotype,
             y = repbyrpoA,
             colour = `Growth condition`,
             fill = `Growth condition`,
             group = interaction(`Growth condition`,Genotype))
         ) +
    geom_violin(alpha = 0.15,
                colour = "black",
                draw_quantiles = c(0.5),
                scale = "width",
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
               position = position_jitterdodge(jitter.width = 0.5)) +
    #scale_x_discrete(labels = mutant_labels[names(mutant_labels) %in% unique(in_situ_int_mutant.df$Genotype)]) +
    scale_x_discrete(labels = c(expression(atop("" ~ phi * "RMV8",italic(int) * "::" * italic(cat) ~ "")),
                                expression(atop("" ~ phi * "RMV8"[italic(immARcat)],italic(int) * "::" * italic(cat) ~ "")))) +
    ylab(expression('Relative'~italic(rep)~'DNA concentration')) +
    scale_y_continuous(trans = "log10") +
    scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour"))

cowplot::plot_grid(
  plotlist = list(
    cowplot::plot_grid(
      plotlist = list(
        in_situ_replication_plot + theme(legend.position = "none"),
        int_in_situ_replication_plot + theme(legend.position = "none")
      ),
      nrow = 1,
      rel_widths = c(0.6,0.3),
      labels = "AUTO"
    ),
    cowplot::get_plot_component(in_situ_replication_plot + theme(legend.position = "bottom"),
                                'guide-box-bottom',
                                return_all = TRUE)
    ),
    nrow = 2,
  rel_heights = c(0.9,0.1)
)

```

# Effect of DprA on competence gene expression

``` {r 08B02743 hybrid with and without DprA}

dprA_standardised_wt.df <-
  dplyr::bind_rows(
    read_expression_file("./qPCR/2743_dprA_competence_transcriptions.csv") %>%
      dplyr::mutate(Genotype = "atop(phi*'08B02743',italic(dprA)*'::Janus')"),
    read_expression_file("./qPCR/2743_dprA_transcription_two_replicates.csv") %>%
      dplyr::mutate(Genotype = "phi*'08B02743'")
  ) %>%
  dplyr::rename(Fold_change = fold_change) %>%
  dplyr::filter(gene != "RpoA" & gene != "rpoA") %>%
  dplyr::mutate(
      gene = paste0("italic(",gene,")")
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,gene,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

dprA_standardised_wt.df %<>%
  dplyr::bind_rows(
    phage_transcription.df %>%
      dplyr::filter(phage == "phi*'08B02743'" & gene %in% c("italic(lyt)","italic(recA)","italic(int)","italic(dprA)","italic(reg)")) %>%
      dplyr::mutate(Genotype = "phi*'08B02743'"),
    read_expression_file("./qPCR/2743_comEA_foldchange.csv") %>%
      dplyr::mutate(phage = "phi*'08B02743'",
                    Genotype = "phi*'08B02743'") %>%
      dplyr::mutate(
          gene = paste0("italic(",gene,")")
      ) %>%
      dplyr::rename(Fold_change = fold_change) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,gene,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()
  ) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(gene = factor(gene,
                              levels = c("italic(comEA)","italic(recA)","italic(dprA)","italic(int)","italic(reg)","italic(lyt)"))) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("phi*'08B02743'",
                                             "atop(phi*'08B02743',italic(dprA)*'::Janus')")))


(dprA_standardised_wt_plot <-
    ggplot(dprA_standardised_wt.df %>%
             dplyr::filter(
               gene %in% c("italic(recA)","italic(comEA)","italic(dprA)") &
                 Condition %in% c("No supplement","CSP","Mitomycin C")
             ),
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Condition,gene,Time))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      theme(legend.position = "right",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed))

```

# Effect of DprA on phage excision

``` {r 2743 dprA mutants and phage excision}

read_dprA_excision_file <- function(fn) {
  
  df <-
    read.csv(fn) %>%
    dplyr::mutate(Time = dplyr::case_when(
            grepl("_T0",mutants) ~ "0",
            grepl("_T20",mutants) ~ "20",
            grepl("_T60",mutants) ~ "60",
            grepl("_T120",mutants) ~ "120",
            grepl("T0_",mutants) ~ "0",
            grepl("T20_",mutants) ~ "20",
            grepl("T60_",mutants) ~ "60",
            grepl("T120_",mutants) ~ "120",
            grepl("T0+",mutants) ~ "0",
            grepl("T20+",mutants) ~ "20",
            grepl("T60+",mutants) ~ "60",
            grepl("T120+",mutants) ~ "120",
        )
    ) %>%
    dplyr::mutate(Condition = dplyr::case_when(
            grepl("+CSP",mutants) ~ "CSP",
            grepl("_CSP",mutants) ~ "CSP",
            grepl("+MMC",mutants) ~ "Mitomycin C",
            grepl("_MMC",mutants) ~ "Mitomycin C",
            TRUE ~ "No supplement"
        )
    )
  
}

phage_excision_dprA.df <-
  dplyr::bind_rows(
    phage_excision.df %>%
      dplyr::filter(phage == "phi*'08B02743'") %>%
      dplyr::mutate(mutants = "2743",
                    Genotype = "atop(phi*'08B02743','')") %>%
      dplyr::rename(value = ratio),
      
    # Extracellular BC in dprA mutant
    read_dprA_excision_file("./qPCR/2743_dprA_extracellular_1706.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "atop(phi*'08B02743',italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743'"
                    ) %>%
      dplyr::rename(value = BCbyrpoA),
    
    # ZnCat mutants - extracellular
    read_dprA_excision_file("./qPCR/2743_dprA_zncat_extracellular_1706.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743',italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyrpoA),
    
    # WT - intracellular
    read_dprA_excision_file("./qPCR/2743_dprA_intracellular_BC_1706.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "atop(phi*'08B02743',italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743'"
                    ) %>%
      dplyr::rename(value = BCbyrpoA),
    
    # DprA zncat intracellular
    read_dprA_excision_file("./qPCR/2743_zncat_dprA_BC_intracellular_1706.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyrpoA),
    
    # zncat extracellular
    read_dprA_excision_file("./qPCR/2743_zncat_extracellular_BC.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    # zncat total
    read_dprA_excision_file("./qPCR/2743_zncat_intracellular_BC.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    # RecA extracellular replicates
    read_dprA_excision_file("./qPCR/recA_KO_extracellular_DNA_three.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "atop(phi*'08B02743',italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    # RecA total replicates
    read_dprA_excision_file("./qPCR/2743_recA_intracellular_BC_three_replicates.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "atop(phi*'08B02743',italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    
    # 2743 zncat RecA intracellular replicates
    read_dprA_excision_file("./qPCR/2743_zncat_recA_intracellular_BC.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    # 2743 zncat RecA extracellular replicates
    read_dprA_excision_file("./qPCR/2743_zncat_recA_extracellular_BC.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = BCbyRpoA),
    
    # 2743 RecA excision
    read_dprA_excision_file("./qPCR/2743_recA_ADbyAB_two.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "atop(phi*'08B02743',italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'"
                    ) %>%
      dplyr::rename(value = ADbyAB),
    
    # 2743 DprA excision
    read_dprA_excision_file("./qPCR/2743_dprA_ADbyAB_two_replicates.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "atop(phi*'08B02743',italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743'",
                    mutants = dplyr::if_else(grepl("_2nd_",mutants),gsub("_2nd_","_",mutants),mutants)
                    ) %>%
      dplyr::rename(value = ADbyAB),
    
    # 2743 zncat excision
    read_dprA_excision_file("./qPCR/2743_zncat_ADbyAB_three.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = ADbyAB),
    
    ######
    # 2743 zncat dprA excision
    read_dprA_excision_file("./qPCR/2743_zncat_dprA_two_ADbyAB.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(dprA)*'::Janus')",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = ADbyAB),
    
    # 2743 zncat recA excision
    read_dprA_excision_file("./qPCR/2743_zncat_recA_ADbyAB_two.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(recA)*'::Janus')",
                    phage = "phi*'08B02743'~italic(szaL)*'::'*italic(cat)"
                    ) %>%
      dplyr::rename(value = ADbyAB)
    
  ) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C")),
                Genotype = factor(Genotype,
                                  levels = c("atop(phi*'08B02743','')",
                                             "atop(phi*'08B02743',italic(recA)*'::Janus')",
                                             "atop(phi*'08B02743',italic(dprA)*'::Janus')",
                                             "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))",
                                            "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(recA)*'::Janus')", "atop(phi*'08B02743'~italic(szaL)*'::'*italic(cat),italic(dprA)*'::Janus')"))) %>%
  dplyr::rename(ratio = value) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(Measurement = factor(dplyr::case_when(
                  Measurement == "AD to AB ratio" ~ "italic(attB)~'to'~italic(attL)~'ratio'",
                  Measurement == "BC total" ~ "'Total'~italic(attP)",
                  Measurement == "BC extracellular" ~ "'Extracellular'~italic(attP)",
                  TRUE ~ Measurement
                ),
                levels = c("italic(attB)~'to'~italic(attL)~'ratio'","'Total'~italic(attP)","'Extracellular'~italic(attP)"))
  ) %>%
  dplyr::group_by(Condition,Measurement,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

(phage_excision_dprA_plot <-
  ggplot(phage_excision_dprA.df,
         aes(x = Time,
             y = ratio,
             colour = Condition,
             fill = Condition,
             group = interaction(Time,Genotype,Condition,Measurement))
         ) +
    geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                alpha = 0.15,
                scale = "width",
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
               position = position_jitterdodge(jitter.width = 0.25),
               size = 0.75) +
    scale_y_continuous(trans = "log10") +
    scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
    ylab(expression('DNA molecule ratio')) +
    xlab("Time (min)") +
    facet_grid(Measurement~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed) +
    theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines"),
          strip.text = element_text(size = 7.5))
)

```

Is the decision made at excision?

``` {r Decision at excision, fig.width = 10}

decision_at_excision_df <-
  phage_excision_dprA.df %>%
    dplyr::group_by(Time,Condition,phage,`Biological replicate`,mutants,Genotype,Measurement) %>%
    mutate(`Technical replicate` = as.character(row_number())) %>%
    dplyr::ungroup() %>%
    tidyr::pivot_wider(id_cols = c(Time,Condition,`Biological replicate`,`Technical replicate`,Genotype),
                       names_from = Measurement,
                       values_from = ratio) %>%
    dplyr::filter(!is.na(`italic(attB)~'to'~italic(attL)~'ratio'`) & !is.na(`'Total'~italic(attP)`) & !is.na(`'Extracellular'~italic(attP)`))

cowplot::plot_grid(
  plotlist = list(
    ggplot(decision_at_excision_df,
       aes(x = `italic(attB)~'to'~italic(attL)~'ratio'`,
           y = `'Total'~italic(attP)`,
           colour = Condition,
             fill = Condition,
           shape = Genotype)) +
      geom_point() +
      scale_colour_manual(values = condition_palette,
                          aesthetics = c("fill","colour")) +
      scale_y_continuous(trans = "log10") +
      scale_x_continuous(trans = "log10") +
      theme_bw() +
      theme(legend.position = "none"),
    ggplot(decision_at_excision_df,
       aes(x = `'Total'~italic(attP)`,
           y = `'Extracellular'~italic(attP)`,
           colour = Condition,
             fill = Condition,
           shape = Genotype)) +
      geom_point() +
      scale_colour_manual(values = condition_palette,
                          aesthetics = c("fill","colour")) +
      scale_y_continuous(trans = "log10") +
      scale_x_continuous(trans = "log10") +
      theme_bw() +
      theme(legend.position = "right")
  ),
  ncol = 2,
  rel_widths = c(1,2.15)
)


```

# Effect of SpaL on phage excision

``` {r RMV4 spaL mutants and phage excision}

phage_excision_spaL.df <-
  dplyr::bind_rows(
    # Wild type genotype
    phage_excision.df %>%
      dplyr::filter(phage == "phi*'RMV4'") %>%
      dplyr::mutate(mutants = phage,
                    Genotype = phage),
    dplyr::bind_rows(
      # Excision
      read_dprA_excision_file("./qPCR/RMV4_spaL_ADbyAB_three_replicates.csv") %>%
        dplyr::mutate(Measurement = "AD to AB ratio",
                      Genotype = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)",
                      phage = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)"
                      ) %>%
        dplyr::rename(value = ADbyAB),
      # Extracellular
      read_dprA_excision_file("./qPCR/RMV4_spaL_extracellular_BC_three_replicates.csv") %>%
        dplyr::mutate(Measurement = "BC extracellular",
                      Genotype = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)",
                      phage = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)"
                      ) %>%
        dplyr::rename(value = BCbyRpoA),
      # Total
      read_dprA_excision_file("./qPCR/RMV4_spaL_intracellular_BC_three_replicates.csv") %>%
        dplyr::mutate(Measurement = "BC total",
                      Genotype = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)",
                      phage = "phi*'RMV4'~italic(spaL)*'::'*italic(cat)"
                      ) %>%
        dplyr::rename(value = BCbyRpoA),
      # DprA
      read_dprA_excision_file("./qPCR/RMV4_dprA_induction_ADbyAB.csv") %>%
      dplyr::mutate(Measurement = "AD to AB ratio",
                    Genotype = "phi*'RMV4'~italic(dprA)*'::Janus'",
                    phage = "phi*'RMV4'~italic(dprA)*'::Janus'"
                    ) %>%
            dplyr::rename(value = ADbyAB),
      read_dprA_excision_file("./qPCR/RMV4_dprA_intracellular_BCbyRpoA_two_replicates.csv") %>%
      dplyr::mutate(Measurement = "BC total",
                    Genotype = "phi*'RMV4'~italic(dprA)*'::Janus'",
                    phage = "phi*'RMV4'~italic(dprA)*'::Janus'"
                    ) %>%
            dplyr::rename(value = BCbyRpoA),
      read_dprA_excision_file("./qPCR/RMV4_dprA_extracellular_BCbyRpoA_two_replicates.csv") %>%
      dplyr::mutate(Measurement = "BC extracellular",
                    Genotype = "phi*'RMV4'~italic(dprA)*'::Janus'",
                    phage = "phi*'RMV4'~italic(dprA)*'::Janus'"
                    ) %>%
            dplyr::rename(value = BCbyRpoA)
    ) %>%
    dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                  Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))
                  ) %>%
    dplyr::rename(ratio = value) %>%
    dplyr::mutate(Condition = factor(Condition,
                                  levels = c("No supplement","CSP","Mitomycin C"))) %>%
    dplyr::mutate(Measurement = factor(dplyr::case_when(
                    Measurement == "AD to AB ratio" ~ "italic(attB)~'to'~italic(attL)~'ratio'",
                    Measurement == "BC total" ~ "'Total'~italic(attP)",
                    Measurement == "BC extracellular" ~ "'Extracellular'~italic(attP)"
                  ),
                  levels = c("italic(attB)~'to'~italic(attL)~'ratio'","'Total'~italic(attP)","'Extracellular'~italic(attP)"))
    ) %>%
    dplyr::group_by(Condition,Measurement,Time,Genotype) %>%
    mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
    dplyr::ungroup()
  ) %>%
    dplyr::group_by(Condition,Measurement,Time,Genotype,`Biological replicate`) %>%
    dplyr::mutate(median_y = median(ratio),
                  min_y = min(ratio),
                  max_y = max(ratio)) %>%
    dplyr::ungroup()

add_median_range <- function(p, data, group_vars, value_var) {
  # Ensure group_vars and value_var are symbols
  value_var <- enquo(value_var)
  
  # Summarize the data to get the median, min, and max for each group
  summary_data <- data %>%
    group_by_at(group_vars) %>%
    summarize(
      median_value = median(!!value_var, na.rm = TRUE),
      min_value = min(!!value_var, na.rm = TRUE),
      max_value = max(!!value_var, na.rm = TRUE)
    )
  
  # Convert group_vars to symbols for aes mapping
  group_vars_syms <- syms(group_vars)
  
  # Add the median points and error bars to the existing ggplot object
  p <- p +
    geom_point(data = summary_data, aes(!!!group_vars_syms, y = median_value), color = "blue", size = 3) +
    geom_errorbar(data = summary_data, aes(!!!group_vars_syms, ymin = min_value, ymax = max_value), width = 0.2, color = "red") +
    labs(y = quo_name(value_var))
  
  return(p)
}

(phage_excision_spaL_plot <-
  ggplot(phage_excision_spaL.df,
         aes(x = Time,
             y = ratio,
             colour = Condition,
             fill = Condition,
             group = interaction(Time,Genotype,Condition,Measurement))
         ) +
    geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
    geom_violin(draw_quantiles = c(0.5),
                colour = "black",
                alpha = 0.15,
                scale = "width",
                position = position_dodge(preserve = 'single')) +
    geom_point(aes(shape = `Biological replicate`),
                position = position_jitterdodge(jitter.width = 0.25),
               size = 0.75) +
    #add_median_range() +
    scale_y_continuous(trans = "log10") +
    scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
    ylab(expression('DNA molecule ratio')) +
    xlab("Time (min)") +
    facet_grid(Measurement~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed) +
    theme(legend.position = "bottom",
            panel.spacing = unit(0, "lines"),
          strip.text.y = element_text(size = 7.5))
)

```

# Kinetics of deletion

``` {r Kinetics of deletions, fig.height = 8, fig.width = 8}

deletion_kinetics.df <-
  read.csv("./transformation_data/prophage_deletion_over_time.csv") %>%
  dplyr::filter(time < 500) %>%
  dplyr::mutate(time = factor(time)) %>%
  dplyr::mutate(
    Genotype = dplyr::case_when(
      genotype == "RMV8" ~ "RMV8",
      genotype == "RMV8 prophage::cat" ~ "RMV8_phageCat",
      genotype == "RMV8 Integrase::cat" ~ "RMV8_intcat",
      genotype == "RMV8 lytA::cat" ~ "RMV8_lytAcat"
    )
  )

cowplot::plot_grid(
  plotlist = list(
    ggplot(deletion_kinetics.df,
       aes(x = time,
           y = kan_per_ul,
           colour = Genotype,
           fill = Genotype)) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  position = position_dodge(preserve = 'single'),
                  width = 0.5) +
      geom_point(position = position_jitterdodge(dodge.width = 0.5, jitter.width = 0.05)) +
      ylab(expression('Kanamycin-resistant colonies per'~mu*'l')) +
      xlab("Time (min)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(deletion_kinetics.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
      theme(legend.position = "none"),
    ggplot(deletion_kinetics.df,
           aes(x = time,
               y = rif_per_ul,
               fill = Genotype,
               colour = Genotype)) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  position = position_dodge(preserve = 'single'),
                  width = 0.5) +
      geom_point(position = position_jitterdodge(dodge.width = 0.5, jitter.width = 0.05)) +
      ylab(expression('Rifampicin-resistant colonies per'~mu*'l')) +
      xlab("Time (min)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(deletion_kinetics.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
      theme(legend.position = "none"),
    cowplot::get_plot_component(
      ggplot(deletion_kinetics.df,
           aes(x = time,
               y = rif_per_ul,
               fill = Genotype,
               colour = Genotype)) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  position = position_dodge(preserve = 'single')) +
      geom_point(position = position_jitterdodge()) +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(deletion_kinetics.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
      theme(legend.position = "bottom"),
      'guide-box-bottom',
      return_all = TRUE
    )
  ),
  nrow = 3,
  labels = c("A","B",""),
  rel_heights = c(1,1,0.2)
)

```

# Effect of RecA induction by IPTG

``` {r LacR induction}

iptg_induction.df <-
  dplyr::bind_rows(
    # RMV8
    get_growth_curve_df(read.csv("./qPCR/RMV8_pLac_recA.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8'~italic(recA)[italic(lacI)]"),
    # RMV8 hybrid
    get_growth_curve_df(read.csv("./qPCR/MMC_IPTG_RMV8_hybrid_plac_recA.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_hybrid_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8_hybrid'") %>%
      dplyr::mutate(phage = "phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]")
  ) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'~italic(recA)[italic(lacI)]","phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"))) %>%
  dplyr::mutate(IPTG = factor(IPTG,
                  levels = c("No IPTG","IPTG")))



# guide legend
guide_nrow = 1
g <- guide_legend(nrow = guide_nrow,
                    title = NULL)

# plot
iptg_growth_plot <-
  ggplot(iptg_induction.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = MMC,
               fill = MMC,
               linetype = IPTG,
               group = interaction(IPTG,MMC))) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g, linetype = g) +
      facet_grid(phage~.,
                 scales = 'free_x',
                 labeller = label_parsed)

```

## Effect of IPTG induction on transcription

``` {r Effect of IPTG on phage transcription}

iptg_phage_expression_df <-
  read.csv("./qPCR/IPTG_recA_indcued_RMV8vsHybrid.csv") %>%
    dplyr::mutate(Time = dplyr::case_when(
            grepl("T0_",mutants) ~ "0",
            grepl("_T0",mutants) ~ "0",
            grepl("T20_",mutants) ~ "20",
            grepl("_T20",mutants) ~ "20",
            grepl("T60_",mutants) ~ "60",
            grepl("_T60",mutants) ~ "60",
            grepl("T120_",mutants) ~ "120",
            grepl("_T120",mutants) ~ "120"
        )
    ) %>%
    dplyr::mutate(Condition = dplyr::case_when(
            grepl("_CSP",mutants) ~ "CSP",
            grepl("_MMC",mutants) ~ "Mitomycin C",
            TRUE ~ "No supplement"
        )
    ) %>%
    dplyr::mutate(gene  = dplyr::case_when(
        gene == "RMV8_int" ~ "int",
        gene == "RMV8_lytA" ~ "lyt",
        gene == "Imma_2743" ~ "reg",
        gene == "immA" ~ "reg",
        gene == "Lex" ~ "reg",
        gene == "Int_RMV8" ~ "int",
        gene == "dnaC" ~ "rep",
        TRUE ~ gene
      )
    ) %>%
    dplyr::mutate(gene = paste0('italic(',gene,')')) %>%
    dplyr::mutate(
      IPTG = dplyr::if_else(grepl("IPTG",mutants),"IPTG","No IPTG"),
      Genotype = dplyr::if_else(grepl("_hybrid",mutants),"RMV8_hybrid","RMV8"),
      phage = dplyr::if_else(grepl("_hybrid",mutants),"phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]","phi*'RMV8'~italic(recA)[italic(lacI)]")
    ) %>%
    dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'~italic(recA)[italic(lacI)]","phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"))) %>%
    dplyr::mutate(IPTG = factor(paste0("'",IPTG,"'"),
                  levels = c("'No IPTG'","'IPTG'"))) %>%
  dplyr::group_by(Genotype,Time,IPTG,phage,gene) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(Time = factor(as.character(Time), levels = c("0","60","120")))

ggplot(iptg_phage_expression_df,
       aes(x = Time,
           y = fold_change,
           colour = Genotype,
           fill = Genotype,
           group = interaction(mutants,Time,gene),
           shape = `Biological replicate`),
       ) +
  geom_violin(draw_quantiles = c(0.5),
              colour = "black",
              scale = "width",
              alpha = 0.15,
              position = position_dodge(preserve = 'single'),
              width = 0.5) +
  geom_point(position = position_jitter(width = 0.1),
             size = 0.75) +
  geom_hline(yintercept = 1, linetype = 2) +
  ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
  xlab("Time (min)") +
  scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(iptg_phage_expression_df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
  facet_grid(gene~phage, labeller = label_parsed) +
  theme(legend.position = "none",
            panel.spacing = unit(0, "lines"))

```

# Validation of IPTG

``` {r Validation of IPTG}

iptg_validation.df <-
  read.csv("./qPCR/IPTG_recA_Inudction_final.csv") %>%
  dplyr::mutate(Time = as.numeric(gsub("T","",condition))) %>%
  dplyr::mutate(phage = factor(dplyr::if_else(mutants == "RMV8_plac_recA",
                                       "phi*'RMV8'~italic(recA)[italic(lacI)]",
                                       "phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"),
                               levels = c("phi*'RMV8'~italic(recA)[italic(lacI)]","phi*'RMV8'[italic(immARcat)]~italic(recA)[italic(lacI)]"))
  ) %>%
  dplyr::group_by(mutants,condition) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()

ggplot(iptg_validation.df,
       aes(x = as.factor(Time),
           y = RecA_fold_change,
           colour = mutants,
           fill = mutants,
           group = interaction(mutants,as.factor(Time)),
           shape = `Biological replicate`),
       ) +
  geom_violin(draw_quantiles = c(0.5),
              colour = "black",
              scale = "width",
              alpha = 0.15,
              position = position_dodge(preserve = 'single'),
              width = 0.5) +
  geom_point(position = position_jitter(width = 0.1)) +
  geom_hline(yintercept = 1, linetype = 3) +
  ylab(expression('Relative'~italic(recA)~'RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
  xlab("Time (min)") +
  scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(iptg_validation.df$mutants)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
  facet_wrap(.~phage, labeller = label_parsed) +
  theme(legend.position = "none",
            panel.spacing = unit(0, "lines"))

```

# Effect of RecA induction by IPTG with half MMC

``` {r Replot growth curves with half MMC}

iptg_half_mmc_induction.df <-
  dplyr::bind_rows(
    # RMV8
    get_growth_curve_df(read.csv("./growth_curves/halfMMC_RMV8.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8'"),
    # RMV8 pLac mutant 1
    get_growth_curve_df(read.csv("./growth_curves/halfMMC_RMV8_DT3_plac_recA.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_hybrid_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8 LacI mutant 1'") %>%
      dplyr::mutate(phage = "phi*'RMV8'"),
    # RMV8 hybrid
    get_growth_curve_df(read.csv("./growth_curves/halfMMC_hybrid.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_hybrid_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8 hybrid'"),
    # RMV8 hybrid pLac mutant 1
    get_growth_curve_df(read.csv("./growth_curves/halfMMC_hybrid_DT1_plac_recA.csv") %>%
                          dplyr::rename_with(
                            ~ gsub("_\\.",".",gsub("_$","",paste0("RMV8_hybrid_", gsub("control","",.x), recycle0 = TRUE))),
                            !starts_with("OD600")
                          )
                        ) %>%
      dplyr::mutate(Genotype = "'RMV8 LacI mutant 1'") %>%
      dplyr::mutate(phage = "phi*'RMV8 hybrid'")#,
  ) %>%
  dplyr::mutate(recA = dplyr::if_else(grepl("Lac",Genotype),"italic(recA)[italic(lacI)]","italic(recA)[native]")) %>%
  dplyr::mutate(Genotype =
                    dplyr::case_when(
                      phage == "phi*'RMV8'" & Genotype == "'RMV8'" ~ "RMV8",
                      phage == "phi*'RMV8'" & Genotype == "'RMV8 LacI mutant 1'" ~ "RMV8_plac_recA",
                      phage == "phi*'RMV8'" & Genotype == "'RMV8 LacI mutant 2'" ~ "RMV8_plac_recA",
                      phage == "phi*'RMV8 hybrid'" & Genotype == "'RMV8'" ~ "RMV8_hybrid",
                      phage == "phi*'RMV8 hybrid'" & Genotype == "'RMV8 LacI mutant 1'" ~ "RMV8_hybrid_plac_recA",
                      phage == "phi*'RMV8 hybrid'" & Genotype == "'RMV8 LacI mutant 2'" ~ "RMV8_hybrid_plac_recA",
                    )
                  )

iptg_half_mmc_induction_normalised.df <-
  iptg_half_mmc_induction.df %>%
    dplyr::group_by(Genotype,MMC,Time,recA) %>%
    dplyr::mutate(no_IPTG = dplyr::if_else(IPTG == "No IPTG",Median,NA_real_)) %>%
    tidyr::fill(no_IPTG,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(IPTG_Median = Median - no_IPTG,
                  IPTG_Min = Min - no_IPTG,
                  IPTG_Max = Max - no_IPTG) %>%
    dplyr::filter(IPTG != "No IPTG") %>%
    dplyr::mutate(MMC = factor(dplyr::if_else(MMC == "Mitomycin C","'IPTG & Mitomycin C'","'IPTG'"),
                  levels = c("'IPTG'","'IPTG & Mitomycin C'"))) %>%
    dplyr::mutate(phage = factor(dplyr::if_else(phage=="phi*'RMV8'","phi*'RMV8'","phi*'RMV8'[italic(immARcat)]"),
                                 levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV8'[italic(immARcat)]"))) %>%
    dplyr::mutate(recA = factor(recA, levels = c("italic(recA)[native]","italic(recA)[italic(lacI)]")))

# plot
ggplot(iptg_half_mmc_induction_normalised.df %>%
         dplyr::mutate(IPTG = paste0("'",IPTG,"'")),
         aes(x = Time,
             y = IPTG_Median,
             ymin = IPTG_Min,
             ymax = IPTG_Max,
             colour = Genotype,
             fill = Genotype)) +
    geom_ribbon(alpha = 0.5, linetype = 0) +
    geom_line() +
    ylab(expression("Effect on growth ("*Delta*OD[600]*")")) +
    xlab("Time (h)") +
    geom_hline(yintercept = 0, linetype = 3) +
    scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(iptg_half_mmc_induction_normalised.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
    theme_bw() +
    theme(legend.position = "none",
          legend.text = element_text(size = 8),
          strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
    guides(colour = g, fill = g) +
    facet_grid(IPTG~phage+recA,
               labeller = label_parsed)

```

# Plot growth curves with different stimuli

``` {r Growth curves with different stimuli, fig.width = 8, fig.height = 11}

raw_growth_comparison_induction.df <-
  dplyr::bind_rows(
    # RMV4
    get_growth_curve_df(read.csv("./growth_curves/RMV4_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4'") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Background = "'Wild type'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_prophage_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4'") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Background = "'prophage::Janus'") %>%
      dplyr::filter(`Growth condition` != "Mitomycin C"),
    # Re-run of MMC
    get_growth_curve_df(read.csv("./growth_curves/RMV4-prophageKO.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4'") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Background = "'prophage::Janus_rerun'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_recAKO_metals_2nd.csv")) %>%
      dplyr::mutate(phage = "phi*'RMV4'",
                    Genotype = "'RMV4'",
                    Background = "italic(recA)*'::Janus'"),
    # RMV8
    get_growth_curve_df(read.csv("./growth_curves/RMV8_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Background = "'Wild type'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_recAKO_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Background = "italic(recA)*'::Janus'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_prophageKO_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV8'") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Background = "'prophage::Janus'"),
    # 2743
    get_growth_curve_df(read.csv("./growth_curves/2743_metals_1st.csv")) %>%
      dplyr::mutate(Genotype = "'08B02743'") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Background = "'Wild type'"),
    get_growth_curve_df(read.csv("./growth_curves/2743_recA_metals_test.csv")) %>%
      dplyr::mutate(Genotype = "'08B02743'") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Background = "italic(recA)*'::Janus'"),
    get_growth_curve_df(read.csv("./growth_curves/2743_prophageKO_metals.csv")) %>%
      dplyr::mutate(Genotype = "'08B02743'") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Background = "'prophage::Janus'"),
    # RMV8 hybrid
    get_growth_curve_df(read.csv("./growth_curves/RMV8_hybrid_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV8_Znregcat'") %>%
      dplyr::mutate(phage = "phi*'RMV8'[italic(immARcat)]") %>%
      dplyr::mutate(Background = "'Wild type'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_hybrid_recAKO_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV8_Znregcat'") %>%
      dplyr::mutate(phage = "phi*'RMV8'[italic(immARcat)]") %>%
      dplyr::mutate(Background = "italic(recA)*'::Janus'"),
    get_growth_curve_df(read.csv("./growth_curves/29022024_2743_sc1.csv")) %>%
      dplyr::mutate(Background = "'Wild type'") %>%
      dplyr::mutate(phage = "phi*'08B02743_MMC_repeat'") %>%
      dplyr::mutate(Genotype = "'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/29022024_2743_R1_RecA.csv")) %>%
      dplyr::mutate(Background = "italic(recA)*'::Janus'") %>%
      dplyr::mutate(phage = "phi*'08B02743_MMC_repeat'") %>%
      dplyr::mutate(Genotype = "'08B02743'")
  )

growth_comparison_induction.df <-
  raw_growth_comparison_induction.df %>%
  # Calculate relative growth
  dplyr::group_by(Genotype,phage,Background,Time) %>%
  dplyr::mutate(
    Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
    Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
    Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
  ) %>%
  tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
              .direction = "updown") %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Adjusted_growth_median = Median - Unsupplemented_growth_median,
    Adjusted_growth_min = Min - Unsupplemented_growth_median,
    Adjusted_growth_max = Max - Unsupplemented_growth_median
  ) %>%
  # Replace MMC data for 2743 with repeat
  dplyr::filter(!((`Growth condition` == "Mitomycin C" | `Growth condition` == "No supplement") & Background == "italic(recA)*'::Janus'" & phage == "phi*'08B02743'")) %>%
  dplyr::filter(!((`Growth condition` == "Mitomycin C" | `Growth condition` == "No supplement") & Background == "'Wild type'" & phage == "phi*'08B02743'")) %>%
  dplyr::mutate(phage = dplyr::if_else(phage == "phi*'08B02743_MMC_repeat'","phi*'08B02743'",phage)) %>%
  # Replace MMC data for RMV4 with repeat
  dplyr::filter(!(`Growth condition` == "Mitomycin C" & Background == "'prophage::Janus'" & phage == "phi*'RMV4'")) %>%
  dplyr::filter(!(`Growth condition` == "No supplement" & Background == "'prophage::Janus_rerun'" & phage == "phi*'RMV4'")) %>%
  dplyr::mutate(Background = dplyr::if_else(Background == "'prophage::Janus_rerun'","'prophage::Janus'",Background)) %>%
  # Reorder
  dplyr::mutate(Background = factor(Background,
                                    levels = c("'Wild type'","italic(recA)*'::Janus'","'prophage::Janus'"))) %>%
  dplyr::mutate(phage = factor(phage,
                                    levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]")))

stimuli_mutants_raw_growth.df <-
  dplyr::bind_rows(
    # immA KO
    get_growth_curve_df(read.csv("./growth_curves/2743_immAKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(immA)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Genotype = "'08B02743'"),
    # rep KO
    get_growth_curve_df(read.csv("./growth_curves/2743_repKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(rep)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Genotype = "'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_repKO_growth_metals.csv")) %>%
      dplyr::mutate(Background = "italic(rep)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Genotype = "'RMV8'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_repKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(rep)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Genotype = "'RMV4'"),
    # int KO
    get_growth_curve_df(read.csv("./growth_curves/2743_intKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(int)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Genotype = "'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_intKO_growth_metals.csv")) %>%
      dplyr::mutate(Background = "italic(int)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Genotype = "'RMV8'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_intcat_metals.csv")) %>%
      dplyr::mutate(Background = "italic(int)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Genotype = "'RMV4'"),
    # lyt KO
    get_growth_curve_df(read.csv("./growth_curves/2743_lytAKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(lyt)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Genotype = "'08B02743'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV8_lytAKO_growth_2_metals.csv")) %>%
      dplyr::mutate(Background = "italic(lyt)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV8'") %>%
      dplyr::mutate(Genotype = "'RMV8'"),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_lytKO_metals.csv")) %>%
      dplyr::mutate(Background = "italic(lyt)*'::'*italic(cat)") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Genotype = "'RMV4'")
  )

stimuli_mutants_induction.df <-
  stimuli_mutants_raw_growth.df %>%
  # Calculate relative growth
  dplyr::group_by(Genotype,phage,Time) %>%
  dplyr::mutate(
    Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
    Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
    Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
  ) %>%
  tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
              .direction = "updown") %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Adjusted_growth_median = Median - Unsupplemented_growth_median,
    Adjusted_growth_min = Min - Unsupplemented_growth_median,
    Adjusted_growth_max = Max - Unsupplemented_growth_median
  ) %>%
  dplyr::bind_rows(
    growth_comparison_induction.df# %>%
      #dplyr::filter(Background == "'Wild type'")
  ) %>%
  dplyr::mutate(Background = factor(Background,
                                  levels = c("'Wild type'",
                                             "italic(recA)*'::Janus'",
                                             "italic(int)*'::'*italic(cat)",
                                             "italic(immA)*'::'*italic(cat)",
                                             "italic(rep)*'::'*italic(cat)",
                                             "italic(lyt)*'::'*italic(cat)",
                                             "'prophage::Janus'"))) %>%
  dplyr::mutate(phage = factor(phage,
                                    levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]")))

g <- guide_legend(nrow = 2,
                    title = NULL)

# plot
stimuli_mutants_induction_plot <-
  ggplot(stimuli_mutants_induction.df,
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = `Growth condition`,
               fill = `Growth condition`)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
              panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(Background~phage,
                 scales = 'free_x',
                   labeller = label_parsed)

stimuli_mutants_induction_plot

```

# Role of RecA in activation and lysis

``` {r RecA in activation and lysis}

process_growth_data <- function(df) {
  df %>%
    dplyr::filter(`Growth condition` %in% c("No supplement","Mitomycin C") & Background %in% c("'Wild type'","italic(recA)*'::Janus'","italic(int)*'::'*italic(cat)")) %>%
    dplyr::mutate(Background = factor(Background,
                                      levels = c("'Wild type'","italic(recA)*'::Janus'","italic(int)*'::'*italic(cat)"))) %>%
    dplyr::mutate(Genotype = factor(gsub("'","",Genotype),
                                    levels = c("RMV8","08B02743","RMV4","RMV8_Znregcat")))
}

raw_recA_growth_df <-
  process_growth_data(dplyr::bind_rows(
                        raw_growth_comparison_induction.df %>%
                          dplyr::filter(!(phage == "phi*'08B02743'")) %>% # remove in favour of re-run with fresh MMC
                          dplyr::mutate(phage = dplyr::if_else(phage == "phi*'08B02743_MMC_repeat'","phi*'08B02743'", phage)),
                        stimuli_mutants_raw_growth.df# %>%
                      )
    )

recA_growth.df <-
  raw_recA_growth_df %>%
    dplyr::filter(`Growth condition` == "Mitomycin C") %>%
    dplyr::mutate(phage = dplyr::if_else(phage == "phi*'08B02743_MMC_repeat'","phi*'08B02743'",phage)) %>%
    dplyr::group_by(Time,phage,Genotype) %>%
    dplyr::mutate(wild_type_growth = dplyr::if_else(Background=="'Wild type'",Median,NA_real_)) %>%
    tidyr::fill(wild_type_growth,.direction = "updown") %>%
    dplyr::mutate(mutation_effect = Median - wild_type_growth,
                  mutation_effect_min = Min - wild_type_growth,
                  mutation_effect_max = Max - wild_type_growth) %>%
    dplyr::filter(Background != "'Wild type'")

recA_activation.df <-
  process_growth_data(stimuli_mutants_induction.df) %>% 
    dplyr::filter(`Growth condition` == "Mitomycin C")

g <- guide_legend(ncol = 1,
                    title = NULL)

effect_of_mmc_compared_to_wt_plot <-
  ggplot(recA_growth.df,
           aes(x = Time,
               y = mutation_effect,
               ymin = mutation_effect_min,
               ymax = mutation_effect_max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression(atop('Growth relative to wild type','in mitomycin C ('*Delta*Delta*'OD'[600]*')'))) +
      scale_colour_manual(values = mutant_palette,
                          labels = mutant_labels[names(mutant_labels) %in% unique(recA_activation.df$Genotype)],
                        aesthetics = c("fill","colour")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "top",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
              panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(.~Background,
                 scales = 'free_x',
                   labeller = label_parsed)

effect_of_mmc_plot <-
  ggplot(recA_activation.df,
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression(atop('Effect of mitomycin C','on growth ('*Delta*'OD'[600]*')'))) +
      scale_colour_manual(values = mutant_palette,
                          labels = mutant_labels[names(mutant_labels) %in% unique(recA_activation.df$Genotype)],
                        aesthetics = c("fill","colour")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "top",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
              panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(.~Background,
                 scales = 'free_x',
                   labeller = label_parsed)

recA_activation_plot <-
  cowplot::plot_grid(
    plotlist = list(
      effect_of_mmc_plot + theme(legend.position = "none"),
        cowplot::plot_grid(
          plotlist = list(
            cowplot::get_plot_component(effect_of_mmc_plot + theme(legend.position = "left"), 'guide-box-bottom', return_all = TRUE),
            effect_of_mmc_compared_to_wt_plot + theme(legend.position = "none")
          ),
          rel_widths = c(0.3,0.7),
          nrow = 1,
          ncol = 2
        )
      ),
    nrow = 2
    )

```

We need the raw data plot as well:

``` {r Raw MMC effect plot}

(raw_effect_of_mmc_plot <-
  ggplot(raw_recA_growth_df %>%
           dplyr::mutate(`Growth condition` = factor(paste0("'",`Growth condition`,"'"),
                                            levels = c("'No supplement'","'Mitomycin C'"))),
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(atop('Effect of mitomycin C','on growth ('*Delta*'OD'[600]*')'))) +
      scale_colour_manual(values = mutant_palette,
                          labels = mutant_labels[names(mutant_labels) %in% unique(recA_activation.df$Genotype)],
                        aesthetics = c("fill","colour")) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "right",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
              panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(`Growth condition`~Background,
                 scales = 'free_x',
                   labeller = label_parsed)
)

```

# 08B02743 szaL::Janus mutant behaviour

Loss of szaL results in an increased sensitivity to MMC, which is RecA-dependent; this corresponds with increased activation of phage genes in response to MMC in the szaL gene. This is independent of the expression of comX or recA, both of which are only transcribed at low levels in 08B02743 regardless of whether or not szaL is present.

``` {r szaL growth phenotype, fig.height = 11, fig.width = 8}

immA_df <-
  dplyr::bind_rows(
      get_growth_curve_df(read.csv("./growth_curves/2743_immA_2nd_repeat_022024.csv")) %>%
                 dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(immA)*"::"*italic(cat)~""',
                          Background = "italic(recA)^'+'~italic(dprA)^'+'"),
      get_growth_curve_df(read.csv("./growth_curves/2743_immA_dprA_022024.csv")) %>%
            dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(dprA)*"::Janus"~italic(immA)*"::"*italic(cat)~""',
                          Background = "italic(recA)^'+'~italic(dprA)^'-'"),
      get_growth_curve_df(read.csv("./growth_curves/2743_immA_recA_metals_growth.csv")) %>%
            dplyr::mutate(Genotype = '""~phi*"08B02743"~italic(recA)*"::Janus"~italic(immA)*"::"*italic(cat)~""',
                          Background = "italic(recA)^'-'~italic(dprA)^'+'")
    ) %>%
    dplyr::group_by(Genotype,Time) %>%
    dplyr::mutate(Control = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_)) %>%
    tidyr::fill(Control,.direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(Adjusted_Median = Median - Control,
                  Adjusted_Min = Min - Control,
                  Adjusted_Max = Max - Control) %>%
  dplyr::mutate(phage = "atop(phi*'08B02743',italic(immA)*'::'*italic(cat))") %>%
  # Calculate relative growth
  dplyr::group_by(phage,Background,Time) %>%
  dplyr::mutate(
    Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
    Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
    Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
  ) %>%
  tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
              .direction = "updown") %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Adjusted_growth_median = Median - Unsupplemented_growth_median,
    Adjusted_growth_min = Min - Unsupplemented_growth_median,
    Adjusted_growth_max = Max - Unsupplemented_growth_median
  )

zn_cat_janus_growth.df <-
  dplyr::bind_rows(
    immA_df,
    growth_comparison_induction.df %>%
      dplyr::filter(phage == "phi*'08B02743'" & (Background == "'Wild type'" | Background == "italic(recA)*'::Janus'")) %>%
      dplyr::mutate(Background = dplyr::if_else(Background == "'Wild type'",
                                                "italic(recA)^'+'~italic(dprA)^'+'",
                                                "italic(recA)^'-'~italic(dprA)^'+'")),
    get_growth_curve_df(read.csv("./growth_curves/2743_zncat_metals.csv")) %>%
      dplyr::mutate(Genotype = "phi*'08B02743 szaL::cat'") %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'+'") %>%
      dplyr::mutate(phage = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))"),
    get_growth_curve_df(read.csv("./growth_curves/2743_zncat_recAKO_metals.csv")) %>%
      dplyr::mutate(Genotype = "'08B02743 szaL::cat'") %>%
      dplyr::mutate(phage = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))") %>%
      dplyr::mutate(Background = "italic(recA)^'-'~italic(dprA)^'+'"),
    
    get_growth_curve_df(read.csv("./growth_curves/04032024_2743_dprA_1st.csv") %>% dplyr::select(!dplyr::contains('_twice'))) %>%
      dplyr::mutate(Genotype = "'08B02743'") %>%
      dplyr::mutate(phage = "phi*'08B02743'") %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'-'"),
    
    get_growth_curve_df(read.csv("./growth_curves/04032024_2743_dprA_zncat_2nd.csv") %>% dplyr::select(!dplyr::contains('_twice'))) %>%
      dplyr::mutate(Genotype = "'08B02743 szaL::cat'") %>%
      dplyr::mutate(phage = "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))") %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'-'")
    
  ) %>%
  # Calculate relative growth
  dplyr::group_by(phage,Background,Time) %>%
  dplyr::mutate(
    Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
    Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
    Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
  ) %>%
  tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
              .direction = "updown") %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Adjusted_growth_median = Median - Unsupplemented_growth_median,
    Adjusted_growth_min = Min - Unsupplemented_growth_median,
    Adjusted_growth_max = Max - Unsupplemented_growth_median
  ) %>%
  dplyr::mutate(Background = factor(Background,
                                    levels = c("italic(recA)^'+'~italic(dprA)^'+'",
                                               "italic(recA)^'-'~italic(dprA)^'+'",
                                               "italic(recA)^'+'~italic(dprA)^'-'"))) %>%
  dplyr::mutate(phage = factor(phage,
                                    levels = c("phi*'08B02743'",
                                               "atop(phi*'08B02743',italic(immA)*'::'*italic(cat))",
                                               "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))")
                                    )
                )

g <- guide_legend(nrow = 1,
                    title = NULL)

(refined_szaL_growth_plot <-
  ggplot(zn_cat_janus_growth.df %>%
           dplyr::filter(`Growth condition` %in% c("No supplement","Mitomycin C","CSP")),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = `Growth condition`,
               fill = `Growth condition`,
               group = interaction(`Growth condition`,Background))) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      xlab("Time (h)") +
      scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines"),
          strip.text.y = element_text(size = 7.5)) +
      guides(colour = g, fill = g) +
      facet_grid(Background~phage,
                 #scales = 'free_y',
                 labeller = label_parsed)
)

```

## Compare expression in 2743 and szaL mutant

``` {r 2743 szaL gene expression comparison}

szaL_expression_comparison_df <-
  read_expression_file("./qPCR/2743vs2743zncat_lytA_rep_T120_August.csv") %>%
      dplyr::rename(Fold_change = fold_change_MMC) %>%
  dplyr::mutate(Time = factor(Time, levels = c("0","20","60","120")),
                Condition = factor(Condition, levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(Genotype = dplyr::if_else(
                      grepl("_zncat_",read.csv("./qPCR/2743vs2743zncat_lytA_rep_T120_August.csv",header=TRUE) %>%
                      dplyr::select(mutants) %>%
                      dplyr::pull()),
                      "phi*'08B02743'",
                      "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))"
                  )
                ) %>%
  dplyr::filter(gene != "rpoA") %>%
  dplyr::mutate(gene = dplyr::if_else(gene=="lytA","lyt",gene)) %>%
  dplyr::mutate(gene = paste0("italic(",gene,")")) %>% 
  dplyr::group_by(Condition,gene,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::mutate(gene = factor(gene,
                              levels = c("italic(rep)","italic(lyt)"))) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c("phi*'08B02743'",
                                              "atop(phi*'08B02743',italic(szaL)*'::'*italic(cat))")))


(szaL_expression_comparison_plot <-
    ggplot(szaL_expression_comparison_df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Condition,gene,Time,Genotype))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      geom_vline(xintercept = 1.5, linetype = 3) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      geom_hline(yintercept = 1, linetype = 2) +
      theme(legend.position = "right",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed)
  )

```

## Effect of SzaL on competence induction

``` {r Plot effect of SzaL on competence induction}

szaL_effect_on_com_df <-
  dplyr::bind_rows(
    read_dprA_excision_file("./qPCR/RMV8_comX_transcriptions.csv"),
    read_dprA_excision_file("./qPCR/2743vs2743zncat_comX.csv"),
    read_dprA_excision_file("./qPCR/RMV8vsRMV8hybrid_recA_transcriptions.csv"),
    read_dprA_excision_file("./qPCR/2743vs_2743zncat_recA_transcirptions.csv")
  ) %>%
  dplyr::mutate(
    Genotype = 
      dplyr::case_when(
        grepl("hybrid",mutants) ~ '""~phi*"RMV8"[italic(immARcat)]~""',
        grepl("zncat",mutants) ~ '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~""',
        grepl("RMV8",mutants) ~ '""~phi*"RMV8"~""',
        grepl("2743",mutants) ~ '""~phi*"08B02743"~""'
      )
  ) %>%
  dplyr::mutate(gene = paste0('italic(',gene,')')) %>% 
  dplyr::group_by(Condition,gene,Time,Genotype) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP"))) %>%
  dplyr::mutate(gene = factor(gene,
                              levels = c("italic(comX)","italic(recA)"))) %>%
  dplyr::mutate(Genotype = factor(Genotype,
                                  levels = c('""~phi*"08B02743"~""',
                                             '""~phi*"08B02743"~italic(szaL)*"::"*italic(cat)~""',
                                             '""~phi*"RMV8"~""',
                                             '""~phi*"RMV8"[italic(immARcat)]~""'
                                             )
                                  )
  )

(szaL_effect_on_com_plot <-
    ggplot(szaL_effect_on_com_df,
           aes(x = Time,
               y = fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Condition,gene,Time,Genotype))
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      geom_vline(xintercept = 1.5, linetype = 3) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      geom_hline(yintercept = 1, linetype = 2) +
      theme(legend.position = "none",
            panel.spacing = unit(0, "lines")) +
      facet_grid(gene~Genotype,
                 scales = 'free_y',
                 labeller = label_parsed)
  )

```

## Regulation of RMV4 lysogeny

``` {r spaL growth phenotype, fig.height = 11, fig.width = 8}

spaL_growth.df <-
  dplyr::bind_rows(
    growth_comparison_induction.df %>%
      dplyr::filter(phage == "phi*'RMV4'" & (Background == "'Wild type'" | Background == "italic(recA)*'::'*italic(cat)" | Background == "italic(recA)*'::Janus'")) %>%
      dplyr::mutate(Background = dplyr::if_else(Background == "'Wild type'",
                                                "italic(recA)^'+'~italic(dprA)^'+'",
                                                "italic(recA)^'-'~italic(dprA)^'+'")
                    ),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_spaL_repeat_metals.csv")) %>%
      dplyr::mutate(Genotype = "phi*'RMV4 spaL::cat'") %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'+'") %>%
      dplyr::mutate(phage = 'phi*"RMV4"~italic(spaL)*"::"*italic(cat)'),
    get_growth_curve_df(read.csv("./growth_curves/RMV4_spaL_recA_repeat.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4 spaL::cat'") %>%
      dplyr::mutate(phage = 'phi*"RMV4"~italic(spaL)*"::"*italic(cat)') %>%
      dplyr::mutate(Background = "italic(recA)^'-'~italic(dprA)^'+'"),
    
    #get_growth_curve_df(read.csv("RMV4_metals_dprAKO.csv")) %>%
    get_growth_curve_df(read.csv("./growth_curves/RMV4_1st_dprA_repeat_July_2024.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4'") %>%
      dplyr::mutate(phage = "phi*'RMV4'") %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'-'"),
    
    get_growth_curve_df(read.csv("./growth_curves/RMV4_spaL_dprA_metals.csv")) %>%
      dplyr::mutate(Genotype = "'RMV4 spaL::cat'") %>%
      dplyr::mutate(phage = 'phi*"RMV4"~italic(spaL)*"::"*italic(cat)') %>%
      dplyr::mutate(Background = "italic(recA)^'+'~italic(dprA)^'-'")
    
  ) %>%
  # Calculate relative growth
  dplyr::group_by(phage,Background,Time) %>%
  dplyr::mutate(
    Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
    Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
    Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
  ) %>%
  tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
              .direction = "updown") %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Adjusted_growth_median = Median - Unsupplemented_growth_median,
    Adjusted_growth_min = Min - Unsupplemented_growth_median,
    Adjusted_growth_max = Max - Unsupplemented_growth_median
  ) %>%
  dplyr::mutate(Background = factor(Background,
                                    levels = c("italic(recA)^'+'~italic(dprA)^'+'",
                                               "italic(recA)^'-'~italic(dprA)^'+'",
                                               "italic(recA)^'+'~italic(dprA)^'-'"))) %>%
  dplyr::mutate(phage = factor(phage,
                                    levels = c("phi*'RMV4'",
                                               'phi*"RMV4"~italic(spaL)*"::"*italic(cat)')
                                    )
                )

g <- guide_legend(nrow = 2,
                    title = NULL)

# plot
(refined_spaL_growth_plot <-
  ggplot(spaL_growth.df %>%
           dplyr::filter(`Growth condition` %in% c("No supplement","Mitomycin C","CSP")),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = `Growth condition`,
               fill = `Growth condition`,
               group = interaction(`Growth condition`,Background))) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      xlab("Time (h)") +
      scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines"),
          strip.text.y = element_text(size = 7.5)) +
      guides(colour = g, fill = g) +
      facet_grid(Background~phage,
                 #scales = 'free_y',
                 labeller = label_parsed)
)

(spaL_growth_plot <-
  ggplot(spaL_growth.df,
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = `Growth condition`,
               fill = `Growth condition`,
               group = interaction(`Growth condition`,Background))) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      xlab("Time (h)") +
      scale_colour_manual(values = condition_palette,
                        aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(Background~phage,
                 #scales = 'free_y',
                 labeller = label_parsed)
)

```

# RMV4 transcriptional activation

``` {r Transcriptional activation of RMV4 spaL mutant, fig.height=10}


RMV4_expression_df <-
  dplyr::bind_rows(
    phage_transcription.df %>%
      dplyr::filter(phage == "phi*'RMV4'" & gene != 'italic(reg)'),
    read_expression_file("./qPCR/RMV4_spaL_transcriptions_All.csv") %>%
        dplyr::mutate(phage = "RMV4 spaL") %>%
        dplyr::filter(gene != "rpoA") %>%
        dplyr::mutate(gene = dplyr::case_when(
                              gene == "lytA" ~ "lyt",
                              gene == "Int" ~ "int",
                              TRUE ~ gene
                      )
        ) %>%
        dplyr::mutate(gene = factor(paste0('italic(',gene,')'),
                                      levels = c("italic(recA)","italic(int)","italic(rep)","italic(lyt)")))
  ) %>%
  dplyr::mutate(Time = factor(Time,
                                levels = c("0","20","60","120"))) %>%
  dplyr::mutate(phage = factor(dplyr::case_when(
                  phage == "RMV4 spaL" ~ "phi*'RMV4'~italic('spaL')*'::'*italic(cat)",
                  phage == "RMV4" ~ "phi*'RMV4'",
                  TRUE ~ phage
                ),
                levels = c("phi*'RMV4'","phi*'RMV4'~italic('spaL')*'::'*italic(cat)"))
  ) %>%
  dplyr::mutate(Condition = factor(Condition,
                                levels = c("No supplement","CSP","Mitomycin C"))) %>%
  dplyr::group_by(Condition,gene,phage,Time) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup()


RMV4_expression_plot <-
   # Expression data
    ggplot(RMV4_expression_df,
           aes(x = Time,
               y = Fold_change,
               colour = Condition,
               fill = Condition,
               group = interaction(Time,Condition,phage,gene),
               )
           ) +
      geom_violin(draw_quantiles = c(0.5),
                  colour = "black",
                  scale = "width",
                  alpha = 0.15,
                  adjust = 1,
                  position = position_dodge(preserve = 'single')) +
      geom_point(aes(shape = `Biological replicate`),
                 position = position_jitterdodge(),
                 size = 0.75) +
      geom_vline(xintercept = c(seq(from=1.5,to=3.5,by=1)), linetype = 3) +
      geom_hline(yintercept = 1, linetype = 2) +
      scale_y_continuous(trans = "log10") +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
      xlab("Time (min)") +
      facet_grid(gene~phage, scales = 'free_y', labeller = label_parsed) +
      theme(legend.position = "right",
            panel.spacing = unit(0, "lines"))

cowplot::plot_grid(
  plotlist = list(
    cowplot::plot_grid(  
      plotlist = list(
        cowplot::plot_grid(
          plotlist = list(
            szaL_expression_comparison_plot + theme(legend.position = "none"),
            cowplot::get_plot_component(RMV4_expression_plot, 'guide-box-right', return_all = TRUE)
          ),
          nrow = 2,
          ncol = 1,
          labels = c("A","")
        ),
        RMV4_expression_plot + theme(legend.position = "none")
      ),
      labels = c("","C"),
      ncol = 2
    ),
    szaL_effect_on_com_plot + theme(legend.position = "none")
  ),
  nrow = 2,
  rel_heights = c(0.7,0.3),
  labels = c("","B")
)

```

# Rates of activation

``` {r Rates of activation of all phage}

rates_of_activation.df <-
  dplyr::bind_rows(
    dplyr::bind_rows(
      # RMV8 and RMV8 hybrid
      get_growth_curve_df(
        read.csv("./growth_curves/20min_MMC_treatment.csv") %>%
                            rename_with(~stringr::str_replace(., "_1MMC", "_5ng_µl_MMC")) %>%
                            rename_with(~stringr::str_replace(., "_2MMC", "_10ng_µl_MMC"))
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "20 min"),
      get_growth_curve_df(
        read.csv("./growth_curves/60min_MMC_treatment.csv") %>%
                            rename_with(~stringr::str_replace(., "_1MMC", "_5ng_µl_MMC")) %>%
                            rename_with(~stringr::str_replace(., "_2MMC", "_10ng_µl_MMC"))
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "60 min"),
      get_growth_curve_df(
        read.csv("./growth_curves/2hr_MMC_treatment.csv") %>%
                            rename_with(~stringr::str_replace(., "_1MMC", "_5ng_µl_MMC")) %>%
                            rename_with(~stringr::str_replace(., "_2MMC", "_10ng_µl_MMC"))
      ) %>%
        dplyr::mutate(Time = Time + 2,
                      Exposure = "120 min"),
      # 2743
      get_growth_curve_df(
        read.csv("./growth_curves/2743_20min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "20 min") %>%
        dplyr::mutate(Genotype = "08B02743"),
      get_growth_curve_df(
        read.csv("./growth_curves/2743_60min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "60 min") %>%
        dplyr::mutate(Genotype = "08B02743"),
      get_growth_curve_df(
        read.csv("./growth_curves/2743_120min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "120 min") %>%
        dplyr::mutate(Genotype = "08B02743"),
      # RMV4
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_20min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "20 min") %>%
        dplyr::mutate(Genotype = "RMV4"),
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_60min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "60 min") %>%
        dplyr::mutate(Genotype = "RMV4"),
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_120min_MMC_treatment.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "120 min") %>%
        dplyr::mutate(Genotype = "RMV4"),
      # High concentration, continuous exposure
      get_growth_curve_df(
        read.csv("./growth_curves/2743_high_MMC.csv")
      ) %>%
        dplyr::mutate(Exposure = "Continuous") %>%
        dplyr::mutate(Genotype = "08B02743"),
      # High concentration, continuous exposure
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_high_MMC.csv")
      ) %>%
        dplyr::mutate(Exposure = "Continuous") %>%
        dplyr::mutate(Genotype = "RMV4"),
      get_growth_curve_df(
        read.csv("./growth_curves/RMV8_high_MMC.csv")
      ) %>%
        dplyr::mutate(Exposure = "Continuous") %>%
        dplyr::mutate(Genotype = "RMV8"),
      get_growth_curve_df(
        read.csv("./growth_curves/RMV8_hybrid_high_MMC.csv")
      ) %>%
        dplyr::mutate(Exposure = "Continuous") %>%
        dplyr::mutate(Genotype = "RMV8_Znregcat")
    ) %>%
    dplyr::group_by(Time,Genotype,Exposure) %>%
    dplyr::mutate(
      Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
      Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
      Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
    ) %>%
    tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
                .direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(
      Adjusted_growth_median = Median - Unsupplemented_growth_median,
      Adjusted_growth_min = Min - Unsupplemented_growth_median,
      Adjusted_growth_max = Max - Unsupplemented_growth_median
    ) %>%
      dplyr::mutate(phage = dplyr::case_when(Genotype == "RMV4" ~ "phi*'RMV4'",
                                                Genotype == "RMV8" ~ "phi*'RMV8'",
                                                Genotype == "08B02743" ~ "phi*'08B02743'",
                                                Genotype == "RMV8_Znregcat" ~ "phi*'RMV8'[italic(immARcat)]"
                                                )
    )
    ,
    growth_comparison_induction.df %>%
      dplyr::mutate(Genotype = dplyr::if_else(phage == "phi*'RMV8'[italic(immARcat)]","RMV8_Znregcat",gsub("'","",Genotype))) %>%
      dplyr::filter(MMC == "Mitomycin C" & Background == "'Wild type'") %>%
      dplyr::mutate(MMC = "Low mitomycin C") %>%
      dplyr::mutate(Exposure = "Continuous")
  ) %>%
  dplyr::mutate(Exposure = factor(paste0("'",Exposure,"'"),
                                  levels = c("'20 min'","'60 min'","'120 min'","'Continuous'"))) %>%
  dplyr::mutate(MMC = factor(MMC,
                            levels = c("No mitomycin C","Low mitomycin C","High mitomycin C"))) %>%
  dplyr::mutate(MMC_expression = factor(paste0("'",MMC,"'"),
                                  levels = c("'No mitomycin C'","'Low mitomycin C'","'High mitomycin C'"))) %>%
  dplyr::mutate(phage = factor(phage,
                               levels = c("phi*'RMV8'","phi*'08B02743'","phi*'RMV4'","phi*'RMV8'[italic(immARcat)]")))

# Split data by pulse v continuous
mmc_pulse.df <-
  rates_of_activation.df %>%
    dplyr::filter(Exposure != "'Continuous'")

mmc_continuous.df <-
  rates_of_activation.df %>%
    dplyr::filter(Exposure == "'Continuous'")

# guide legend
g <- guide_legend(nrow = 1,
                    title = NULL)

# plot
(mmc_plot <-
  ggplot(mmc_pulse.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = MMC,
               fill = MMC)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_grid(Exposure~phage,
                 scales = 'free_x',
                 labeller = label_parsed)
)

```

# Pulse effect by regulator

``` {r MMC pulse effects}

# plot
(continuous_mmc_growth_plot <-
  ggplot(mmc_continuous.df %>%
           dplyr::filter(`Growth condition` != "No supplement") %>%
           dplyr::mutate(MMC = factor(paste0("'",MMC,"'"),
                                      levels = c("'Low mitomycin C'","'High mitomycin C'"))) %>%
           dplyr::mutate(Genotype = factor(Genotype,
                                           levels = c("RMV8","08B02743","RMV4","RMV8_Znregcat"))),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = Genotype,
               fill = Genotype)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      geom_hline(yintercept = 0, linetype = 3) +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(labels = mutant_labels[names(mutant_labels) %in% unique(rates_of_activation.df$Genotype)],
                        values = mutant_palette,
                        aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(.~MMC,
                 scales = 'free_x',
                 labeller = label_parsed)
)

```

# Pulse effect by time

``` {r MMC pulse effects by time}

time_palette <-
  c(
    "20 min" = "steelblue1",
    "60 min" = "steelblue2",
    "120 min" = "steelblue3",
    "Continuous" = as.character(condition_palette[names(condition_palette)=="MMC"])
  )

g <- guide_legend(nrow = 1,
                    title = NULL)

# plot
mmc_pulse_by_time_growth_plot <-
  ggplot(mmc_pulse.df %>%
           dplyr::filter(`Growth condition` != "No supplement") %>%
           dplyr::mutate(Exposure = factor(gsub("'","",Exposure),
                         levels = names(time_palette))),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = Exposure,
               fill = Exposure)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      xlab("Time (h)") +
      scale_colour_manual(values = time_palette,
                      aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(phage~MMC_expression,
                 scales = 'free_x',
                 labeller = label_parsed) +
   geom_hline(yintercept = 0, linetype = 3)

```

# Pulse effect on regulatory region

``` {r Pulse effect on regulatory region}

operator_mmc_pulse.df <-
  dplyr::bind_rows(
    dplyr::bind_rows(
      # RMV4
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_noncoding_KO_20min_MMC_treatment.csv") %>%
                            rename_with(~stringr::str_replace(., "_(\\d)MMC", "_MMC_replicate_\\1"))
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "'20 min'") %>%
        dplyr::mutate(phage = "phi*'RMV4 operator mutant'"),
      get_growth_curve_df(
        read.csv("./growth_curves/RMV4_noncoding_KO_60min_MMC_treatment.csv") %>%
                            rename_with(~stringr::str_replace(., "_(\\d)MMC", "_MMC_replicate_\\1"))
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "'60 min'") %>%
        dplyr::mutate(phage = "phi*'RMV4 operator mutant'")
    ) %>%
    dplyr::group_by(Time,Genotype,Exposure) %>%
    dplyr::mutate(
      Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
      Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
      Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
    ) %>%
    tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
                .direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(
      Adjusted_growth_median = Median - Unsupplemented_growth_median,
      Adjusted_growth_min = Min - Unsupplemented_growth_median,
      Adjusted_growth_max = Max - Unsupplemented_growth_median
    ),
    rates_of_activation.df %>%
      dplyr::filter(Genotype == "RMV4" & Exposure %in% c("'20 min'","'60 min'"))
  ) %>%
  dplyr::mutate(MMC = factor(paste0("'",MMC,"'"),
                             levels = c("'Low mitomycin C'","'High mitomycin C'")))

# plot
(operator_mmc_pulse_plot <-
  ggplot(operator_mmc_pulse.df %>%
           dplyr::filter(`Growth condition` != "No supplement") %>%
           dplyr::mutate(Exposure = factor(gsub("'","",Exposure))),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = Exposure,
               fill = Exposure)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      scale_colour_manual(values = time_palette,
                      aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(MMC~phage,
                 scales = 'free_x',
                 labeller = label_parsed)
)

```

# RMV4 rates of activation

``` {r Rates of activation of RMV4}

RMV4_rates_of_activation.df <-
  dplyr::bind_rows(
    dplyr::bind_rows(
      # RMV4
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_spaL_2_20min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "20 min") %>%
        dplyr::mutate(Genotype = "RMV4_spaL"),
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_spaL_2_60min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "60 min") %>%
        dplyr::mutate(Genotype = "RMV4_spaL"),
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_spaL_2_120min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "120 min") %>%
        dplyr::mutate(Genotype = "RMV4_spaL"),
      
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_proKO_20min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1/3,
                      Exposure = "20 min") %>%
        dplyr::mutate(Genotype = "RMV4_prophage_KO"),
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_proKO_2_60min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "60 min") %>%
        dplyr::mutate(Genotype = "RMV4_prophage_KO"),
      get_growth_curve_df(
        read.csv("./growth_curves/19032024_RMV4_proKO_120min.csv")
      ) %>%
        dplyr::mutate(Time = Time + 1,
                      Exposure = "120 min") %>%
        dplyr::mutate(Genotype = "RMV4_prophage_KO")
    ) %>%
    dplyr::group_by(Time,Genotype,Exposure) %>%
    dplyr::mutate(
      Unsupplemented_growth_median = dplyr::if_else(`Growth condition` == "No supplement",Median,NA_real_),
      Unsupplemented_growth_min = dplyr::if_else(`Growth condition` == "No supplement",Min,NA_real_),
      Unsupplemented_growth_max = dplyr::if_else(`Growth condition` == "No supplement",Max,NA_real_)
    ) %>%
    tidyr::fill(c(Unsupplemented_growth_median,Unsupplemented_growth_min,Unsupplemented_growth_max),
                .direction = "updown") %>%
    dplyr::ungroup() %>%
    dplyr::mutate(
      Adjusted_growth_median = Median - Unsupplemented_growth_median,
      Adjusted_growth_min = Min - Unsupplemented_growth_median,
      Adjusted_growth_max = Max - Unsupplemented_growth_median
    ) %>%
      dplyr::mutate(phage = dplyr::case_when(Genotype == "RMV4" ~ "phi*'RMV4'",
                                                Genotype == "RMV4_spaL" ~ "phi*'RMV4'~italic(spaL)*'::Janus'",
                                                Genotype == "RMV4_prophage_KO" ~ "phi*'RMV4'*'::Janus'"
                                                )
    ) %>%
  dplyr::mutate(Exposure = factor(paste0("'",Exposure,"'"),
                                  levels = c("'20 min'","'60 min'","'120 min'","'Continuous'"))) %>%
  dplyr::mutate(MMC = factor(MMC,
                            levels = c("No mitomycin C","Low mitomycin C","High mitomycin C"))) %>%
  dplyr::mutate(MMC_expression = factor(paste0("'",MMC,"'"),
                                  levels = c("'No mitomycin C'","'Low mitomycin C'","'High mitomycin C'"))) 
    ,
    rates_of_activation.df %>%
      dplyr::filter(Genotype == "RMV4" & Exposure != "'Continuous'")
  )

# Split data by pulse v continuous
RMV4_mmc_pulse.df <-
  RMV4_rates_of_activation.df %>%
    dplyr::filter(Exposure != "'Continuous'")

# guide legend
g <- guide_legend(nrow = 1,
                    title = NULL)

# plot
(RMV4_mmc_plot <-
  ggplot(RMV4_mmc_pulse.df,
           aes(x = Time,
               y = Median,
               ymin = Min,
               ymax = Max,
               colour = MMC,
               fill = MMC)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression(OD[600])) +
      xlab("Time (h)") +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      scale_colour_manual(values = condition_palette,
                      aesthetics = c("fill","colour")) +
      guides(colour = g, fill = g) +
      facet_grid(Exposure~phage,
                 scales = 'free_x',
                 labeller = label_parsed)
)

```

# RMV4 pulse effect by time for RMV4

``` {r MMC pulse effects by time for RMV4}

g <- guide_legend(nrow = 1,
                    title = NULL)

# plot
(rmv4_mmc_pulse_by_time_growth_plot <-
  ggplot(RMV4_mmc_pulse.df %>%
           dplyr::filter(`Growth condition` != "No supplement") %>%
           dplyr::mutate(Exposure = factor(gsub("'","",Exposure),
                         levels = names(time_palette))),
           aes(x = Time,
               y = Adjusted_growth_median,
               ymin = Adjusted_growth_min,
               ymax = Adjusted_growth_max,
               colour = Exposure,
               fill = Exposure)) +
      geom_ribbon(alpha = 0.5, linetype = 0) +
      geom_line() +
      ylab(expression('Effect on growth ('*Delta*'OD'[600]*')')) +
      xlab("Time (h)") +
      scale_colour_manual(values = time_palette,
                      aesthetics = c("fill","colour")) +
      theme_bw() +
      theme(legend.position = "bottom",
            legend.text = element_text(size = 8),
            strip.background = element_blank(),
            panel.spacing = unit(0, "lines")) +
      guides(colour = g, fill = g) +
      facet_grid(MMC_expression~phage,
                 scales = 'free_x',
                 labeller = label_parsed) +
   geom_hline(yintercept = 0, linetype = 3)
)

```

# Comparisons of prophage

``` {r Comparison of prophage annotations}

##################
# Input parameters
# Files
annotation_file_names<-list(
  "./annotations/phiRMV8.embl",
  "./annotations/phiARI0460-2.embl",
  "./annotations/phiRMV4.embl"
)

comparison_file_names<-list(
  "./annotations/phiRMV8.dna.phiARI0460-2.dna.crunch",
  "./annotations/phiARI0460-2.dna.phiRMV4.dna.crunch"
)

# Names
names<-c("phiRMV8","phiARI460-2","phiRMV4")
seq_labels<-c(expression(atop(phi*'RMV8',italic(attB)[OXC])),
              expression(atop(phi*'08B02743',italic(attB)[MM1])),
              expression(atop(phi*'RMV4',italic(attB)[MM1]))
)

# Seq ID
min_seq_id<-75
max_seq_id<-100
range_bounds <- round(seq(from=min_seq_id,to=max_seq_id,length.out=10))
reds <- c("#FFF5F0", "#FEE0D2", "#FCBBA1", "#FC9272", 
          "#FB6A4A", "#EF3B2C", "#CB181D", "#A50F15", "#67000D")
##################
library(genoPlotR)
library(vcd)
library(magrittr)
library(tidyverse)
source("modified_genoplotR.R")

# Read sequences
dna_sequences <- lapply(annotation_file_names,
                        read_dna_seg_from_embl,
                        tagsToParse = c("CDS"),
                        extra_fields = c("note"),
                        cex = 1,
                        lwd = 0.5)

prophage_palette <- list(
  "phiRMV8" = c("lytic" = "darkgreen",
                "structural" = "forestgreen",
                "replication" = "green",
                "lysogeny" = "lightgreen"
  ),
  "phiARI460-2" = c("lytic" = "navy",
                    "structural" = "blue",
                    "replication" = "cornflowerblue",
                    "lysogeny" = "cyan"
                    ),
  "phiRMV4" = c("lytic" = "darkred",
                    "structural" = "red3",
                    "replication" = "red",
                    "lysogeny" = "tomato"
  )
)

# Style of CDSs
for (i in 1:length(dna_sequences)) {
  dna_sequences[[i]]$gene_type[dna_sequences[[i]]$feature=="CDS"]<-"side_blocks"
  for (j in 1:nrow(dna_sequences[[i]])) {
    dna_sequences[[i]]$col[j]<-"black"
    dna_sequences[[i]]$fill[j]<-prophage_palette[[names[i]]][dna_sequences[[i]]$note[j]]
  }
}

# Annotate segments
annotations<-list()
for (i in 1:length(dna_sequences)) {
    annotations[[i]]<-auto_annotate(dna_sequences[[i]],
                                    locus_tag_pattern=NULL,
                                    names=dna_sequences[[i]]$gene,
                                    rot=45)
}

top_annotation <-
  as.annotation(
    dna_sequences[[1]] %>%
      dplyr::group_by(note) %>%
      dplyr::summarise(x1 = min(start)+100) %>%
      dplyr::left_join(
        dna_sequences[[1]] %>%
          dplyr::group_by(note) %>%
          dplyr::summarise(x2 = max(end)-100),
        by = c("note")
      ) %>%
      dplyr::rename(text = note) %>%
      dplyr::mutate(text = paste0(" ",text)) %>%
      dplyr::mutate(
        color = "black",
        rot = 45
      ) %>%
      dplyr::select(x1,x2,text,color,rot) %>%
      dplyr::mutate(text = dplyr::case_when(
          text == " lytic" ~ " lysis",
          text == " structural" ~ " structure",
          TRUE ~ text
        )
      )
  )

top_bracket_annotation <-
  top_annotation
top_bracket_annotation$text<-""

top_text_annotation <-
  top_annotation
top_text_annotation$x1<-0.5*(top_text_annotation$x1+top_text_annotation$x2)-1000
top_text_annotation$x2<-NA

annotations[[1]] <-
  dplyr::bind_rows(
    top_bracket_annotation,
    top_text_annotation
  )

# Calculate offsets
seq_lengths <- unlist(lapply(dna_sequences, function(x) {max(x$end)}))
max_length <- max(seq_lengths)
offset_lengths <- unlist(lapply(seq_lengths,function(x) {(max_length - x)/2}))

# Read comparison
comparison_files<-list()
comparison_colours.df <-
  data.frame(
    lb = range_bounds[1:9],
    ub = range_bounds[2:10],
    col = reds
  )

for (comparison in comparison_file_names) {
  
  ref.query.comparison.df<-read.table(comparison, comment.char="")
  colnames(ref.query.comparison.df) <- c("name1", "name2", "per_id", "aln_len", "mism", 
                                         "gaps", "start1", "end1", "start2", "end2", "e_value", 
                                         "bit_score")
  ref.query.comparison.df <- ref.query.comparison.df[, c(colnames(ref.query.comparison.df)[c(7:10, 1:6, 11:12)])]
  ref.query.comparison.df<-ref.query.comparison.df[order(ref.query.comparison.df$aln_len),]
  ref.query.comparison.actOrder<-as.comparison(ref.query.comparison.df)
  
  # Reverse comparison file from ACT order
  ref.query.comparison<-ref.query.comparison.actOrder
  ref.query.comparison$start1<-ref.query.comparison.actOrder$start2
  ref.query.comparison$end1<-ref.query.comparison.actOrder$end2
  ref.query.comparison$start2<-ref.query.comparison.actOrder$start1
  ref.query.comparison$end2<-ref.query.comparison.actOrder$end1
  
  # Add colour
  ref.query.comparison %<>%
    dplyr::mutate(col = dplyr::case_when(
                      per_id >= 75 & per_id < 78 ~ "#FFF5F0",
                      per_id >= 78 & per_id < 81 ~ "#FEE0D2",
                      per_id >= 81 & per_id < 83 ~ "#FCBBA1",
                      per_id >= 83 & per_id < 86 ~ "#FC9272",
                      per_id >= 86 & per_id < 89 ~ "#FB6A4A",
                      per_id >= 89 & per_id < 92 ~ "#EF3B2C",
                      per_id >= 92 & per_id < 94 ~ "#CB181D",
                      per_id >= 94 & per_id < 97 ~ "#A50F15",
                      per_id >= 97 & per_id <= 100 ~ "#67000D"
                    )
                  )
  
  # Store
  comparison_files[[length(comparison_files)+1]]<-ref.query.comparison
}

# Plot
(prophage_comparison_plot <-
    grid.grabExpr(
      plot_gene_map(dna_sequences,
                    comparisons = comparison_files,
                    annotations = annotations,
                    seg_height = 0.75,
                    annotation_height = 0.0,
                    annotation_cex = 0.75,
                    dna_seg_label_cex = 0.75,
                    dna_seg_labels = seq_labels,
                    offsets = offset_lengths,
                    top_annotation_padding = 0.25,
                    annotation_y_val = 0.5,
                    top_annotation_banner = 3.0,
                    plot_new = TRUE
      ),
    wrap.grobs = TRUE
    )
)

```
# Comparison legend

``` {r Comparison legend}

legend_text<-NULL
for (i in 1:9) {
  legend_text<-c(legend_text,
                 paste0(range_bounds[i],"-",range_bounds[i+1]))
}

sequence_legend.df <-
  tibble::tibble(
      "Sequence identity (%)" = legend_text,
      "reds" = c("#FFF5F0","#FEE0D2","#FCBBA1","#FC9272","#FB6A4A","#EF3B2C","#CB181D","#A50F15","#67000D"),
      "y" = 1
  )

seq_pal <- sequence_legend.df$reds
names(seq_pal) <- sequence_legend.df$ID

guide_nrow = 1
g <- guide_legend(direction = "horizontal",
                  ncol = 3,
                  title.position = "top")

seq_id_legend <-
  ggplot(sequence_legend.df,
         aes(x = `Sequence identity (%)`,
             y = y,
             colour = `Sequence identity (%)`,
             fill = `Sequence identity (%)`)) +
    geom_col() +
    scale_colour_manual(values = seq_pal, aesthetics = c("fill","colour")) +
    guides(colour = g, fill = g) +
    theme(legend.position = "bottom")

```

# Lysogeny comparison

``` {r Lysogeny gene cluster comparisons}

##################
# Input parameters
# Files
annotation_file_names<-list(
  "./annotations/RMV8_lysogenic.embl",
  "./annotations/trimmed_RMV8_immARcat_lysogeny.embl",
  "./annotations/2743_lysogenic.embl",
  "./annotations/RMV4_lysogenic.embl"
)

comparison_file_names<-list(
  "./annotations/RMV8_lysogenic.dna.trimmed_RMV8_immARcat_lysogeny.dna.crunch",
  "./annotations/trimmed_RMV8_immARcat_lysogeny.dna.2743_lysogenic.dna.crunch",
  "./annotations/2743_lysogenic.dna.RMV4_lysogenic.dna.crunch"
)
output_fn<-"./annotations/lysogeny_comparison.pdf"
# Names
names<-c("phiRMV8","phiRMV8_immARcat","phiARI460-2","phiRMV4")
seq_labels<-c(expression(phi*'RMV8'),
              expression(atop(phi*'RMV8',""[italic(immARcat)])),
              expression(phi*'08B02743'),
              expression(phi*'RMV4')
)

# Seq ID
min_seq_id<-75
max_seq_id<-100
range_bounds <- round(seq(from=min_seq_id,to=max_seq_id,length.out=10))
reds <- c("#FFF5F0", "#FEE0D2", "#FCBBA1", "#FC9272", 
          "#FB6A4A", "#EF3B2C", "#CB181D", "#A50F15", "#67000D")
##################

# Read sequences
dna_sequences <- lapply(annotation_file_names,
                        read_dna_seg_from_embl,
                        tagsToParse = c("CDS"),
                        extra_fields = c("note"),
                        cex=0.75,lwd=0.25,lty=1,
                        col = "black")

prophage_palette <- list(
  "phiRMV8" = "lightgreen",
  "phiRMV8_immARcat" = "magenta",
  "phiARI460-2" = "cyan",
  "phiRMV4" = "tomato"
)

# Style of CDSs
for (i in 1:length(dna_sequences)) {
  dna_sequences[[i]]$gene_type[dna_sequences[[i]]$feature=="CDS"]<-"side_blocks"
  dna_sequences[[i]]$fill<-prophage_palette[[i]]
}

# Annotate segments
annotations<-list()
for (i in 1:length(dna_sequences)) {
  annotations[[i]]<-auto_annotate(dna_sequences[[i]],
                                  locus_tag_pattern=NULL,
                                  names=dna_sequences[[i]]$gene,
                                  rot=0)
}

# Calculate offsets
seq_lengths <- unlist(lapply(dna_sequences, function(x) {max(x$end)}))
max_length <- max(seq_lengths)
offset_lengths <- unlist(lapply(seq_lengths,function(x) {(max_length - x)/2}))

# Read comparison
comparison_files<-list()
comparison_colours.df <-
  data.frame(
    lb = range_bounds[1:9],
    ub = range_bounds[2:10],
    col = reds
  )

for (comparison in comparison_file_names) {
  
  ref.query.comparison.df<-read.table(comparison, comment.char="")
  colnames(ref.query.comparison.df) <- c("name1", "name2", "per_id", "aln_len", "mism", 
                                         "gaps", "start1", "end1", "start2", "end2", "e_value", 
                                         "bit_score")
  ref.query.comparison.df <- ref.query.comparison.df[, c(colnames(ref.query.comparison.df)[c(7:10, 1:6, 11:12)])]
  ref.query.comparison.df<-ref.query.comparison.df[order(ref.query.comparison.df$aln_len),]
  ref.query.comparison.actOrder<-as.comparison(ref.query.comparison.df)
  
  # Reverse comparison file from ACT order
  ref.query.comparison<-ref.query.comparison.actOrder
  ref.query.comparison$start1<-ref.query.comparison.actOrder$start2
  ref.query.comparison$end1<-ref.query.comparison.actOrder$end2
  ref.query.comparison$start2<-ref.query.comparison.actOrder$start1
  ref.query.comparison$end2<-ref.query.comparison.actOrder$end1
  
  # Add colour
  ref.query.comparison %<>%
    dplyr::mutate(col = dplyr::case_when(
      per_id >= 75 & per_id < 78 ~ "#FFF5F0",
      per_id >= 78 & per_id < 81 ~ "#FEE0D2",
      per_id >= 81 & per_id < 83 ~ "#FCBBA1",
      per_id >= 83 & per_id < 86 ~ "#FC9272",
      per_id >= 86 & per_id < 89 ~ "#FB6A4A",
      per_id >= 89 & per_id < 92 ~ "#EF3B2C",
      per_id >= 92 & per_id < 94 ~ "#CB181D",
      per_id >= 94 & per_id < 97 ~ "#A50F15",
      per_id >= 97 & per_id <= 100 ~ "#67000D"
    )
    )
  
  # Store
  comparison_files[[length(comparison_files)+1]]<-ref.query.comparison
}

# Adjust annotation spacing
annotations[[2]][2,1] <- annotations[[2]][2,1] + 100
annotations[[3]][2,1] <- annotations[[3]][2,1] + 100

# Formatting
for (i in 1:length(annotations)) {
  annotations[[i]]$text <- paste0("italic(",annotations[[i]]$text,")")
}

# Plot
(lysogeny_comparison_plot <-
    grid.grabExpr(
      plot_gene_map(dna_sequences,
                comparisons = comparison_files,
                annotations = annotations,
                seg_height = 0.75,
                annotation_height = 0.4,
                annotation_cex = 0.7,
                dna_seg_label_cex = 0.75,
                dna_seg_labels = seq_labels,
                offsets = offset_lengths,
                top_annotation_banner = 3,
                scale_cex = 0.5,
                plot_new = TRUE
      ),
    wrap.grobs = TRUE
    )
)

```

# *att* site right replichores

``` {r att site comparisons 1, fig.width = 8, fig.height = 11}


# Seq ID
min_seq_id<-75
max_seq_id<-100
range_bounds <- round(seq(from=min_seq_id,to=max_seq_id,length.out=10))
reds <- c("#FFF5F0", "#FEE0D2", "#FCBBA1", "#FC9272", 
          "#FB6A4A", "#EF3B2C", "#CB181D", "#A50F15", "#67000D")
att_path = "./att_figures"

process_input_files <- function(annotation_file_names,comparison_file_names, seq_labels, att_gene, site) {
  
  # Read sequences
  dna_sequences <- lapply(annotation_file_names,
                          read_dna_seg_from_embl,
                          tagsToParse = c("CDS","repeat_region"),
                          extra_fields = c("note","rpt_family"))
  
  
  # Style of CDSs
  for (i in 1:length(dna_sequences)) {
    dna_sequences[[i]] %<>%
      dplyr::filter(feature == "CDS" | rpt_family == "BOX element" | rpt_family == "RUP element")
    dna_sequences[[i]]$gene_type[dna_sequences[[i]]$feature=="CDS"]<-"blocks"
    for (j in 1:nrow(dna_sequences[[i]])) {
      dna_sequences[[i]]$fill[j]<-dna_sequences[[i]]$col[j]
      dna_sequences[[i]]$col[j]<-"black"
      dna_sequences[[i]]$lty[j]<-1
      dna_sequences[[i]]$lwd[j]<-0.25
      if (!is.na(dna_sequences[[i]]$rpt_family[j]) & dna_sequences[[i]]$rpt_family[j] == "BOX element") {
        dna_sequences[[i]]$fill[j] = "magenta"
        dna_sequences[[i]]$col[j] = "magenta"
      }
      if (!is.na(dna_sequences[[i]]$rpt_family[j]) & dna_sequences[[i]]$rpt_family[j] == "RUP element") {
        dna_sequences[[i]]$fill[j] = "purple"
        dna_sequences[[i]]$col[j] = "purple"
      }
    }
  }
  
  # Annotate segments
  annotations<-list()
  for (i in 1:length(dna_sequences)) {
    annotations[[i]]<-auto_annotate(dna_sequences[[i]],
                                    locus_tag_pattern=NULL,
                                    names=dna_sequences[[i]]$gene,
                                    rot=45)
  }
  
  annotations[[1]] <-
    as.annotation(
      dna_sequences[[1]] %>%
        dplyr::filter(gene == att_gene) %>%
        dplyr::mutate(x1 = 0.5*(start+end)-600,
                      x2 = NA) %>%
        dplyr::rename(text = name) %>%
        dplyr::mutate(text = paste0("~italic(",text,")")) %>%
        dplyr::mutate(
          color = "black",
          rot = 45
        ) %>%
        dplyr::select(x1,x2,text,color,rot)
    )
  
  annotations[[2]] %<>%
    dplyr::filter(text == "NONE")
  
  # Calculate offsets
  seq_lengths <- unlist(lapply(dna_sequences, function(x) {max(x$end)}))
  max_length <- max(seq_lengths)
  offset_lengths <- unlist(lapply(seq_lengths,function(x) {(max_length - x)/2}))
  
  # Read comparison
  comparison_files<-list()
  comparison_colours.df <-
    data.frame(
      lb = range_bounds[1:9],
      ub = range_bounds[2:10],
      col = reds
    )
  
  for (comparison in comparison_file_names) {
    
    ref.query.comparison.df<-read.table(comparison, comment.char="")
    colnames(ref.query.comparison.df) <- c("name1", "name2", "per_id", "aln_len", "mism", 
                                           "gaps", "start1", "end1", "start2", "end2", "e_value", 
                                           "bit_score")
    ref.query.comparison.df <- ref.query.comparison.df[, c(colnames(ref.query.comparison.df)[c(7:10, 1:6, 11:12)])]
    ref.query.comparison.df<-ref.query.comparison.df[order(ref.query.comparison.df$aln_len),]
    ref.query.comparison.actOrder<-as.comparison(ref.query.comparison.df)
    
    # Reverse comparison file from ACT order
    ref.query.comparison<-ref.query.comparison.actOrder
    ref.query.comparison$start1<-ref.query.comparison.actOrder$start2
    ref.query.comparison$end1<-ref.query.comparison.actOrder$end2
    ref.query.comparison$start2<-ref.query.comparison.actOrder$start1
    ref.query.comparison$end2<-ref.query.comparison.actOrder$end1
    
    # Add colour
    ref.query.comparison %<>%
      dplyr::mutate(col = dplyr::case_when(
        per_id >= 75 & per_id < 78 ~ "#FFF5F0",
        per_id >= 78 & per_id < 81 ~ "#FEE0D2",
        per_id >= 81 & per_id < 83 ~ "#FCBBA1",
        per_id >= 83 & per_id < 86 ~ "#FC9272",
        per_id >= 86 & per_id < 89 ~ "#FB6A4A",
        per_id >= 89 & per_id < 92 ~ "#EF3B2C",
        per_id >= 92 & per_id < 94 ~ "#CB181D",
        per_id >= 94 & per_id < 97 ~ "#A50F15",
        per_id >= 97 & per_id <= 100 ~ "#67000D"
      )
      )
    
    # Store
    comparison_files[[length(comparison_files)+1]]<-ref.query.comparison
  }

  # convert to gg object
  imported_comparison_figure <-
        grid.grabExpr(
          plot_gene_map(dna_sequences,
                        comparisons = comparison_files,
                        annotations = annotations,
                        annotation_height = 0.0,
                        annotation_cex = 1.0,
                        dna_seg_label_cex = 1.0,
                        dna_seg_labels = seq_labels,
                        gene_type  = "side_blocks",
                        seg_height = 0.75,
                        offsets = offset_lengths,
                        top_annotation_banner = 3.0
          ),
          wrap.grobs = TRUE
    )

  # add title
  imported_comparison_figure <-
    if (site %in% c("OXC","MM1")) {
      cowplot::plot_grid(plotlist = list(imported_comparison_figure),
                         nrow = 1,
                         ncol = 1) +
        ggtitle(bquote(italic("attB")[.(site)])) +
        theme(plot.title = element_text(hjust = 0.5))
    } else {
      cowplot::plot_grid(plotlist = list(imported_comparison_figure),
                         nrow = 1,
                         ncol = 1) +
        ggtitle(bquote(italic("attB")[italic(.(site))])) +
        theme(plot.title = element_text(hjust = 0.5))
    }
  
  # return
  return(imported_comparison_figure)
  
}


##################
# att_ccrB - 21999_7#175
ccrB_annotation_file_names<-list(
  paste0(att_path,"/ccrB_R6.embl"),
  paste0(att_path,"/ccrB_phage.embl")
)

ccrB_comparison_file_names<-list(
  paste0(att_path,"/ccrB_R6.dna.ccrB_phage.dna.crunch")
)

# Names
ccrB_seq_labels<-c(expression("R6"),
         expression("ERS1813719"))

ccrB_figure <- process_input_files(ccrB_annotation_file_names,
                                   ccrB_comparison_file_names,
                                   ccrB_seq_labels,
                                   "ccrB",
                                   "ccrB"
)

# att_hlpA - 22841_2#133
hlpA_annotation_file_names<-list(
  paste0(att_path,"/hlpA_R6.embl"),
  paste0(att_path,"/hlpA_phage.embl")
)

hlpA_comparison_file_names<-list(
  paste0(att_path,"/hlpA_R6.dna.hlpA_phage.dna.crunch")
)

# Names
hlpA_seq_labels<-c(expression("R6"),
              expression("ERS1699606"))

hlpA_figure <- process_input_files(hlpA_annotation_file_names,
                                   hlpA_comparison_file_names,
                                   hlpA_seq_labels,
                                   "hlpA",
                                   "hlpA"
)

# att_ssbB - 11893_1#48
ssbB_annotation_file_names<-list(
  paste0(att_path,"/ssbB_R6.embl"),
  paste0(att_path,"/ssbB_phage.embl")
)

ssbB_comparison_file_names<-list(
  paste0(att_path,"/ssbB_R6.dna.ssbB_phage.dna.crunch")
)

# Names
ssbB_seq_labels<-c(expression("R6"),
                   expression("ERS367616"))

ssbB_figure <- process_input_files(ssbB_annotation_file_names,
                                   ssbB_comparison_file_names,
                                   ssbB_seq_labels,
                                   "ssbB",
                                   "ssbB"
)

# att_ccnB - 22695_3#282
ccnB_annotation_file_names<-list(
  paste0(att_path,"/ccnB_R6.embl"),
  paste0(att_path,"/ccnB_phage.embl")
)

ccnB_comparison_file_names<-list(
  paste0(att_path,"/ccnB_R6.dna.ccnB_phage.dna.crunch")
)

# Names
ccnB_seq_labels<-c(expression("R6"),
                   expression("ERS1698859"))

ccnB_figure <- process_input_files(ccnB_annotation_file_names,
                                   ccnB_comparison_file_names,
                                   ccnB_seq_labels,
                                   "ccnB",
                                   "ccnB"
)

# att_cbpG - 15841_5#24
cbpG_annotation_file_names<-list(
  paste0(att_path,"/cbpG_R6.embl"),
  paste0(att_path,"/cbpG_phage.embl")
)

cbpG_comparison_file_names<-list(
  paste0(att_path,"/cbpG_R6.dna.cbpG_phage.dna.crunch")
)

# Names
cbpG_seq_labels<-c(expression("R6"),
                   expression("ERS638944"))

cbpG_figure <- process_input_files(cbpG_annotation_file_names,
                                   cbpG_comparison_file_names,
                                   cbpG_seq_labels,
                                   "cbpG",
                                   "cbpG"
)

# att_tgt - 13710_2#13
tgt_annotation_file_names<-list(
  paste0(att_path,"/tgt_R6.embl"),
  paste0(att_path,"/tgt_phage.embl")
)

tgt_comparison_file_names<-list(
  paste0(att_path,"/tgt_R6.dna.tgt_phage.dna.crunch")
)

# Names
tgt_seq_labels<-c(expression("R6"),
                   expression("ERS455132"))

tgt_figure <- process_input_files(tgt_annotation_file_names,
                                  tgt_comparison_file_names,
                                  tgt_seq_labels,
                                   "tgt",
                                  "tgt"
)

# att_ydiL - 19341_2#124
ydiL_annotation_file_names<-list(
  paste0(att_path,"/ydiL_R6.embl"),
  paste0(att_path,"/ydiL_phage.embl")
)

ydiL_comparison_file_names<-list(
  paste0(att_path,"/ydiL_R6.dna.ydiL_phage.dna.crunch")
)

# Names
ydiL_seq_labels<-c(expression("R6"),
                  expression("ERS1021109"))

ydiL_figure <- process_input_files(ydiL_annotation_file_names,
                                   ydiL_comparison_file_names,
                                   ydiL_seq_labels,
                                  "ydiL",
                                  "ydiL"
)

# att MM1 - ATCC700669
MM1_annotation_file_names<-list(
  paste0(att_path,"/MM1_R6.embl"),
  paste0(att_path,"/MM1_phage.embl")
)

MM1_comparison_file_names<-list(
  paste0(att_path,"/MM1_R6.dna.MM1_phage.dna.crunch")
)

# Names
MM1_seq_labels<-c(expression("R6"),
                  expression("FM211187"))

MM1_figure <- process_input_files(MM1_annotation_file_names,
                                  MM1_comparison_file_names,
                                  MM1_seq_labels,
                                  "trxB",
                                  "MM1"
)

# att OXC - OXC141
OXC_annotation_file_names<-list(
  paste0(att_path,"/OXC_R6.embl"),
  paste0(att_path,"/OXC_phage.embl")
)

OXC_comparison_file_names<-list(
  paste0(att_path,"/OXC_R6.dna.OXC_phage.dna.crunch")
)

# Names
OXC_seq_labels<-c(expression("R6"),
                  expression("FQ312027"))

OXC_figure <- process_input_files(OXC_annotation_file_names,
                                  OXC_comparison_file_names,
                                  OXC_seq_labels,
                                  "ccnC",
                                  "OXC"
)

# att comYC - 670-6B
comYC_annotation_file_names<-list(
  paste0(att_path,"/comYC_R6.embl"),
  paste0(att_path,"/comYC_phage.embl")
)

comYC_comparison_file_names<-list(
  paste0(att_path,"/comYC_R6.dna.comYC_phage.dna.crunch")
)

# Names
comYC_seq_labels<-c(expression("R6"),
                  expression("CP002176"))

comYC_figure <- process_input_files(comYC_annotation_file_names,
                                    comYC_comparison_file_names,
                                    comYC_seq_labels,
                                  "comYC",
                                  "comYC"
)

# att_scr - 21946_6#218
scr_annotation_file_names<-list(
  paste0(att_path,"/scr_R6.embl"),
  paste0(att_path,"/scr_phage.embl")
)

scr_comparison_file_names<-list(
  paste0(att_path,"/scr_R6.dna.scr_phage.dna.crunch")
)

# Names
scr_seq_labels<-c(expression("R6"),
                   expression("ERS1460212"))

scr_figure <- process_input_files(scr_annotation_file_names,
                                  scr_comparison_file_names,
                                  scr_seq_labels,
                                  "scr",
                                  "scr"
)

# pfbA_scr - 23164_8#100
pfbA_annotation_file_names<-list(
  paste0(att_path,"/pfbA_R6.embl"),
  paste0(att_path,"/pfbA_phage.embl")
)

pfbA_comparison_file_names<-list(
  paste0(att_path,"/pfbA_R6.dna.pfbA_phage.dna.crunch")
)

# Names
pfbA_seq_labels<-c(expression("R6"),
                  expression("ERS2527991"))

pfbA_figure <- process_input_files(pfbA_annotation_file_names,
                                   pfbA_comparison_file_names,
                                  pfbA_seq_labels,
                                  "pfbA",
                                  "pfbA"
)

# relA_scr - 18090_2#20
relA_annotation_file_names<-list(
  paste0(att_path,"/relA_R6.embl"),
  paste0(att_path,"/relA_phage.embl")
)

relA_comparison_file_names<-list(
  paste0(att_path,"/relA_R6.dna.relA_phage.dna.crunch")
)

# Names
relA_seq_labels<-c(expression("R6"),
                   expression("ERS740678"))

relA_figure <- process_input_files(relA_annotation_file_names,
                                   relA_comparison_file_names,
                                   relA_seq_labels,
                                   "relA",
                                   "relA"
)

# pyrG_scr - 21127_1#110
pyrG_annotation_file_names<-list(
  paste0(att_path,"/pyrG_R6.embl"),
  paste0(att_path,"/pyrG_phage.embl")
)

pyrG_comparison_file_names<-list(
  paste0(att_path,"/pyrG_R6.dna.pyrG_phage.dna.crunch")
)

# Names
pyrG_seq_labels<-c(expression("R6"),
                   expression("ERS1299787"))

pyrG_figure <- process_input_files(pyrG_annotation_file_names,
                                   pyrG_comparison_file_names,
                                   pyrG_seq_labels,
                                   "pyrG",
                                   "pyrG"
)

# att_yjbK - 17175_6#28
yjbK_annotation_file_names<-list(
  paste0(att_path,"/yjbK_R6.embl"),
  paste0(att_path,"/yjbK_phage.embl")
)

yjbK_comparison_file_names<-list(
  paste0(att_path,"/yjbK_R6.dna.yjbK_phage.dna.crunch")
)

# Names
yjbK_seq_labels<-c(expression("R6"),
                   expression("ERS708124"))

yjbK_figure <- process_input_files(yjbK_annotation_file_names,
                                   yjbK_comparison_file_names,
                                   yjbK_seq_labels,
                                   "yjbK",
                                   "yjbK"
)

# att_ply - 22841_1#183
ply_annotation_file_names<-list(
  paste0(att_path,"/ply_R6.embl"),
  paste0(att_path,"/ply_phage.embl")
)

ply_comparison_file_names<-list(
  paste0(att_path,"/ply_R6.dna.ply_phage.dna.crunch")
)

# Names
ply_seq_labels<-c(expression("R6"),
                   expression("ERS1699476"))

ply_figure <- process_input_files(ply_annotation_file_names,
                                   ply_comparison_file_names,
                                   ply_seq_labels,
                                   "ply",
                                  "ply"
)

# Make combined plots
(right_replichore_atts <-
  cowplot::plot_grid(
    plotlist = list(
      OXC_figure,
      scr_figure,
      ccnB_figure,
      ydiL_figure,
      cbpG_figure,
      pyrG_figure,
      ccrB_figure,
      yjbK_figure
    ),
    nrow = 4,
    ncol = 2
  )
)

```

# *att* site left replichores

``` {r left replichore att sites, fig.width = 8, fig.height = 11}

(left_replichore_atts <-
  cowplot::plot_grid(
    plotlist = list(
      tgt_figure,
      comYC_figure,
      ply_figure,
      ssbB_figure,
      pfbA_figure,
      relA_figure,
      MM1_figure,
      hlpA_figure
    ),
    nrow = 4,
    ncol = 2
  )
)

```

## Genetic experiments with CIPHR

``` {r Plot positions of primers on the CIPHR element}


##################
# Input parameters
# Files
annotation_file_names<-list(
  "./att_figures/23F_CIPHR.embl",
  "./att_figures/R6_CIPHR.embl"
)

comparison_file_names<-list(
  "./att_figures/23F_CIPHR.dna.R6_CIPHR.dna.crunch"
)

# Names
names<-c("ATCC 700669","R6")
seq_labels<-c(expression(atop('ATCC','700669')),
              expression(atop('','R6'))
)

# Seq ID
min_seq_id<-75
max_seq_id<-100
range_bounds <- round(seq(from=min_seq_id,to=max_seq_id,length.out=10))
reds <- c("#FFF5F0", "#FEE0D2", "#FCBBA1", "#FC9272", 
          "#FB6A4A", "#EF3B2C", "#CB181D", "#A50F15", "#67000D")
##################
library(genoPlotR)
library(vcd)
library(magrittr)
library(tidyverse)
source("modified_genoplotR.R")

# Read sequences
dna_sequences <- lapply(annotation_file_names,
                        read_dna_seg_from_embl,
                        tagsToParse = c("CDS","ncRNA"),
                        extra_fields = c("note"),
                        cex = 1,
                        lwd = 0.5)

# Style of CDSs
for (i in 1:length(dna_sequences)) {
  dna_sequences[[i]]$gene_type<-"side_blocks"
  for (j in 1:nrow(dna_sequences[[i]])) {
    dna_sequences[[i]]$fill[j]<-dna_sequences[[i]]$col[j]
    dna_sequences[[i]]$col[j]<-"black"
     #dna_sequences[[i]]$fill[j]<-prophage_palette[[names[i]]][dna_sequences[[i]]$note[j]]
  }
}

# Annotate segments
annotations<-list()
for (i in 1:length(dna_sequences)) {
    annotations[[i]]<-auto_annotate(dna_sequences[[i]],
                                    locus_tag_pattern=NULL,
                                    names=dna_sequences[[i]]$gene,
                                    rot=45)
}

top_annotation <-
  as.annotation(
    dna_sequences[[1]] %>%
      dplyr::group_by(gene) %>%
      dplyr::filter(gene != "-") %>%
      dplyr::summarise(x1 = 0.5*(start+end)-200) %>%
      dplyr::left_join(
        dna_sequences[[1]] %>%
          dplyr::group_by(gene),
        by = c("gene")
      ) %>%
      dplyr::mutate(x2 = x1) %>%
      dplyr::rename(text = gene) %>%
      dplyr::mutate(text = paste0("~italic(",text,")")) %>%
      dplyr::mutate(
        color = "black",
        rot = 45
      ) %>%
      dplyr::select(x1,x2,text,color,rot)
  )

# Calculate offsets
seq_lengths <- unlist(lapply(dna_sequences, function(x) {max(x$end)}))
max_length <- max(seq_lengths)
offset_lengths <- unlist(lapply(seq_lengths,function(x) {(max_length - x)/2}))

# Read comparison
comparison_files<-list()
comparison_colours.df <-
  data.frame(
    lb = range_bounds[1:9],
    ub = range_bounds[2:10],
    col = reds
  )

for (comparison in comparison_file_names) {
  
  ref.query.comparison.df<-read.table(comparison, comment.char="")
  colnames(ref.query.comparison.df) <- c("name1", "name2", "per_id", "aln_len", "mism", 
                                         "gaps", "start1", "end1", "start2", "end2", "e_value", 
                                         "bit_score")
  ref.query.comparison.df <- ref.query.comparison.df[, c(colnames(ref.query.comparison.df)[c(7:10, 1:6, 11:12)])]
  ref.query.comparison.df<-ref.query.comparison.df[order(ref.query.comparison.df$aln_len),]
  ref.query.comparison.actOrder<-as.comparison(ref.query.comparison.df)
  
  # Reverse comparison file from ACT order
  ref.query.comparison<-ref.query.comparison.actOrder
  ref.query.comparison$start1<-ref.query.comparison.actOrder$start2
  ref.query.comparison$end1<-ref.query.comparison.actOrder$end2
  ref.query.comparison$start2<-ref.query.comparison.actOrder$start1
  ref.query.comparison$end2<-ref.query.comparison.actOrder$end1
  
  # Add colour
  ref.query.comparison %<>%
    dplyr::mutate(col = dplyr::case_when(
                      per_id >= 75 & per_id < 78 ~ "#FFF5F0",
                      per_id >= 78 & per_id < 81 ~ "#FEE0D2",
                      per_id >= 81 & per_id < 83 ~ "#FCBBA1",
                      per_id >= 83 & per_id < 86 ~ "#FC9272",
                      per_id >= 86 & per_id < 89 ~ "#FB6A4A",
                      per_id >= 89 & per_id < 92 ~ "#EF3B2C",
                      per_id >= 92 & per_id < 94 ~ "#CB181D",
                      per_id >= 94 & per_id < 97 ~ "#A50F15",
                      per_id >= 97 & per_id <= 100 ~ "#67000D"
                    )
                  )
  
  # Store
  comparison_files[[length(comparison_files)+1]]<-ref.query.comparison
}

# Plot
(ciphr_comparison_plot <-
    grid.grabExpr(
      plot_gene_map(dna_sequences,
                    comparisons = comparison_files,
                    annotations = top_annotation,
                    seg_height = 0.75,
                    annotation_height = 0.0,
                    annotation_cex = 1,
                    dna_seg_label_cex = 1,
                    dna_seg_labels = seq_labels,
                    offsets = offset_lengths,
                    top_annotation_padding = 0.5,
                    annotation_y_val = 0.5,
                    top_annotation_banner = 3.0,
                    plot_new = TRUE
      ),
    wrap.grobs = TRUE
    )
)

# Add legend!

```

``` {r Assemble CIPHR figure}

ciphr_gel_image <-
  magick::image_ggplot(image_crop(image_trim((image_read_pdf("./gel_images/CIPHR_gel_image.pdf")))))

scRNA_pcr_df <-
  read.csv("./qPCR/R6x_modified_scRNA.csv") %>%
  dplyr::group_by(gene,mutants) %>%
  mutate(`Biological replicate` = as.character((row_number()-1)%/%3+1)) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(Genotype = dplyr::if_else(
                    mutants == "R6x",
                    '"R6"',
                    '"R6"~italic(scr)[italic(attL)]'
                  )
                ) %>%
  dplyr::mutate(primer = dplyr::case_when(
                    gene == 'Long' ~ '"Full"~italic(scr)',
                    gene == 'Long_modified' ~ '"Full"~italic(scr)[italic(attL)]',
                    gene == 'short' ~ 'atop("Conserved","region")'
                  )
                ) %>%
  dplyr::mutate(
    primer = factor(
      primer,
      levels = c(
        'atop("Conserved","region")',
        '"Full"~italic(scr)[italic(attL)]',
        '"Full"~italic(scr)'
      )
    )
  )

scr_df <-
  tibble::tibble(
    "gene" = c('italic(scr)[italic(attL)]','italic(scr)'),
    "start" = c(1,1),
    "end" = c(215,189)
  )

primer_binding_df <-
  dplyr::bind_rows(
    tibble::tibble(
      "primer" = c('atop(italic(scr),"left")','atop("Conserved","region")','atop("Full",italic(scr)[italic(attL)])','atop("Full",italic(scr))'),
      "start" = c(31,57,NA,158),
      "end" = c(14,78,NA,179),
      "gene" = c('italic(scr)')
    ),
    tibble::tibble(
    "primer" = c('atop(italic(scr),"left")','atop("Conserved","region")','atop("Full",italic(scr)[italic(attL)])','atop("Full",italic(scr))'),
    "start" = c(30, 57, 147, 185),
    "end" = c(13, 78, 164, 206),
    "gene" = c('italic(scr)[italic(attL)]')
  )
) %>%
  dplyr::rowwise() %>%
  dplyr::mutate(label_pos = min(start,end)) %>%
  dplyr::ungroup()

scr_primer_binding_plot <-
  ggplot(data = NULL,
         aes(x = start,
             xend = end,
             y = gene)) +
    geom_segment(data = scr_df, linewidth =6, colour = 'palegreen1') +
    geom_segment(data = primer_binding_df, arrow = arrow(length=unit(0.30,"cm"), ends="first", type = "closed"), colour = 'black') +
    geom_text(data = primer_binding_df,aes(label = primer, x = label_pos),nudge_y = 0.25,nudge_x = 10,parse = TRUE) +
    scale_y_discrete(labels = function(l) parse(text=l)) +
    xlab("Position (bp)") +
    ylab("Allele") +
    theme_bw()
  
scRNA_pcr_plot <-
  ggplot(scRNA_pcr_df,
         aes(x = primer,
             y = fold_change,
             shape = `Biological replicate`)) +
    geom_point(position = position_jitter(width = 0.1)) +
    ylab(expression('Relative RNA amount ('*2^{-Delta*Delta*'C'[t]}*')')) +
    xlab("Right primer") +
    scale_x_discrete(labels = function(l) parse(text=l)) +
    facet_grid(.~Genotype, labeller = label_parsed) +
    theme_bw() +
    theme(strip.background = element_blank(),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none")

cowplot::plot_grid(
  plotlist = list(
    ciphr_comparison_plot,
    ciphr_gel_image,
    scr_primer_binding_plot,
    scRNA_pcr_plot
  ),
  nrow = 2,
  labels = "AUTO"
)

```


# Assemble Figure 1

``` {r Figure 1, fig.height = 11, fig.width = 8}

(figure_1 <-
  cowplot::plot_grid(
    plotlist = list(
      # upper part
      cowplot::plot_grid(
        plotlist = list(
          regulator_by_site_plot,
          cowplot::plot_grid(
            plotlist = list(
              phage_by_isolate_plot,
              three_rna_plot
            ),
            labels = c("B","C"),
            nrow = 2,
            rel_widths = c(0.5,0.5)
          )
        ),
        labels = c("A",""),
        ncol = 2,
        rel_widths = c(0.5,0.5)
      ),
      # lower part
      cowplot::plot_grid(
        plotlist = list(
          cowplot::plot_grid(
            plotlist = list(
                    r6_scr_growth_plot + theme(legend.position = "none"),
                    ncrna_transformation_changes_plot + theme(legend.position = "none")
                  ),
                  labels = c("D","E"),
                  ncol = 2,
                  nrow = 1
                ),
                cowplot::get_plot_component(r6_scr_growth_plot, 'guide-box-bottom', return_all = TRUE)
            ),
          rel_heights = c(1,0.15),
          nrow = 2
      )
    ),
    nrow = 2
  )
)

ggsave("./figures/prophage_transformation_Figure_1.pdf",
       figure_1,
       dpi = 600,
       height = 11,
       width = 8)

```

# Assemble Figure 2

``` {r Figure 2, fig.height = 11, fig.width = 8}

(figure_2 <-
  cowplot::plot_grid(
    plotlist = list(
      cowplot::plot_grid(
        plotlist = list(
            prophage_comparison_plot,
            lysogeny_comparison_plot
          ),
          nrow = 1,
          labels = c("A","B"),
          rel_widths = c(0.55,0.45)
      ),
      cowplot::plot_grid(
        plotlist = list(
          cowplot::get_plot_component(phage_mutant_growth_plot, 'guide-box-bottom', return_all = TRUE),
          cowplot::get_plot_component(seq_id_legend, 'guide-box-bottom', return_all = TRUE)
        ),
        nrow = 1,
        rel_widths = c(0.65,0.35)
      ),
      cowplot::plot_grid(
        plotlist = list(
          phage_mutant_growth_plot + theme(legend.position = "none"),
          effect_of_mmc_plot + theme(legend.position = "none")
        ),
        nrow = 1,
        labels = c("C","D")
      ),
      deletion_plot + theme(axis.title.x = element_blank())
    ),
    labels = c("","","","E"),
    rel_heights = c(0.4,0.15,0.25,0.3),
    nrow = 4
  )
)

ggsave("./figures/prophage_transformation_Figure_2.pdf",
       figure_2,
       dpi = 600,
       height = 11,
       width = 8)

```

# Assemble Figure 3

``` {r Figure 3, fig.height = 11, fig.width = 8}

(figure_3 <-
  combined_expression_excision_plot)

ggsave("./figures/prophage_transformation_Figure_3.pdf",
       figure_3,
       dpi = 600,
       height = 11,
       width = 8)

```

# Assemble Figure 4

``` {r Figure 4, fig.height = 11, fig.width = 8}

combined_legend.df <-
  tibble::tibble(
    Condition = c("No supplement","CSP","Mitomycin C","CSP"),
    IPTG = c("IPTG","No IPTG","IPTG","No IPTG"),
    `Biological replicate` = c("1","2","3","4"),
    x = 1,
    y = 1
  ) %>%
  dplyr::mutate(
    Condition = factor(Condition,
                       levels = c("No supplement","CSP","Mitomycin C")),
    IPTG = factor(IPTG,
                  levels = c("No IPTG","IPTG")),
    `Biological replicate` = factor(`Biological replicate`,
                                    levels = c("1","2","3","4"))
  )

guide_nrow = 1
g <- guide_legend(direction = "vertical",
                  ncol = 3,
                  title.position = "top")
s <- guide_legend(direction = "vertical",
                  ncol = 4,
                  title.position = "top")

combined_legend_plot <-
  ggplot(combined_legend.df,
         aes(x = x,
             y = y,
             shape = `Biological replicate`,
             group = interaction(Condition,IPTG),
         linetype = IPTG,
         colour = Condition,
         fill = Condition)) +
  geom_line () + geom_point() +
  scale_colour_manual(values = condition_palette) +
  guides(colour = g, fill = g, linetype = g, shape = s) +
  theme(legend.position = "bottom")


combined_legend_obj <-
  cowplot::get_plot_component(combined_legend_plot, 'guide-box-bottom', return_all = TRUE)

(figure_4 <-
  cowplot::plot_grid(
    plotlist = list(excision_levels_plot,
                    cowplot::plot_grid(
                      plotlist = list(
                          recA_standardised_wt_plot + theme(legend.position = "none"),
                          iptg_growth_plot + theme(legend.position = "none")
                        ),
                      nrow = 1,
                      rel_widths = c(0.6,0.4),
                      labels = c("B","C")
                    ),
                    cowplot::plot_grid(
                      plotlist = list(comEA_phage_induction_plot + theme(legend.position = "none"),
                                      comEA_phage_excision_plot + theme(legend.position = "none")
                                      ),
                      labels = c("D","E"),
                      nrow = 1
                    ),
                    combined_legend_obj
                  ),
    labels = c("A","","",""),
    nrow = 4,
    rel_heights = c(0.275,0.35,0.375,0.1)
  )
)

ggsave("./figures/prophage_transformation_Figure_4.pdf",
       figure_4,
       dpi = 600,
       height = 11,
       width = 8)

```

# Assembly Figure 5

``` {r Figure 5, fig.height = 11, fig.width = 8}

# RMV4_expression_plot
# 

(figure_5 <-
  cowplot::plot_grid(
    plotlist = list(
      cowplot::plot_grid(
        plotlist = list(refined_szaL_growth_plot + theme(legend.position = "none"),
                        dprA_standardised_wt_plot + theme(legend.position = "none")
                        ),
        labels = "AUTO",
        nrow = 1,
        rel_widths = c(0.6,0.4)
      ),
      cowplot::plot_grid(
        plotlist = list(
          phage_excision_dprA_plot + theme(legend.position = "none")
        ),
        labels = c("C")
      ),
      cowplot::plot_grid(
        plotlist = list(refined_spaL_growth_plot + theme(legend.position = "none"),
                        phage_excision_spaL_plot + theme(legend.position = "none")
                        ),
        labels = c("D","E"),
        nrow = 1,
        rel_widths = c(0.45,0.55)
      ),
      cowplot::get_plot_component(phage_excision_dprA_plot, 'guide-box-bottom', return_all = TRUE)
    ),
    nrow = 4,
    rel_heights = c(0.3,0.3,0.3,0.1)
  )
)

ggsave("./figures/prophage_transformation_Figure_5.pdf",
       figure_5,
       dpi = 600,
       height = 11,
       width = 8)

```

# Assemble Figure 6

``` {r Figure 6, fig.height = 8, fig.width = 8}

prophage_summary_plot <-
  magick::image_ggplot(image_crop(image_trim((image_read_pdf("./summary_figure/prophage_transformation_summary_figure.pdf")))))

(figure_6 <-
  cowplot::plot_grid(
    plotlist = list(
      mmc_pulse_by_time_growth_plot,
      prophage_summary_plot
    ),
    nrow = 1,
    ncol = 2,
    labels = "AUTO",
    rel_widths = c(0.4,0.6)
  )
)

ggsave("./figures/prophage_transformation_Figure_6.pdf",
       figure_6,
       dpi = 600,
       height = 6,
       width = 8)

```